// @deriv-com/quill-ui/dist/components/SegmentedControl/single-choice/index.js
import { j as p } from "../../../jsx-runtime-DmTjM30f.js";
import h, { useEffect as x } from "react";
import { c as S } from "../../../clsx-OuTLNxxd.js";
import { SegmentedControl as j } from "../base.js";
const w = ({
  className: i,
  options: n = [],
  onChange: o,
  hasContainerWidth: u,
  selectedItemIndex: r = 0,
  size: d = "md"
}) => {
  const [l, f] = h.useState(n), a = (e, t) => {
    const c = t < e.length ? t : 0;
    f([
      ...e.map((s, m) => ({
        ...s,
        selected: !s.disabled && m === c
      }))
    ]);
  }, g = (e) => {
    a(l, e), o == null || o(e);
  };
  return x(() => {
    var s;
    const e = l.findIndex((m) => m.selected), t = e !== -1 ? e : 0, c = (s = l[r]) != null && s.disabled ? t : r;
    a(
      n,
      r === e ? t : c
    );
  }, [n, r]), n.length ? /* @__PURE__ */ p.jsx(
    j,
    {
      className: S(
        "segmented-control-single",
        `segmented-control-single--${d}`,
        u && "segmented-control-single--has-container-width",
        i
      ),
      hasAnimation: !0,
      options: l,
      onChange: g,
      size: d
    }
  ) : null;
};
export {
  w as SegmentedControlSingleChoice,
  w as default
};
