// @deriv-com/quill-ui/dist/components/Button/social/index.js
import { j as n } from "../../../jsx-runtime-DmTjM30f.js";
import * as e from "react";
import { forwardRef as f } from "react";
import { c as w } from "../../../clsx-OuTLNxxd.js";
import { ButtonSize as g, loaderIcons as x } from "../base/index.js";
import "../../Typography/base.js";
import "../../Typography/heading/index.js";
import { Text as B } from "../../Typography/text/index.js";
import { s as d } from "../../../index-CQbz1p6_.js";
import '../../../assets/index21.css';const M = ({ iconSize: t = "md", title: a, titleId: l, ...c }, o) => e.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", fill: "none", viewBox: "0 0 32 32", ...d[t], role: "img", ref: o, "aria-labelledby": l, ...c },
  a ? e.createElement("title", { id: l }, a) : null,
  e.createElement(
    "g",
    { fill: "#000", clipPath: "url(#a9d15f9ba4e6d01d9944fb3b81462dcf__a)" },
    e.createElement("path", { d: "M22.248 0c.195 1.894-.542 3.749-1.627 5.129-1.125 1.34-2.907 2.405-4.69 2.248-.233-1.815.66-3.746 1.667-4.93C18.721 1.066 20.66.08 22.248 0M18.064 8.672c1.158-.464 2.59-1.038 4.165-.943 1.006.08 3.904.395 5.76 3.185l-.06.041c-.476.317-3.378 2.248-3.342 6.057.036 4.537 3.678 6.238 4.16 6.463l.053.026-.01.032c-.092.3-.72 2.36-2.153 4.49-1.315 1.97-2.668 3.896-4.833 3.935-1.02.02-1.706-.28-2.422-.592-.75-.326-1.53-.667-2.757-.667-1.286 0-2.104.351-2.891.69-.68.29-1.336.572-2.25.61-2.087.077-3.67-2.087-4.985-4.053-2.707-3.973-4.755-11.21-1.971-16.088C5.88 9.42 8.354 7.887 11.02 7.847c1.164-.022 2.278.426 3.25.816.739.297 1.396.562 1.93.562.485 0 1.118-.254 1.864-.553" })
  ),
  e.createElement(
    "defs",
    null,
    e.createElement(
      "clipPath",
      { id: "a9d15f9ba4e6d01d9944fb3b81462dcf__a" },
      e.createElement("path", { fill: "#fff", d: "M0 0h32v32H0z" })
    )
  )
), y = f(M), H = ({ iconSize: t = "md", title: a, titleId: l, ...c }, o) => e.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", fill: "none", viewBox: "0 0 32 32", ...d[t], role: "img", ref: o, "aria-labelledby": l, ...c },
  a ? e.createElement("title", { id: l }, a) : null,
  e.createElement(
    "g",
    { fill: "#fff", clipPath: "url(#1fc97e5c6da17f810ade6bed350dfe50__a)" },
    e.createElement("path", { d: "M22.248 0c.195 1.894-.542 3.749-1.627 5.129-1.125 1.34-2.907 2.405-4.69 2.248-.233-1.815.66-3.746 1.667-4.93C18.721 1.066 20.66.08 22.248 0M18.064 8.672c1.158-.464 2.59-1.038 4.165-.943 1.006.08 3.904.395 5.76 3.185l-.06.041c-.476.317-3.378 2.248-3.342 6.057.036 4.537 3.678 6.238 4.16 6.463l.053.026-.01.032c-.092.3-.72 2.36-2.153 4.49-1.315 1.97-2.668 3.896-4.833 3.935-1.02.02-1.706-.28-2.422-.592-.75-.326-1.53-.667-2.757-.667-1.286 0-2.104.351-2.891.69-.68.29-1.336.572-2.25.61-2.087.077-3.67-2.087-4.985-4.053-2.707-3.973-4.755-11.21-1.971-16.088C5.88 9.42 8.354 7.887 11.02 7.847c1.164-.022 2.278.426 3.25.816.739.297 1.396.562 1.93.562.485 0 1.118-.254 1.864-.553" })
  ),
  e.createElement(
    "defs",
    null,
    e.createElement(
      "clipPath",
      { id: "1fc97e5c6da17f810ade6bed350dfe50__a" },
      e.createElement("path", { fill: "#fff", d: "M0 0h32v32H0z" })
    )
  )
), S = f(H), R = ({ iconSize: t = "md", title: a, titleId: l, ...c }, o) => e.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", fill: "none", viewBox: "0 0 32 32", ...d[t], role: "img", ref: o, "aria-labelledby": l, ...c },
  a ? e.createElement("title", { id: l }, a) : null,
  e.createElement(
    "g",
    { clipPath: "url(#ec88df304ef3a5aa5b00f6294a6153c7__a)" },
    e.createElement("path", { fill: "#1877F2", fillRule: "evenodd", d: "M16 32c8.837 0 16-7.163 16-16S24.837 0 16 0 0 7.163 0 16s7.163 16 16 16m2.5-11.249v11.055a16.1 16.1 0 0 1-5 0V20.75H9.438v-4.653H13.5V12.55c0-4.034 2.389-6.263 6.043-6.263 1.751 0 3.582.315 3.582.315v3.961h-2.018c-1.987 0-2.607 1.241-2.607 2.514v3.02h4.438l-.71 4.653z", clipRule: "evenodd" }),
    e.createElement("path", { fill: "#fff", d: "M18.5 20.751v11.055a16.1 16.1 0 0 1-5 0V20.75H9.438v-4.653H13.5V12.55c0-4.034 2.389-6.263 6.043-6.263 1.751 0 3.582.315 3.582.315v3.961h-2.018c-1.987 0-2.607 1.241-2.607 2.514v3.02h4.438l-.71 4.653z" })
  ),
  e.createElement(
    "defs",
    null,
    e.createElement(
      "clipPath",
      { id: "ec88df304ef3a5aa5b00f6294a6153c7__a" },
      e.createElement("path", { fill: "#fff", d: "M0 0h32v32H0z" })
    )
  )
), C = f(R), F = ({ iconSize: t = "md", title: a, titleId: l, ...c }, o) => e.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", fill: "none", viewBox: "0 0 32 32", ...d[t], role: "img", ref: o, "aria-labelledby": l, ...c },
  a ? e.createElement("title", { id: l }, a) : null,
  e.createElement(
    "g",
    { clipPath: "url(#7cef8aa016b044abbb02bcc44ea9b488__a)" },
    e.createElement("path", { fill: "#fff", fillRule: "evenodd", d: "M16 32c8.837 0 16-7.163 16-16S24.837 0 16 0 0 7.163 0 16s7.163 16 16 16m2.5-11.249v11.055a16.1 16.1 0 0 1-5 0V20.75H9.438v-4.653H13.5V12.55c0-4.034 2.389-6.263 6.043-6.263 1.751 0 3.582.315 3.582.315v3.961h-2.018c-1.987 0-2.607 1.241-2.607 2.514v3.02h4.438l-.71 4.653z", clipRule: "evenodd" }),
    e.createElement("path", { fill: "#1877F2", d: "M18.5 20.751v11.055a16.1 16.1 0 0 1-5 0V20.75H9.438v-4.653H13.5V12.55c0-4.034 2.389-6.263 6.043-6.263 1.751 0 3.582.315 3.582.315v3.961h-2.018c-1.987 0-2.607 1.241-2.607 2.514v3.02h4.438l-.71 4.653z" })
  ),
  e.createElement(
    "defs",
    null,
    e.createElement(
      "clipPath",
      { id: "7cef8aa016b044abbb02bcc44ea9b488__a" },
      e.createElement("path", { fill: "#fff", d: "M0 0h32v32H0z" })
    )
  )
), j = f(F), P = ({ iconSize: t = "md", title: a, titleId: l, ...c }, o) => e.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", fill: "none", viewBox: "0 0 32 32", ...d[t], role: "img", ref: o, "aria-labelledby": l, ...c },
  a ? e.createElement("title", { id: l }, a) : null,
  e.createElement(
    "g",
    { clipPath: "url(#fbe6a261e5b0889a4a51a17750ecf085__a)" },
    e.createElement("path", { fill: "#3E82F1", d: "M32 16.375c0-1.097-.1-2.194-.294-3.273H16.325v6.186h8.787a7.34 7.34 0 0 1-3.256 4.829l5.274 4.02C30.22 25.348 32 21.248 32 16.374" }),
    e.createElement("path", { fill: "#32A753", d: "M21.858 24.119c-1.458.962-3.33 1.529-5.53 1.529-4.257 0-7.852-2.815-9.136-6.6l-5.458 4.145c2.77 5.404 8.42 8.82 14.593 8.812 4.412 0 8.108-1.43 10.805-3.876z" }),
    e.createElement("path", { fill: "#F9BB00", d: "M7.19 12.966 1.735 8.82a15.77 15.77 0 0 0 0 14.377l5.457-4.145a9.35 9.35 0 0 1 0-6.087" }),
    e.createElement("path", { fill: "#E74133", d: "M16.327 0C10.154 0 4.503 3.417 1.734 8.82l5.457 4.146c1.284-3.786 4.88-6.6 9.136-6.6 2.394 0 4.55.81 6.237 2.392l4.687-4.595C24.426 1.583 20.73 0 16.327 0" })
  ),
  e.createElement(
    "defs",
    null,
    e.createElement(
      "clipPath",
      { id: "fbe6a261e5b0889a4a51a17750ecf085__a" },
      e.createElement("path", { fill: "#fff", d: "M0 0h32v32H0z" })
    )
  )
), V = f(P), m = {
  md: {
    width: 24,
    height: 24
  },
  lg: {
    width: 24,
    height: 24
  },
  xl: {
    width: 32,
    height: 32
  }
}, z = {
  md: "social-button__icon-only--md",
  lg: "social-button__icon-only--lg",
  xl: "social-button__icon-only--xl"
}, $ = f(
  ({
    size: t = "md",
    variant: a = "primary",
    social: l = "google",
    hideLabel: c = !1,
    disabled: o = !1,
    fullWidth: i,
    isLoading: r = !1,
    className: p,
    ...s
  }, h) => {
    const b = a === "primary" ? `social-button__variant--${l}--primary` : "social-button__variant--secondary", u = l.charAt(0).toUpperCase() + l.slice(1), E = t === "md" ? "sm" : t === "lg" ? "md" : "xl", v = `social-button__variant--${l}--${a}-color`, _ = x[t];
    return /* @__PURE__ */ n.jsxs(
      "button",
      {
        className: w(
          "quill-button",
          b,
          p,
          i && "quill-button__full-width",
          !i && c ? z[t] : g[t],
          { ...s }
        ),
        ...s,
        disabled: o,
        ref: h,
        children: [
          l === "google" && !r && /* @__PURE__ */ n.jsx(V, { ...m[t] }),
          l === "facebook" && !r && (a === "primary" ? /* @__PURE__ */ n.jsx(
            j,
            {
              ...m[t],
              fill: "#fff"
            }
          ) : /* @__PURE__ */ n.jsx(
            C,
            {
              ...m[t],
              fill: "#fff"
            }
          )),
          l === "apple" && !r && a === "secondary" ? /* @__PURE__ */ n.jsx(y, { ...m[t] }) : l === "apple" && !r && /* @__PURE__ */ n.jsx(S, { ...m[t] }),
          r && !o && /* @__PURE__ */ n.jsx(
            _,
            {
              "data-testid": "button-loader",
              className: "quill-button__loader-icon"
            }
          ),
          !c && !r && /* @__PURE__ */ n.jsx(B, { size: E, bold: !0, color: v, children: u })
        ]
      }
    );
  }
);
$.displayName = "SocialButton";
export {
  $ as SocialButton,
  z as SocialButtonSize,
  $ as default,
  m as socialButtonIconSize
};
