// @deriv-com/quill-ui/dist/components/Input/text-field-addon/index.js
import { j as l } from "../../../jsx-runtime-DmTjM30f.js";
import { forwardRef as x } from "react";
import _ from "../base/index.js";
import { c as e } from "../../../clsx-OuTLNxxd.js";
import "../../Typography/base.js";
import "../../Typography/heading/index.js";
import { Text as u } from "../../Typography/text/index.js";
import '../../../assets/index3.css';const w = x(
  ({
    addOnPosition: r = "left",
    variant: o = "outline",
    fillAddonBorderColor: t,
    rightIcon: n,
    addonLabel: i,
    inputSize: d = "lg",
    disabled: s,
    ...a
  }, m) => {
    const c = o === "fill" ? { borderColor: t } : {}, p = /* @__PURE__ */ l.jsx(
      "div",
      {
        className: e(
          "quill-addon",
          `quill-addon__position--${r}`,
          `quill-addon__position--${r}--${o}`,
          `quill-addon__variant--${o}`
        ),
        style: c,
        children: /* @__PURE__ */ l.jsx(
          u,
          {
            size: d === "lg" ? "md" : d,
            className: e(s && "quill-addon__label--disabled"),
            children: i
          }
        )
      }
    ), f = /* @__PURE__ */ l.jsx("span", { className: "icon_wrapper", children: n });
    return /* @__PURE__ */ l.jsx(
      _,
      {
        addOn: p,
        disabled: s,
        addOnIcon: f,
        variant: o,
        inputSize: d,
        ref: m,
        ...a
      }
    );
  }
);
export {
  w as TextFieldAddon,
  w as default
};
