// @deriv-com/quill-ui/dist/components/Tab/list/index.js
import { j as n } from "../../../jsx-runtime-DmTjM30f.js";
import { c as s } from "../../../clsx-OuTLNxxd.js";
import { useContext as u, useRef as h, useEffect as b } from "react";
import { TabContext as x } from "../container/index.js";
import '../../../assets/index13.css';const W = ({ children: c, className: r }) => {
  const { activeTab: o, contentStyle: a } = u(x), i = h(null);
  return b(() => {
    const t = i.current;
    if (t) {
      const e = t.querySelectorAll("button")[o];
      if (e) {
        const l = t.clientWidth, m = e.clientWidth, f = e.offsetLeft - l / 2 + m / 2;
        t.scrollTo({
          left: f,
          behavior: "smooth"
        });
      }
    }
  }, [o]), /* @__PURE__ */ n.jsx("div", { className: s("tab-list--container", r), children: /* @__PURE__ */ n.jsx(
    "div",
    {
      ref: i,
      className: s(
        "tab-list--item",
        a === "hug" && "tab-list--item-hug-content"
      ),
      role: "tablist",
      "aria-orientation": "horizontal",
      children: c
    }
  ) });
};
export {
  W as TabList
};
