// @deriv-com/ui/dist/components/AppLayout/index.js
import { AccountSwitcher as e } from "./AccountSwitcher/index.js";
import { DerivLogo as m } from "./DerivLogo/index.js";
import { Drawer as p } from "./Drawer/index.js";
import { Footer as a } from "./Footer/index.js";
import { Header as c } from "./Header/index.js";
import { MenuItem as u } from "./MenuItem/index.js";
import { Wrapper as w } from "./Wrapper/index.js";
import { Notifications as s } from "./Notifications/index.js";
import { PlatformSwitcher as S } from "./PlatformSwitcher/index.js";
import { PlatformSwitcherItem as L } from "./PlatformSwitcher/PlatformSwitcherItem.js";
import { DesktopLanguagesModal as b } from "./LanguagesSwitcher/DesktopLanguagesModal.js";
import { MobileLanguagesDrawer as I } from "./LanguagesSwitcher/MobileLanguagesDrawer.js";
import { Submenu as k } from "./Submenu /index.js";
export {
  e as AccountSwitcher,
  m as DerivLogo,
  b as DesktopLanguagesModal,
  p as Drawer,
  a as Footer,
  c as Header,
  u as MenuItem,
  I as MobileLanguagesDrawer,
  s as Notifications,
  S as PlatformSwitcher,
  L as PlatformSwitcherItem,
  k as Submenu,
  w as Wrapper
};
