// @deriv-com/quill-ui/dist/components/Pagination/index.js
import { j as i } from "../../jsx-runtime-DmTjM30f.js";
import * as l from "react";
import { forwardRef as b, useState as y, useMemo as N, useEffect as u } from "react";
import { UsePaginationRange as j } from "../../hooks/usePaginationRange.js";
import { PaginationButton as E } from "./base/index.js";
import { c as v } from "../../clsx-OuTLNxxd.js";
import "../Button/base/index.js";
import "../../snackbar-controller-ChcV5SXl.js";
import "../Button/social/index.js";
import { IconButton as x } from "../Button/icon-button/index.js";
import { s as h } from "../../index-CQbz1p6_.js";
const z = ({ iconSize: n = "md", title: t, titleId: e, ...r }, s) => l.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 32 32", ...h[n], role: "img", ref: s, "aria-labelledby": e, ...r },
  t ? l.createElement("title", { id: e }, t) : null,
  l.createElement("path", { d: "M21.43 16.07a.66.66 0 0 1 0 .899l-7.5 7.5a.66.66 0 0 1-.899 0 .66.66 0 0 1 0-.899l7.07-7.07-7.07-7.031a.66.66 0 0 1 0-.899.66.66 0 0 1 .899 0z" })
), S = b(z), B = ({ iconSize: n = "md", title: t, titleId: e, ...r }, s) => l.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 32 32", ...h[n], role: "img", ref: s, "aria-labelledby": e, ...r },
  t ? l.createElement("title", { id: e }, t) : null,
  l.createElement("path", { d: "M10.531 16.969a.66.66 0 0 1 0-.899l7.5-7.5a.66.66 0 0 1 .899 0 .66.66 0 0 1 0 .899L11.86 16.5l7.07 7.07a.66.66 0 0 1 0 .899.66.66 0 0 1-.899 0z" })
), M = b(B), K = ({
  contentPerPage: n,
  contentLength: t,
  initialPage: e,
  onClickPagination: r,
  variant: s = "number",
  hideChevron: p
}) => {
  const [o, m] = y(e ?? 1), c = N(() => Math.ceil((t ?? 0) / (n ?? 1)), [t, n]), w = j({
    totalPageCount: c,
    currentPage: o,
    variant: s
  });
  u(() => {
    e && m(e);
  }, [e]), u(() => {
    r == null || r({ currentPage: o, totalPageCount: c });
  }, [o]);
  const d = () => m((a) => a + 1), f = () => m((a) => a - 1), _ = (a) => {
    const g = Number(a);
    m(g);
  }, R = (a) => {
    a.key === "ArrowLeft" && o !== 1 ? f() : a.key === "ArrowRight" && o !== c && d();
  };
  return /* @__PURE__ */ i.jsxs(
    "nav",
    {
      role: "navigation",
      "aria-label": "Pagination Navigation",
      className: "pagination__container",
      onKeyDown: R,
      tabIndex: 0,
      children: [
        !p && /* @__PURE__ */ i.jsx(
          x,
          {
            color: "black",
            icon: /* @__PURE__ */ i.jsx(
              M,
              {
                iconSize: "sm",
                className: v(
                  o === 1 ? "pagination__chevron-disabled" : "pagination__chevron-svg"
                )
              }
            ),
            className: "pagination__chevron",
            "aria-label": "Go to Previous page",
            onClick: f,
            disabled: o === 1,
            size: "sm",
            variant: "tertiary"
          }
        ),
        w.map((a, g) => /* @__PURE__ */ i.jsx(
          E,
          {
            pageNumber: a,
            variant: s,
            currentPage: o,
            handleOnClick: _
          },
          `${a}_${g}`
        )),
        !p && /* @__PURE__ */ i.jsx(
          x,
          {
            color: "black",
            icon: /* @__PURE__ */ i.jsx(
              S,
              {
                iconSize: "sm",
                className: v(
                  o === c ? "pagination__chevron-disabled" : "pagination__chevron-svg"
                )
              }
            ),
            className: "pagination__chevron",
            "aria-label": "Go to Next page",
            onClick: d,
            disabled: o === c,
            size: "sm",
            variant: "tertiary"
          }
        )
      ]
    }
  );
};
export {
  K as Pagination,
  K as default
};
