// @deriv-com/quill-ui/dist/components/Navigation/bottom-navigation/bottom-action/index.js
import { j as a } from "../../../../jsx-runtime-DmTjM30f.js";
import { c as s } from "../../../../clsx-OuTLNxxd.js";
import { CaptionText as x } from "../../../Typography/caption/index.js";
import "react";
import "../../../Typography/base.js";
import "../../../Typography/heading/index.js";
import { Badge as h } from "../../../Badge/index.js";
import '../../../../assets/index12.css';const j = (m) => {
  const {
    as: b = "div",
    label: o,
    children: d,
    icon: n,
    activeIcon: e,
    className: p,
    showLabel: l,
    selected: r,
    onChange: t,
    onClick: i,
    value: g,
    badge: _,
    ...u
  } = m, v = (c) => {
    t == null || t(c, g), i == null || i(c);
  };
  return /* @__PURE__ */ a.jsxs(
    b,
    {
      className: s(
        "quill-navigation-bottom-bar__action",
        `quill-navigation-bottom-bar__action-selected--${r}`,
        p
      ),
      onClick: v,
      ...u,
      children: [
        /* @__PURE__ */ a.jsx(
          h,
          {
            color: "danger",
            label: _,
            position: "top-right",
            size: "sm",
            variant: "notification",
            className: s({
              "quill-navigation-bottom-bar__action-badge-container": l && o
            }),
            children: (n || e) && /* @__PURE__ */ a.jsx("span", { className: "quill-navigation-bottom-bar__action-icon", children: r ? e : n })
          }
        ),
        o && l && /* @__PURE__ */ a.jsx("div", { className: "quill-navigation-bottom-bar__action-label-container", children: /* @__PURE__ */ a.jsx(
          x,
          {
            centered: !0,
            color: "quill-navigation-bottom-bar__action-label",
            children: o
          }
        ) }),
        d
      ]
    }
  );
};
j.displayName = "Navigation.BottomAction";
export {
  j as default
};
