// @deriv-com/quill-ui/dist/components/ActionSheet/root/index.js
import { j as x } from "../../../jsx-runtime-DmTjM30f.js";
import { createContext as v, useState as w, useCallback as b, useEffect as y } from "react";
const j = v({
  show: !1
}), n = ({
  isOpen: e,
  children: u,
  className: d,
  position: l = "right",
  type: m = "modal",
  expandable: i = !0,
  shouldBlurOnClose: r = !1,
  onOpen: a,
  onClose: t
}) => {
  const [h, s] = w(e), c = () => {
    a == null || a(), s(!0), document.body.style.overflow = "hidden";
  }, o = b(() => {
    var f;
    t == null || t(), s(!1), document.body.style.overflow = "auto", r && ((f = document == null ? void 0 : document.activeElement) == null || f.blur());
  }, [t, r]);
  return y(() => {
    e ? c() : o();
  }, [e, o]), /* @__PURE__ */ x.jsx(
    j.Provider,
    {
      value: {
        show: h,
        handleOpen: c,
        handleClose: o,
        position: l,
        type: m,
        expandable: i,
        className: d
      },
      children: u
    }
  );
};
export {
  j as ActionSheetContext,
  n as default
};
