// @deriv-com/quill-ui/dist/components/SelectionControl/toggle-switch/index.js
import { j as e } from "../../../jsx-runtime-DmTjM30f.js";
import { useState as x, useEffect as b } from "react";
import '../../../assets/toggle-switch.css';/* empty css                              */
import { c as o } from "../../../clsx-OuTLNxxd.js";
const g = ({
  disabled: r = !1,
  onChange: s,
  className: c,
  defaultChecked: m,
  onFocus: d,
  onBlur: p,
  id: u,
  checked: l
}) => {
  const [i, f] = x(m || !1);
  b(() => {
    l !== void 0 && f(l);
  }, [l]);
  const a = () => {
    if (!r)
      if (l === void 0) {
        const t = !i;
        f(t), s == null || s(t);
      } else
        s == null || s(!l);
  }, w = (t) => {
    (t.key === "Enter" || t.key === " ") && (t.preventDefault(), a());
  };
  return /* @__PURE__ */ e.jsx(
    "button",
    {
      className: o("toggle-switch", c),
      onClick: a,
      onKeyDown: w,
      onFocus: d,
      id: u,
      onBlur: p,
      disabled: r,
      "aria-pressed": i,
      children: /* @__PURE__ */ e.jsx(
        "div",
        {
          className: o("toggle-switch__slider", {
            enabled: i
          }),
          children: /* @__PURE__ */ e.jsx(
            "div",
            {
              className: o("toggle-switch__knob", {
                enabled: i
              })
            }
          )
        }
      )
    }
  );
};
export {
  g as ToggleSwitch,
  g as default
};
