// @deriv-com/quill-ui/dist/components/Breadcrumbs/base/index.js
import { j as o } from "../../../jsx-runtime-DmTjM30f.js";
import { useState as x, useEffect as g, Fragment as v } from "react";
import { maxMobileLinks as k, maxDesktopLinks as q } from "./constants.js";
import '../../../assets/breadcrumbs.css';/* empty css                            */
import { useBreakpoints as R } from "../../../hooks/useBreakpoints/index.js";
import { c as m } from "../../../clsx-OuTLNxxd.js";
import "../../Typography/base.js";
import "../../Typography/heading/index.js";
import { Text as N } from "../../Typography/text/index.js";
import { F as E } from "../../../LabelPairedChevronRightSmRegularIcon-BcvI7e_s.js";
function T({ size: b, links: e = [], className: L }) {
  const [d, l] = x([]), [w, j] = x([]), u = (r) => d.length - 1 > r, i = e.length, { isMobile: f } = R(), p = (f ? k : q) < i;
  return g(() => {
    if (l(e), p) {
      const r = [
        e[0],
        { content: "...", href: "" },
        e[i - 2],
        e[i - 1]
      ];
      l(r);
      const s = e.filter(
        (n) => !new Set(r).has(n)
      );
      j(s);
    }
  }, [e, f]), /* @__PURE__ */ o.jsx("div", { className: m("quill-breadcrumb--container", L), children: d.map(({ content: r, href: s, target: n }, a) => /* @__PURE__ */ o.jsxs(v, { children: [
    p && a === 1 && /* @__PURE__ */ o.jsx(
      "select",
      {
        className: "quill-breadcrumb--dropdown",
        onChange: (c) => {
          if (typeof window < "u") {
            const t = c.target.value;
            try {
              const h = new URL(
                t,
                window.location.origin
              );
              window.location.href = h.href;
            } catch {
              console.error("Invalid URL:", t);
            }
          }
        },
        children: w.map(({ href: c, content: t }) => /* @__PURE__ */ o.jsx(
          "option",
          {
            value: c,
            children: t
          },
          `dropdown-${t}`
        ))
      }
    ),
    /* @__PURE__ */ o.jsx(
      "a",
      {
        href: s,
        target: n || "_self",
        className: m(
          "quill-breadcrumb--item",
          !u(a) && "quill-breadcrumb--item--disabled"
        ),
        children: /* @__PURE__ */ o.jsx(N, { size: b, children: r })
      }
    ),
    /* @__PURE__ */ o.jsx(
      E,
      {
        "data-testid": "dt-link-chevron",
        className: m(
          !u(a) && "quill-breadcrumb--chevron"
        ),
        fill: "var(--component-textIcon-normal-subtle)"
      }
    )
  ] }, `breadcrumbs-${r}`)) });
}
export {
  T as Base,
  T as default
};
