// @deriv-com/quill-ui/dist/components/SectionMessage/index.js
import { j as e } from "../../jsx-runtime-DmTjM30f.js";
import '../../assets/section-message.css';/* empty css                             */
import { c as x } from "../../clsx-OuTLNxxd.js";
import "react";
import "../Typography/base.js";
import "../Typography/heading/index.js";
import { Text as a } from "../Typography/text/index.js";
import { Link as f } from "../Link/index.js";
import { F as j, a as p } from "../../StandaloneCircleInfoBoldIcon-G7lZfb4e.js";
import { F as g, a as h } from "../../StandaloneTriangleExclamationBoldIcon-DbjkWWdj.js";
const l = {
  info: /* @__PURE__ */ e.jsx(
    j,
    {
      fill: "var(--core-color-solid-blue-800)",
      iconSize: "sm"
    }
  ),
  danger: /* @__PURE__ */ e.jsx(
    g,
    {
      fill: "var(--core-color-solid-red-800)",
      iconSize: "sm"
    }
  ),
  success: /* @__PURE__ */ e.jsx(
    h,
    {
      fill: "var(--core-color-solid-green-800)",
      iconSize: "sm"
    }
  ),
  warning: /* @__PURE__ */ e.jsx(
    p,
    {
      fill: "var(--core-color-solid-yellow-800)",
      iconSize: "sm"
    }
  )
}, y = ({
  title: c,
  message: d,
  className: n,
  status: r,
  icon: s,
  size: o = "md",
  linkList: i
}) => /* @__PURE__ */ e.jsxs(
  "div",
  {
    className: x(
      "section-message",
      `section-message--${r}`,
      n
    ),
    children: [
      (r || s) && /* @__PURE__ */ e.jsx("div", { children: r ? l == null ? void 0 : l[r] : s && s }),
      /* @__PURE__ */ e.jsxs("div", { className: "section-message-content", children: [
        /* @__PURE__ */ e.jsxs("div", { className: "title-and-message", children: [
          c && /* @__PURE__ */ e.jsx(a, { bold: !0, size: o === "md" ? "lg" : "md", children: c }),
          /* @__PURE__ */ e.jsx(a, { size: o, children: d })
        ] }),
        i && i.length > 0 && /* @__PURE__ */ e.jsx("div", { className: "section-message-links", children: i.map(({ id: t, linkProps: m }) => /* @__PURE__ */ e.jsx(f, { ...m, size: o, children: m.children }, t)) })
      ] })
    ]
  }
);
export {
  y as SectionMessage,
  y as default
};
