// @deriv-com/ui/dist/components/AppLayout/AccountSwitcher/index.js
import '../../../assets/index24.css';
import { jsx as o, Fragment as x, jsxs as u } from "react/jsx-runtime";
import * as l from "react";
import { forwardRef as A, useState as S, useRef as T } from "react";
import { a as _ } from "../../../index-EY1gwl5O.js";
import { Modal as d } from "../../Modal/index.js";
import { ContextMenu as R } from "../../ContextMenu/index.js";
import { useDevice as g } from "../../../hooks/useDevice.js";
import f from "../../Tabs/Tabs.js";
import { AccountsPanel as y } from "./AccountsPanel/index.js";
import { AccountsItem as F } from "./AccountsItem/index.js";
import { AccountSwitcherFooter as N } from "./Footer/index.js";
import { TotalAsset as k } from "./TotalAsset/index.js";
import { AccountSwitcherTab as C } from "./Tab/index.js";
import { TradershubLink as E } from "./TradershubLink/index.js";
import { c as z } from "../../../clsx-OuTLNxxd.js";
import { s as D } from "../../../index-CQbz1p6_.js";
import { F as O } from "../../../LegacyChevronDown2pxIcon-CQ-CdTZp.js";
const I = ({ iconSize: n = "md", title: e, titleId: i, ...t }, r) => l.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", fill: "none", viewBox: "0 0 16 16", ...D[n], role: "img", ref: r, "aria-labelledby": i, ...t },
  e ? l.createElement("title", { id: i }, e) : null,
  l.createElement("path", { fill: "#333", d: "m8 6.414-5.293 5.293a1 1 0 0 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414z" })
), M = A(I), c = ({
  children: n,
  activeAccount: e,
  isDisabled: i = !1,
  buttonClassName: t,
  tabsLabels: r
}) => {
  const [m, a] = S(!1), s = T(null), { isDesktop: h } = g(), p = () => {
    a(!1);
  }, w = () => {
    a(!0);
  };
  return _(s, p), /* @__PURE__ */ o(x, { children: /* @__PURE__ */ u(
    "div",
    {
      className: z(
        "deriv-account-switcher__button",
        t
      ),
      onClick: () => {
        s.current || w();
      },
      children: [
        /* @__PURE__ */ o("div", { className: "deriv-account-switcher__currency-icon", children: e.icon }),
        /* @__PURE__ */ u("div", { className: "deriv-account-switcher__balance", children: [
          /* @__PURE__ */ o("span", { children: e.balance }),
          /* @__PURE__ */ o("span", { children: e.currency })
        ] }),
        i ? null : m ? /* @__PURE__ */ o(M, { iconSize: "xs" }) : /* @__PURE__ */ o(O, { iconSize: "xs" }),
        i ? null : h ? /* @__PURE__ */ o(
          R,
          {
            ref: s,
            className: "deriv-account-switcher__container",
            isOpen: m,
            children: n && /* @__PURE__ */ o(
              f,
              {
                activeTab: e.isVirtual ? (r == null ? void 0 : r.demo) ?? "Demo" : (r == null ? void 0 : r.real) ?? "Real",
                variant: "secondary",
                children: n
              }
            )
          }
        ) : /* @__PURE__ */ o(
          d,
          {
            closeTimeoutMS: 200,
            onRequestClose: (v) => {
              v.stopPropagation(), p();
            },
            className: "deriv-account-switcher__container--mobile",
            isOpen: m,
            children: n && /* @__PURE__ */ o(
              f,
              {
                activeTab: e.isVirtual ? (r == null ? void 0 : r.demo) ?? "Demo" : (r == null ? void 0 : r.real) ?? "Real",
                variant: "secondary",
                children: n
              }
            )
          }
        )
      ]
    }
  ) });
};
c.Tab = C;
c.AccountsPanel = y;
c.AccountsItem = F;
c.TotalAsset = k;
c.TradersHubLink = E;
c.Footer = N;
c.setAppElement = d.setAppElement;
c.DisplayName = "AccountSwitcher";
export {
  c as AccountSwitcher
};
