// @deriv-com/ui/dist/main.js
import { Accordion as e } from "./components/Accordion/index.js";
import { ActionScreen as m } from "./components/ActionScreen/index.js";
import { Button as f } from "./components/Button/index.js";
import { Badge as a } from "./components/Badge/index.js";
import { Breadcrumbs as s } from "./components/Breadcrumbs/index.js";
import { Checkbox as c } from "./components/Checkbox/index.js";
import { CircularProgressBar as l } from "./components/CircularProgressBar/index.js";
import { Divider as d } from "./components/Divider/index.js";
import { Dialog as T } from "./components/Dialog/index.js";
import { Dropdown as D } from "./components/Dropdown/index.js";
import { InlineMessage as M } from "./components/InlineMessage/index.js";
import { Input as L } from "./components/Input/index.js";
import { LinearProgressBar as h } from "./components/LinearProgressBar/index.js";
import { Loader as A } from "./components/Loader/index.js";
import { Modal as C } from "./components/Modal/index.js";
import { PageLayout as N } from "./components/PageLayout/index.js";
import { PasswordInput as y } from "./components/PasswordInput/index.js";
import { SectionMessage as H } from "./components/SectionMessage/index.js";
import { SideNote as j } from "./components/SideNote/index.js";
import { default as z } from "./components/Tabs/Tab.js";
import { default as G } from "./components/Tabs/Tabs.js";
import { Table as K } from "./components/Table/index.js";
import { Text as Q } from "./components/Text/index.js";
import { TextArea as U } from "./components/TextArea/index.js";
import { ToggleSwitch as Y } from "./components/ToggleSwitch/index.js";
import { Tooltip as _ } from "./components/Tooltip/index.js";
import { useDevice as rr } from "./hooks/useDevice.js";
import { VerticalTab as er } from "./components/VerticalTab/VerticalTab.js";
import { VerticalTabItems as mr } from "./components/VerticalTab/VerticalTabItems.js";
import { AccountSwitcher as fr } from "./components/AppLayout/AccountSwitcher/index.js";
import { DerivLogo as ar } from "./components/AppLayout/DerivLogo/index.js";
import { Drawer as sr } from "./components/AppLayout/Drawer/index.js";
import { Footer as cr } from "./components/AppLayout/Footer/index.js";
import { Header as lr } from "./components/AppLayout/Header/index.js";
import { MenuItem as dr } from "./components/AppLayout/MenuItem/index.js";
import { Wrapper as Tr } from "./components/AppLayout/Wrapper/index.js";
import { Notifications as Dr } from "./components/AppLayout/Notifications/index.js";
import { PlatformSwitcher as Mr } from "./components/AppLayout/PlatformSwitcher/index.js";
import { PlatformSwitcherItem as Lr } from "./components/AppLayout/PlatformSwitcher/PlatformSwitcherItem.js";
import { DesktopLanguagesModal as hr } from "./components/AppLayout/LanguagesSwitcher/DesktopLanguagesModal.js";
import { MobileLanguagesDrawer as Ar } from "./components/AppLayout/LanguagesSwitcher/MobileLanguagesDrawer.js";
import { Submenu as Cr } from "./components/AppLayout/Submenu /index.js";
import { ContextMenu as Nr } from "./components/ContextMenu/index.js";
export {
  e as Accordion,
  fr as AccountSwitcher,
  m as ActionScreen,
  a as Badge,
  s as Breadcrumbs,
  f as Button,
  c as Checkbox,
  l as CircularProgressBar,
  Nr as ContextMenu,
  ar as DerivLogo,
  hr as DesktopLanguagesModal,
  T as Dialog,
  d as Divider,
  sr as Drawer,
  D as Dropdown,
  cr as Footer,
  lr as Header,
  M as InlineMessage,
  L as Input,
  h as LinearProgressBar,
  A as Loader,
  dr as MenuItem,
  Ar as MobileLanguagesDrawer,
  C as Modal,
  Dr as Notifications,
  N as PageLayout,
  y as PasswordInput,
  Mr as PlatformSwitcher,
  Lr as PlatformSwitcherItem,
  H as SectionMessage,
  j as SideNote,
  Cr as Submenu,
  z as Tab,
  K as Table,
  G as Tabs,
  Q as Text,
  U as TextArea,
  Y as ToggleSwitch,
  _ as Tooltip,
  er as VerticalTab,
  mr as VerticalTabItems,
  Tr as Wrapper,
  rr as useDevice
};
