// @deriv-com/ui/dist/components/SectionMessage/index.js
import '../../assets/SectionMessage.css';
import { jsxs as m, jsx as r } from "react/jsx-runtime";
import { c as o } from "../../clsx-OuTLNxxd.js";
import { Text as h } from "../Text/index.js";
/* empty css                            */
import * as e from "react";
import { forwardRef as s } from "react";
const g = ({ title: a, titleId: t, ...c }, l) => e.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 24, viewBox: "0 0 16 24", role: "img", ref: l, "aria-labelledby": t, ...c },
  a ? e.createElement("title", { id: t }, a) : null,
  e.createElement(
    "g",
    null,
    e.createElement("path", { d: "M8 5.5c-2.344 0-4.469 1.25-5.656 3.25a6.52 6.52 0 0 0 0 6.5A6.51 6.51 0 0 0 8 18.5c2.313 0 4.438-1.219 5.625-3.25a6.52 6.52 0 0 0 0-6.5C12.438 6.75 10.313 5.5 8 5.5M8 20c-2.875 0-5.5-1.5-6.937-4-1.438-2.469-1.438-5.5 0-8A8.02 8.02 0 0 1 8 4c2.844 0 5.469 1.531 6.906 4 1.438 2.5 1.438 5.531 0 8A7.94 7.94 0 0 1 8 20m3.531-9.469-4 4c-.312.313-.781.313-1.062 0l-2-2a.684.684 0 0 1 0-1.031c.281-.312.75-.312 1.062 0L7 12.969 10.469 9.5c.281-.312.75-.312 1.062 0a.736.736 0 0 1 0 1.031" })
  ),
  e.createElement(
    "defs",
    null,
    e.createElement(
      "clipPath",
      { id: "1fb059352b46b1cb332ac0b18a67fe03__a" },
      e.createElement("path", { d: "M0 0h16v24H0z" })
    )
  )
), v = s(g), f = ({ title: a, titleId: t, ...c }, l) => e.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 24, viewBox: "0 0 16 24", role: "img", ref: l, "aria-labelledby": t, ...c },
  a ? e.createElement("title", { id: t }, a) : null,
  e.createElement(
    "g",
    null,
    e.createElement("path", { d: "M7.75 6.656 1.563 16.813c-.063.093-.063.156-.063.25 0 .25.188.437.438.437H14.03c.25 0 .469-.187.469-.437 0-.094-.031-.157-.094-.25L8.22 6.656A.235.235 0 0 0 8 6.5c-.125 0-.187.063-.25.156m-1.281-.781A1.8 1.8 0 0 1 8 5c.594 0 1.188.344 1.5.875l6.188 10.156c.187.313.312.657.312 1.032C16 18.125 15.125 19 14.031 19H1.938A1.947 1.947 0 0 1 0 17.063q0-.563.281-1.032zM9 15.5c0 .563-.469 1-1 1-.562 0-1-.437-1-1 0-.531.438-1 1-1 .531 0 1 .469 1 1m-.25-5.75v3a.74.74 0 0 1-.75.75.72.72 0 0 1-.75-.75v-3A.74.74 0 0 1 8 9a.76.76 0 0 1 .75.75" })
  ),
  e.createElement(
    "defs",
    null,
    e.createElement(
      "clipPath",
      { id: "1818d62d0ad2f189ea051505bd61b6d6__a" },
      e.createElement("path", { d: "M0 0h16v24H0z" })
    )
  )
), w = s(f), E = ({ title: a, titleId: t, ...c }, l) => e.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 24, viewBox: "0 0 16 24", role: "img", ref: l, "aria-labelledby": t, ...c },
  a ? e.createElement("title", { id: t }, a) : null,
  e.createElement(
    "g",
    null,
    e.createElement("path", { d: "M8 5.5c-2.344 0-4.469 1.25-5.656 3.25a6.52 6.52 0 0 0 0 6.5A6.51 6.51 0 0 0 8 18.5c2.313 0 4.438-1.219 5.625-3.25a6.52 6.52 0 0 0 0-6.5C12.438 6.75 10.313 5.5 8 5.5M8 20c-2.875 0-5.5-1.5-6.937-4-1.438-2.469-1.438-5.5 0-8A8.02 8.02 0 0 1 8 4c2.844 0 5.469 1.531 6.906 4 1.438 2.5 1.438 5.531 0 8A7.94 7.94 0 0 1 8 20M8 8a.76.76 0 0 1 .75.75v3.5A.74.74 0 0 1 8 13a.72.72 0 0 1-.75-.75v-3.5A.74.74 0 0 1 8 8m1 7c0 .563-.469 1-1 1-.562 0-1-.437-1-1 0-.531.438-1 1-1 .531 0 1 .469 1 1" })
  ),
  e.createElement(
    "defs",
    null,
    e.createElement(
      "clipPath",
      { id: "4434df383b9b26f1cef1fc50792b86e3__a" },
      e.createElement("path", { d: "M0 0h16v24H0z" })
    )
  )
), p = s(E), b = ({ title: a, titleId: t, ...c }, l) => e.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 24, viewBox: "0 0 16 24", role: "img", ref: l, "aria-labelledby": t, ...c },
  a ? e.createElement("title", { id: t }, a) : null,
  e.createElement(
    "g",
    null,
    e.createElement("path", { d: "M8 5.5c-2.344 0-4.469 1.25-5.656 3.25a6.52 6.52 0 0 0 0 6.5A6.51 6.51 0 0 0 8 18.5c2.313 0 4.438-1.219 5.625-3.25a6.52 6.52 0 0 0 0-6.5C12.438 6.75 10.313 5.5 8 5.5M8 20c-2.875 0-5.5-1.5-6.937-4-1.438-2.469-1.438-5.5 0-8A8.02 8.02 0 0 1 8 4c2.844 0 5.469 1.531 6.906 4 1.438 2.5 1.438 5.531 0 8A7.94 7.94 0 0 1 8 20m-1.25-5.5h.75v-2h-.75a.72.72 0 0 1-.75-.75.74.74 0 0 1 .75-.75h1.5a.76.76 0 0 1 .75.75v2.75h.25a.76.76 0 0 1 .75.75.74.74 0 0 1-.75.75h-2.5a.72.72 0 0 1-.75-.75.74.74 0 0 1 .75-.75M8 10c-.562 0-1-.437-1-1 0-.531.438-1 1-1 .531 0 1 .469 1 1 0 .563-.469 1-1 1" })
  ),
  e.createElement(
    "defs",
    null,
    e.createElement(
      "clipPath",
      { id: "d954ee467a6d4f3f4fe390ae7ac8e2f0__a" },
      e.createElement("path", { d: "M0 0h16v24H0z" })
    )
  )
), _ = s(b), M = {
  error: w,
  info: _,
  success: v,
  warning: p
}, P = ({
  children: a,
  className: t,
  ctaSection: c,
  icon: l,
  title: d,
  variant: n = "general"
}) => {
  const i = n !== "general" && M[n];
  return /* @__PURE__ */ m(
    "div",
    {
      className: o(
        "deriv-section-message",
        `deriv-section-message--${n}`,
        t
      ),
      children: [
        (l || i) && /* @__PURE__ */ r("div", { className: "deriv-section-message__icon", children: i ? /* @__PURE__ */ r(
          i,
          {
            className: `deriv-section-message__icon--${n}`
          }
        ) : l }),
        /* @__PURE__ */ m("div", { className: "deriv-section-message__content", children: [
          d && /* @__PURE__ */ r(
            h,
            {
              as: "div",
              align: "start",
              size: "md",
              weight: "bold",
              className: "deriv-section-message__title",
              children: d
            }
          ),
          /* @__PURE__ */ r("div", { className: "deriv-section-message__description", children: a }),
          c && /* @__PURE__ */ r("div", { className: "deriv-section-message__cta-section", children: c })
        ] })
      ]
    }
  );
};
export {
  P as SectionMessage
};
