// @deriv-com/quill-ui/dist/components/Chip/base/index.js
import { j as a } from "../../../jsx-runtime-DmTjM30f.js";
import * as o from "react";
import z, { forwardRef as f } from "react";
import { c as q } from "../../../clsx-OuTLNxxd.js";
import { CaptionText as w } from "../../Typography/caption/index.js";
import "../../Typography/base.js";
import "../../Typography/heading/index.js";
import { Text as g } from "../../Typography/text/index.js";
import { s as p } from "../../../index-CQbz1p6_.js";
import '../../../assets/index18.css';const N = ({ iconSize: s = "md", title: t, titleId: e, ...l }, r) => o.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 32 32", ...p[s], role: "img", ref: r, "aria-labelledby": e, ...l },
  t ? o.createElement("title", { id: e }, t) : null,
  o.createElement("path", { d: "M16 7.75a8.73 8.73 0 0 0-7.578 4.375c-1.602 2.734-1.602 6.055 0 8.75C9.984 23.609 12.836 25.25 16 25.25a8.63 8.63 0 0 0 7.54-4.375c1.6-2.695 1.6-6.016 0-8.75C21.976 9.43 19.124 7.75 16 7.75m0 18.75c-3.594 0-6.875-1.875-8.672-5-1.797-3.086-1.797-6.875 0-10 1.797-3.086 5.078-5 8.672-5 3.555 0 6.836 1.914 8.633 5 1.797 3.125 1.797 6.914 0 10a9.93 9.93 0 0 1-8.633 5m-2.969-12.93a.66.66 0 0 1 .899 0L16 15.64l2.031-2.07a.66.66 0 0 1 .899 0 .66.66 0 0 1 0 .899L16.86 16.5l2.07 2.07a.66.66 0 0 1 0 .899.66.66 0 0 1-.899 0L16 17.399l-2.07 2.07a.66.66 0 0 1-.899 0 .66.66 0 0 1 0-.899l2.07-2.07-2.07-2.031a.66.66 0 0 1 0-.899" })
), T = f(N), $ = ({ iconSize: s = "md", title: t, titleId: e, ...l }, r) => o.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 32 32", ...p[s], role: "img", ref: r, "aria-labelledby": e, ...l },
  t ? o.createElement("title", { id: e }, t) : null,
  o.createElement("path", { d: "M16.43 21.969a.66.66 0 0 1-.899 0l-7.5-7.5a.66.66 0 0 1 0-.899.66.66 0 0 1 .899 0L16 20.64l7.031-7.07a.66.66 0 0 1 .899 0 .66.66 0 0 1 0 .899z" })
), y = f($), A = {
  sm: /* @__PURE__ */ a.jsx(w, {}),
  md: /* @__PURE__ */ a.jsx(g, {}),
  lg: /* @__PURE__ */ a.jsx(g, {})
}, J = f(
  ({
    icon: s,
    size: t = "md",
    label: e,
    labelTag: l,
    dismissible: r = !1,
    dropdown: i = !1,
    className: h,
    selected: n,
    isDropdownOpen: j,
    onChipSelect: c,
    onDismiss: d,
    children: R,
    ..._
  }, b) => {
    const E = (m) => {
      if (!(r || i))
        if (n === void 0) {
          const u = m.currentTarget, x = u.getAttribute("data-state") === "selected", k = x ? "" : "selected";
          u.setAttribute("data-state", k), c == null || c(m, !x);
        } else
          c == null || c(m, n);
    }, v = (m) => {
      d == null || d(m);
    }, L = r || i;
    return /* @__PURE__ */ a.jsxs(
      "button",
      {
        onClick: E,
        className: q(
          "quill-chip",
          `quill-chip__size--${t}`,
          L && `quill-chip__custom-right-padding__size--${t}`,
          h
        ),
        "data-state": n ? "selected" : "",
        ref: b,
        ..._,
        children: [
          s && /* @__PURE__ */ a.jsx(s, {}),
          e && z.cloneElement(A[t], {
            children: e
          }),
          R,
          l && /* @__PURE__ */ a.jsx(w, { bold: !0, children: l }),
          r && /* @__PURE__ */ a.jsx(
            T,
            {
              iconSize: "sm",
              onClick: v,
              "data-testid": "dt-chip-dismissible-btn",
              className: "dismissible-icon"
            }
          ),
          i && /* @__PURE__ */ a.jsx(
            y,
            {
              width: 24,
              height: 24,
              "data-state": j ? "open" : "close",
              className: "rotate"
            }
          )
        ]
      }
    );
  }
);
export {
  J as Chip,
  A as LabelTextSizes,
  J as default
};
