// @deriv-com/quill-ui/dist/components/Atom/dropdown/title/index.js
import { j as o } from "../../../../jsx-runtime-DmTjM30f.js";
import { forwardRef as m } from "react";
import { c as p } from "../../../../clsx-OuTLNxxd.js";
import "../../../Typography/base.js";
import "../../../Typography/heading/index.js";
import { Text as n } from "../../../Typography/text/index.js";
import '../../../../assets/index.css';const j = m(
  ({
    as: t = "div",
    label: s,
    size: r = "md",
    icon: l,
    className: d,
    centered: i,
    ...e
  }, a) => /* @__PURE__ */ o.jsxs(
    t,
    {
      className: p(
        "quill-dropdown-title__container",
        `quill-dropdown-title__container--${r}`,
        d
      ),
      ref: a,
      ...e,
      children: [
        l && i && /* @__PURE__ */ o.jsx("div", { className: "quill-dropdown-title__icon" }),
        /* @__PURE__ */ o.jsx(
          n,
          {
            size: r,
            className: "quill-dropdown-title__label",
            color: "quill-typography__color--subtle",
            centered: i,
            children: s
          }
        ),
        l && /* @__PURE__ */ o.jsx("div", { className: "quill-dropdown-title__icon", children: l })
      ]
    }
  )
);
export {
  j as DropdownTitle
};
