// @deriv-com/quill-ui/dist/components/SelectionControl/radio/radio-group/index.js
import { j as e } from "../../../../jsx-runtime-DmTjM30f.js";
import u from "react";
import { c as _ } from "../../../../clsx-OuTLNxxd.js";
import { R as h } from "../../../../index-C7RXo2Oj.js";
import '../../../../assets/index6.css';const I = ({
  children: r,
  shouldWrapItems: s
}) => s ? /* @__PURE__ */ e.jsx("div", { className: "quill-radio-group__item-wrapper", children: r }) : /* @__PURE__ */ e.jsx(u.Fragment, { children: r }), b = ({
  children: r,
  className: s,
  name: a,
  onToggle: i,
  required: m,
  selected: o,
  shouldWrapItems: n,
  size: c
}) => {
  const [l, d] = u.useState(o);
  u.useEffect(() => {
    d(o);
  }, [o]);
  const f = (t) => {
    d(t.target.value), i == null || i(t);
  };
  return /* @__PURE__ */ e.jsx("div", { className: _("quill-radio-group", s), children: Array.isArray(r) && r.filter((t) => !t.props.hidden).map((t) => {
    const { id: x, value: p, label: j, disabled: v, ...R } = t.props;
    return /* @__PURE__ */ e.jsx(
      I,
      {
        shouldWrapItems: n,
        children: /* @__PURE__ */ e.jsx(
          h,
          {
            id: x,
            name: a,
            value: p,
            defaultChecked: l === p,
            onChange: f,
            disabled: v,
            required: m,
            size: c,
            ...R,
            children: j
          }
        )
      },
      p
    );
  }) });
}, q = ({
  children: r,
  hidden: s = !1,
  ...a
}) => /* @__PURE__ */ e.jsx("div", { hidden: s, ...a, children: r });
b.Item = q;
export {
  b as RadioGroup,
  b as default
};
