// @deriv-com/quill-ui/dist/components/Notification/items-list/index.js
import { j as p } from "../../../jsx-runtime-DmTjM30f.js";
import { useState as I, useRef as h, useEffect as x, createElement as v } from "react";
import { c as E } from "../../../clsx-OuTLNxxd.js";
import { NotificationItem as g } from "../item/index.js";
const B = ({
  items: o = [],
  className: n,
  isMobile: e,
  onClick: r,
  onClose: m,
  onMarkAsRead: u
}) => {
  const [f, a] = I(""), c = h([]);
  x(() => () => {
    c.current.length && c.current.forEach((t) => clearTimeout(t));
  }, []);
  const s = (t) => {
    const i = setTimeout(() => {
      m == null || m(t);
    }, 160);
    c.current.push(i);
  };
  return o.length ? /* @__PURE__ */ p.jsx("div", { className: E("quill-notification__items-list", n), children: o.map(({ id: t, ...i }) => /* @__PURE__ */ v(
    g,
    {
      ...i,
      key: t,
      isMobile: e,
      onClick: () => r == null ? void 0 : r(t),
      onClose: () => s == null ? void 0 : s(t),
      onMarkAsRead: () => u == null ? void 0 : u(t),
      onShowingButtons: e ? () => a(t) : void 0,
      showButtons: !!e && f === t
    }
  )) }) : null;
};
export {
  B as NotificationItemsList,
  B as default
};
