// @deriv-com/quill-ui/dist/components/Badge/base.js
import { j as e } from "../../jsx-runtime-DmTjM30f.js";
import { forwardRef as p } from "react";
import { c as x } from "../../clsx-OuTLNxxd.js";
import { CaptionText as _ } from "../Typography/caption/index.js";
import "../Typography/base.js";
import "../Typography/heading/index.js";
import { Text as a } from "../Typography/text/index.js";
import '../../assets/base.css';const B = p(
  ({
    size: o = "sm",
    color: l = "success",
    label: t,
    variant: c,
    className: m,
    badgeClassName: d,
    ...n
  }, i) => {
    const s = "badge__base-color", r = t && t.length > 2 ? "99+" : t;
    return /* @__PURE__ */ e.jsx(
      "span",
      {
        className: x(
          m,
          d,
          "badge__base",
          `badge__variant__${c}--size-${o}`,
          `badge__color--${l}`
        ),
        ref: i,
        ...n,
        children: r && (o === "sm" ? /* @__PURE__ */ e.jsx(_, { bold: !0, color: s, children: r }) : o === "md" ? /* @__PURE__ */ e.jsx(a, { bold: !0, color: s, children: r }) : o === "lg" ? /* @__PURE__ */ e.jsx(a, { size: "xl", bold: !0, color: s, children: r }) : null)
      }
    );
  }
);
export {
  B as BadgeBase,
  B as default
};
