// @deriv-com/quill-ui/dist/components/Badge/index.js
import { j as o } from "../../jsx-runtime-DmTjM30f.js";
import { forwardRef as g } from "react";
import { BadgeBase as b } from "./base.js";
import { c as f } from "../../clsx-OuTLNxxd.js";
const u = g(
  ({
    size: t,
    color: r,
    label: s,
    children: e,
    variant: i,
    position: a,
    badgeClassName: d,
    contentSize: m = "md",
    className: n,
    ...c
  }, _) => {
    const p = {
      "bottom-right": `badge__position-bottom-${t}`,
      "top-right": `badge__position-top-${m}`,
      center: "badge__position-center"
    };
    return /* @__PURE__ */ o.jsxs("div", { className: "badge__container", children: [
      /* @__PURE__ */ o.jsx("div", { className: "badge__container-children", children: e && e }),
      /* @__PURE__ */ o.jsx(
        b,
        {
          ...c,
          size: t,
          color: r,
          label: s,
          variant: i,
          badgeClassName: d,
          className: f(
            a && p[a],
            n
          ),
          ref: _
        }
      )
    ] });
  }
);
export {
  u as Badge,
  u as default
};
