// @deriv-com/quill-ui/dist/components/Button/dropdown/dropdown-content.js
import { j as o } from "../../../jsx-runtime-DmTjM30f.js";
import { D as h } from "../../../index-C1BOoRZp.js";
import '../../../assets/wheel-picker-base.css';import '../../../assets/password-validation.css';import '../../../assets/date-picker.css';import '../../../assets/divider.css';/* empty css                        */
import { DropdownTitle as F } from "../../Atom/dropdown/title/index.js";
import { ItemContainer as B } from "../../Atom/dropdown/item-container/index.js";
import "react";
/* empty css                            */
import "../../Typography/base.js";
import "../../Typography/heading/index.js";
import "../../../dayjs.min-DrYbymKu.js";
/* empty css                                    */
/* empty css                                  */
import "../../../lodash-rsqVyylH.js";
import { useDropdown as E } from "../../../hooks/useDropdown/index.js";
import { useBreakpoints as M } from "../../../hooks/useBreakpoints/index.js";
const Q = ({
  options: a,
  size: r,
  closeContentOnClick: j,
  height: x,
  label: p,
  className: f,
  checkbox: m,
  onItemClick: t,
  centered: i,
  ...u
}) => {
  const { close: D } = E(), { isMobile: n } = M(), e = () => /* @__PURE__ */ o.jsxs(o.Fragment, { children: [
    !n && p && /* @__PURE__ */ o.jsx(F, { label: p, size: r, centered: i }),
    a.map((d) => {
      const { id: c, selected: w, onClick: s, ...l } = d;
      return /* @__PURE__ */ o.jsx(
        h,
        {
          size: r,
          onClick: (g) => {
            j && D(), t == null || t(c), s == null || s(g);
          },
          as: "button",
          selected: m && w,
          checkbox: m,
          centered: i,
          ...u,
          ...l
        },
        d.id
      );
    })
  ] });
  return /* @__PURE__ */ o.jsx(o.Fragment, { children: n ? /* @__PURE__ */ o.jsx(e, {}) : /* @__PURE__ */ o.jsx(
    B,
    {
      size: r,
      className: f,
      height: x,
      children: /* @__PURE__ */ o.jsx(e, {})
    }
  ) });
};
export {
  Q as default
};
