// @deriv-com/quill-ui/dist/components/Stepper/base.js
import { j as t } from "../../jsx-runtime-DmTjM30f.js";
import * as e from "react";
import f, { forwardRef as m } from "react";
import { c as n } from "../../clsx-OuTLNxxd.js";
import '../../assets/stepper.css';/* empty css                     */
import "../Typography/base.js";
import "../Typography/heading/index.js";
import { Text as g } from "../Typography/text/index.js";
const v = ({ title: l, titleId: a, ...c }, s) => e.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", width: 13, height: 22, viewBox: "0 0 13 22", role: "img", ref: s, "aria-labelledby": a, ...c },
  l ? e.createElement("title", { id: a }, l) : null,
  e.createElement(
    "g",
    null,
    e.createElement("path", { d: "M12.352 6.648a.843.843 0 0 1 0 1.23l-7 7a.843.843 0 0 1-1.23 0l-3.5-3.5a.843.843 0 0 1 0-1.23.843.843 0 0 1 1.23 0L4.75 13.02l6.371-6.372a.843.843 0 0 1 1.23 0" })
  ),
  e.createElement(
    "defs",
    null,
    e.createElement(
      "clipPath",
      { id: "ed561ce027ed5d1961241930691b028c__a" },
      e.createElement("path", { d: "M0 0h13v22H0z" })
    )
  )
), w = m(v), E = ({ title: l, titleId: a, ...c }, s) => e.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", width: 14, height: 24, viewBox: "0 0 14 24", role: "img", ref: s, "aria-labelledby": a, ...c },
  l ? e.createElement("title", { id: a }, l) : null,
  e.createElement(
    "g",
    null,
    e.createElement("path", { d: "M13.688 7.313a.964.964 0 0 1 0 1.406l-8 8a.964.964 0 0 1-1.407 0l-4-4a.964.964 0 0 1 0-1.406.964.964 0 0 1 1.406 0L5 14.593l7.281-7.28a.964.964 0 0 1 1.406 0" })
  ),
  e.createElement(
    "defs",
    null,
    e.createElement(
      "clipPath",
      { id: "a1c9c9f97002b7fd5350890599a3ce85__a" },
      e.createElement("path", { d: "M0 0h14v24H0z" })
    )
  )
), x = m(E), b = ({ title: l, titleId: a, ...c }, s) => e.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", width: 18, height: 30, viewBox: "0 0 18 30", role: "img", ref: s, "aria-labelledby": a, ...c },
  l ? e.createElement("title", { id: a }, l) : null,
  e.createElement(
    "g",
    null,
    e.createElement("path", { d: "M17.36 9.64c.507.47.507 1.29 0 1.758l-10 10a1.205 1.205 0 0 1-1.758 0l-5-5a1.205 1.205 0 0 1 0-1.757 1.205 1.205 0 0 1 1.757 0L6.5 18.742l9.102-9.101a1.205 1.205 0 0 1 1.757 0" })
  ),
  e.createElement(
    "defs",
    null,
    e.createElement(
      "clipPath",
      { id: "9d32da9c3661f7bacf2e982c3709a225__a" },
      e.createElement("path", { d: "M0 0h18v30H0z" })
    )
  )
), _ = m(b), d = "var(--semantic-color-slate-solid-surface-frame-low)", u = {
  sm: /* @__PURE__ */ t.jsx(w, { fill: d }),
  md: /* @__PURE__ */ t.jsx(x, { fill: d }),
  lg: /* @__PURE__ */ t.jsx(_, { fill: d })
}, j = ({ size: l }) => /* @__PURE__ */ t.jsx(t.Fragment, { children: u[l] }), k = ({
  currentStep: l,
  Icon: a = j,
  size: c = "md",
  labels: s = [],
  lineSize: p = "md"
}) => /* @__PURE__ */ t.jsx("div", { className: "quill-steps-container", children: typeof l == "number" && s.map((h, r) => {
  const i = r <= l, o = r > l + 1;
  return r !== 0 && /* @__PURE__ */ t.jsx(f.Fragment, { children: /* @__PURE__ */ t.jsxs("div", { className: "step", children: [
    /* @__PURE__ */ t.jsxs("div", { className: "step-circle__pointer", children: [
      /* @__PURE__ */ t.jsx(
        "div",
        {
          className: n(
            "step-circle",
            `step-circle__size-${c}`,
            {
              "step-circle--active": i,
              "step-circle--inactive": !i,
              "step-circle--disabled": o
            }
          ),
          "data-testid": "dt-step-circle",
          children: i && /* @__PURE__ */ t.jsx(a, { size: c })
        }
      ),
      r < s.length - 1 && /* @__PURE__ */ t.jsx(
        "div",
        {
          className: n(
            "step-line",
            `step-line--size-${p}`,
            {
              "step-line--disabled": r > l
            }
          ),
          "data-testid": "dt-step-line"
        }
      )
    ] }),
    /* @__PURE__ */ t.jsx(
      g,
      {
        as: "span",
        bold: !0,
        size: c,
        className: n(
          `step-circle__label-${c}`,
          {
            "step-circle__label--disabled": o
          }
        ),
        children: s[r]
      }
    )
  ] }) }, h);
}) });
export {
  k as BasicStepper,
  k as default
};
