// @deriv-com/quill-ui/dist/components/Tag/index.js
import { j as o } from "../../jsx-runtime-DmTjM30f.js";
import { forwardRef as x } from "react";
import { c as g } from "../../clsx-OuTLNxxd.js";
import { CaptionText as n } from "../Typography/caption/index.js";
import "../Typography/base.js";
import "../Typography/heading/index.js";
import { Text as u } from "../Typography/text/index.js";
import { TagIcon as j } from "./tagIcon/index.js";
const _ = x(
  ({
    label: m,
    className: e,
    variant: s = "fill",
    size: r = "md",
    color: a = "success",
    isBold: c = !1,
    icon: d,
    iconClassName: i,
    showIcon: l = !0,
    ...p
  }, f) => {
    const t = s === "custom" ? "custom" : a;
    return /* @__PURE__ */ o.jsxs(
      "div",
      {
        className: g(
          e,
          "tag",
          `tag__size-${r}`,
          `tag__color--${t}`,
          `tag__color--${t}-${s}`
        ),
        style: s === "custom" ? { backgroundColor: a } : void 0,
        ...p,
        ref: f,
        children: [
          l && /* @__PURE__ */ o.jsx(
            j,
            {
              isBold: c,
              className: i,
              icon: d,
              color: t,
              size: r
            }
          ),
          typeof m != "string" ? m : /* @__PURE__ */ o.jsx(o.Fragment, { children: r === "sm" || r === "xs" ? /* @__PURE__ */ o.jsx(n, { bold: c, color: t, children: m }) : /* @__PURE__ */ o.jsx(
            u,
            {
              size: r === "md" ? "sm" : "md",
              bold: c,
              color: t,
              children: m
            }
          ) })
        ]
      }
    );
  }
);
_.displayName = "Tag";
export {
  _ as Tag,
  _ as default
};
