// @deriv-com/quill-ui/dist/components/Notification/banner/index.js
import { j as v } from "../../../jsx-runtime-DmTjM30f.js";
import { useState as I, useRef as s, useEffect as x } from "react";
import { c as h } from "../../../clsx-OuTLNxxd.js";
import { Notification as j } from "../base.js";
const E = ({
  autohideTimeout: c,
  className: o,
  type: u = "info",
  isMobile: e,
  onClick: n,
  onClose: i,
  ...f
}) => {
  const [m, d] = I(!1), t = s(), a = s();
  x(() => (c && (t.current = setInterval(() => {
    r();
  }, c)), () => {
    t.current && clearInterval(t.current), a.current && clearTimeout(a.current);
  }), []);
  const r = () => {
    t.current && clearInterval(t.current), d(!0), a.current = setTimeout(() => {
      i == null || i();
    }, 240);
  }, l = (p) => {
    r(), n == null || n(p);
  };
  return /* @__PURE__ */ v.jsx(
    j,
    {
      ...f,
      className: h(
        `quill-notification__banner${e ? "--mobile" : ""}`,
        m && "hidden",
        o
      ),
      hasCloseButton: !e,
      isBanner: !0,
      onClick: l,
      onClose: r,
      onSwipeDown: e ? void 0 : r,
      onSwipeUp: e ? r : void 0,
      type: u
    }
  );
};
export {
  E as NotificationBanner,
  E as default
};
