// @deriv-com/quill-ui/dist/components/Loader/skeleton/container/index.js
import { j as h } from "../../../../jsx-runtime-DmTjM30f.js";
import { c as _ } from "../../../../clsx-OuTLNxxd.js";
import n from "react";
import '../../../../assets/index10.css';const k = ({
  className: r,
  direction: l = "row",
  gap: i = 10,
  style: o,
  children: s,
  skeletonWidth: a,
  skeletonHeight: m,
  fullWidth: u = !0,
  alignment: p = "left",
  rounded: c,
  ...d
}) => {
  const f = n.Children.map(s, (e) => {
    if (n.isValidElement(e)) {
      const t = e.props;
      return n.cloneElement(e, {
        width: !t.fullWidth && (t.width || a),
        height: t.height || m,
        rounded: c,
        ...t
      });
    }
    return e;
  });
  return /* @__PURE__ */ h.jsx(
    "div",
    {
      className: _(
        "quill-skeleton-container",
        `quill-skeleton-container__direction--${l}`,
        `quill-skeleton-container__alignment--${l}-${p}`,
        {
          "quill-skeleton-container__full-width": u
        },
        r
      ),
      style: { gap: i, ...o },
      ...d,
      children: f
    }
  );
};
k.displayName = "Skeleton.Container";
export {
  k as default
};
