// @deriv-com/quill-ui/dist/components/Snackbar/snackbar.js
import { j as e } from "../../jsx-runtime-DmTjM30f.js";
import k, { useRef as h, useEffect as T } from "react";
import { c as u } from "../../clsx-OuTLNxxd.js";
import "../Typography/base.js";
import "../Typography/heading/index.js";
import { Text as $ } from "../Typography/text/index.js";
import { Button as j } from "../Button/base/index.js";
import { F as z } from "../../snackbar-controller-ChcV5SXl.js";
import "../Button/social/index.js";
import "../Button/icon-button/index.js";
import { useSnackbar as F } from "../../hooks/useSnackbar.js";
import { a as O } from "../../index-EY1gwl5O.js";
const X = ({
  icon: l,
  id: r,
  isVisible: f,
  message: p,
  actionText: d,
  onActionClick: s,
  onCloseAction: c,
  onSnackbarRemove: i,
  hasCloseButton: n = !0,
  hasFixedHeight: v = !0,
  delay: o,
  standalone: q = !0,
  status: a = "neutral",
  ...w
}) => {
  const { removeSnackbar: m } = F(), t = h(null), y = ({
    id: N,
    onSnackbarRemove: R,
    delay: g = 4e3
  }) => setTimeout(() => {
    m(N, R);
  }, g), b = h(null);
  T(() => (r && (a === "neutral" || !n || o !== void 0) && (t.current = y({
    id: r,
    onSnackbarRemove: i,
    delay: o || 4e3
  })), () => {
    t.current && clearTimeout(t.current);
  }), [p, a, n, o, r, i]);
  const x = () => {
    c == null || c(), t.current && clearTimeout(t.current), r && m(r, i);
  }, E = () => {
    s == null || s(), x();
  };
  if (r && O(b, () => m(r, i)), q && !f) return null;
  const _ = {
    neutral: "white-black",
    fail: "white"
  }[a];
  return /* @__PURE__ */ e.jsxs(
    "div",
    {
      ...w,
      className: u(
        "quill-snackbar",
        `quill-snackbar__status--${a}`,
        f ? "slide-up" : "slide-up slide-down"
      ),
      ref: b,
      children: [
        l && /* @__PURE__ */ e.jsx("div", { className: u("quill-snackbar__icon--container"), children: k.isValidElement(l) && k.cloneElement(l, {
          fill: `var(--component-snackbar-icon-${a})`,
          ...l.props
        }) }),
        /* @__PURE__ */ e.jsx("div", { className: "quill-snackbar__message--container", children: /* @__PURE__ */ e.jsx(
          $,
          {
            className: u(
              "quill-snackbar__message",
              v && "quill-snackbar__message--has-fix-height"
            ),
            size: "sm",
            style: {
              color: `var(--component-snackbar-label-color-${a})`
            },
            children: p
          }
        ) }),
        d && /* @__PURE__ */ e.jsx(
          j,
          {
            variant: "tertiary",
            label: d,
            color: _,
            onClick: E
          }
        ),
        n && /* @__PURE__ */ e.jsx(
          j,
          {
            variant: "tertiary",
            iconPosition: "start",
            icon: /* @__PURE__ */ e.jsx(z, {}),
            color: _,
            size: "md",
            onClick: x,
            "data-testid": "close-button"
          }
        )
      ]
    }
  );
};
export {
  X as Snackbar
};
