// @deriv-com/quill-ui/dist/components/Pagination/base/index.js
import { j as r } from "../../../jsx-runtime-DmTjM30f.js";
import { DOTS as n } from "../../../hooks/usePaginationRange.js";
import { CaptionText as e } from "../../Typography/caption/index.js";
import "react";
import "../../Typography/base.js";
import "../../Typography/heading/index.js";
import '../../../assets/pagination-base.css';/* empty css                                */
const j = ({
  currentPage: a,
  handleOnClick: i,
  pageNumber: t,
  variant: o = "number"
}) => o === "bullet" ? /* @__PURE__ */ r.jsx(
  "button",
  {
    onClick: () => i(String(t)),
    "aria-label": `Go to page ${t}`,
    "aria-current": a === t,
    className: "variant__bullet"
  }
) : t === n ? /* @__PURE__ */ r.jsx("button", { disabled: !0, "aria-label": "Hidden pages", className: "hidden_pages", children: /* @__PURE__ */ r.jsx("p", { className: "hidden_pages-dots", children: n }) }) : /* @__PURE__ */ r.jsx(
  "button",
  {
    "aria-label": `Go to page ${t}`,
    "aria-current": a === t,
    onClick: (s) => i(s.currentTarget.textContent),
    className: "variant__number",
    children: /* @__PURE__ */ r.jsx(
      e,
      {
        className: "number-text",
        "aria-current": a === t,
        children: t
      }
    )
  }
);
export {
  j as PaginationButton,
  j as default
};
