// @deriv-com/quill-ui/dist/components/SegmentedControl/segment.js
import { j as e } from "../../jsx-runtime-DmTjM30f.js";
import j, { forwardRef as D, useEffect as k } from "react";
import { c as w } from "../../clsx-OuTLNxxd.js";
import "../Typography/base.js";
import "../Typography/heading/index.js";
import { Text as y } from "../Typography/text/index.js";
import { KEY as E } from "../../utils/common-utils.js";
const _ = D(
  ({
    allowFocus: r,
    className: m,
    icon: a,
    isAnimated: c,
    isDisabled: i,
    isSelected: s,
    label: u,
    onClick: l,
    onKeyDown: p,
    size: f
  }, o) => {
    const [d, n] = j.useState(!1), x = (t) => {
      t.key !== E.TAB && n(!1), p(t);
    }, h = () => {
      s || (n(!1), l());
    };
    return k(() => {
      var t;
      s && !r && (o != null && o.current) && ((t = o == null ? void 0 : o.current) == null || t.focus({
        preventScroll: !0
      }));
    }, [s]), /* @__PURE__ */ e.jsxs(
      "button",
      {
        className: w(
          "item",
          c && "animated",
          d && r && "focused",
          s && !c && "selected",
          m
        ),
        disabled: i,
        onMouseDown: (t) => t.preventDefault(),
        onClick: h,
        onKeyDown: x,
        onFocus: () => n(!0),
        onBlur: () => n(!1),
        ref: o,
        children: [
          a && /* @__PURE__ */ e.jsx("span", { className: "icon", children: a }),
          u && /* @__PURE__ */ e.jsx(y, { size: f, as: "span", children: u })
        ]
      }
    );
  }
);
export {
  _ as Segment
};
