// @deriv-com/quill-ui/dist/components/Link/index.js
import { j as l } from "../../jsx-runtime-DmTjM30f.js";
import { forwardRef as x } from "react";
import { c as o } from "../../clsx-OuTLNxxd.js";
import { CaptionText as f } from "../Typography/caption/index.js";
import "../Typography/base.js";
import "../Typography/heading/index.js";
import { Text as u } from "../Typography/text/index.js";
import { F as _ } from "../../LabelPairedChevronRightSmRegularIcon-BcvI7e_s.js";
import '../../assets/index5.css';const j = x(
  ({
    className: m,
    size: i = "md",
    disabled: t,
    color: a = "black",
    icon: r,
    hasHoverEffect: n = !0,
    hasChevron: e = !1,
    children: s,
    target: p = "_blank",
    href: d,
    ...c
  }, k) => /* @__PURE__ */ l.jsxs(
    "a",
    {
      ref: k,
      target: p,
      href: t ? "javascript:void(0)" : d,
      ...c,
      className: o(
        "quill-link",
        `quill-link__color-${a}`,
        t && "quill-link--disabled",
        n && "quill-link--hover",
        m
      ),
      children: [
        r && /* @__PURE__ */ l.jsx("span", { "data-testid": "dt-link-icon", className: "quill-link__icon", children: r }),
        i === "caption" ? /* @__PURE__ */ l.jsx(f, { color: a, children: s }) : /* @__PURE__ */ l.jsx(
          u,
          {
            color: a,
            size: i === "sm" ? "sm" : i === "lg" ? "lg" : i === "xl" ? "xl" : "md",
            children: s
          }
        ),
        e && /* @__PURE__ */ l.jsx(
          _,
          {
            "data-testid": "dt-link-chevron",
            className: o(
              `quill-link__icon-color-${a}`,
              `quill-link__icon-size-${i}`
            )
          }
        )
      ]
    }
  )
);
j.displayName = "Link";
export {
  j as Link,
  j as default
};
