// @deriv-com/quill-ui/dist/components/Notification/test-buttons.js
import { j as t } from "../../jsx-runtime-DmTjM30f.js";
import { TYPE as s } from "../../utils/notification-utils.js";
import "../../snackbar-controller-ChcV5SXl.js";
import "../../index-C1BOoRZp.js";
import '../../assets/stepper.css';import '../../assets/toggle-switch.css';import '../../assets/segmented-control.css';import '../../assets/section-message.css';import '../../assets/pagination-base.css';import '../../assets/spinner.css';import '../../assets/breadcrumbs.css';import '../../assets/wheel-picker-base.css';import '../../assets/password-validation.css';import '../../assets/date-picker.css';import '../../assets/divider.css';/* empty css                     */
import "../Atom/dropdown/title/index.js";
import "../Atom/dropdown/item-container/index.js";
import * as p from "react";
import { forwardRef as a } from "react";
/* empty css                         */
import "../Typography/base.js";
import "../Typography/heading/index.js";
import { Text as c } from "../Typography/text/index.js";
import "../../dayjs.min-DrYbymKu.js";
/* empty css                                 */
/* empty css                               */
import "../../lodash-rsqVyylH.js";
import "../Badge/index.js";
/* empty css                         */
import "../../providers/breakpoint/breakpointContext.js";
import { Button as e } from "../Button/base/index.js";
import "../Button/social/index.js";
import "../Button/icon-button/index.js";
import "../Chip/index.js";
import "../Input/password-field/index.js";
import "../Input/search-field/index.js";
import "../Input/text-field/index.js";
import "../Input/text-field-addon/index.js";
import "../Input/textarea/index.js";
import "../Link/index.js";
import "../Loader/skeleton/index.js";
/* empty css                     */
import "../Navigation/index.js";
import "./notification-icon.js";
/* empty css                             */
/* empty css                             */
import "../SegmentedControl/segment.js";
/* empty css                               */
import "../../index-C7RXo2Oj.js";
import "../SelectionControl/radio/radio-group/index.js";
/* empty css                           */
/* empty css                     */
import "../../providers/snackbar/snackbarContext.js";
import "react-dom";
import "../Tab/index.js";
import "../Tag/index.js";
import { s as l } from "../../index-CQbz1p6_.js";
const x = ({ iconSize: m = "md", title: o, titleId: r, ...i }, n) => p.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 32 32", ...l[m], role: "img", ref: n, "aria-labelledby": r, ...i },
  o ? p.createElement("title", { id: r }, o) : null,
  p.createElement("path", { d: "M16 8.375c-2.93 0-5.586 1.563-7.07 4.063-1.446 2.539-1.446 5.624 0 8.124A8.13 8.13 0 0 0 16 24.626c2.89 0 5.547-1.523 7.031-4.062 1.446-2.5 1.446-5.586 0-8.125-1.484-2.5-4.14-4.063-7.031-4.063M16 26.5c-3.594 0-6.875-1.875-8.672-5-1.797-3.086-1.797-6.875 0-10 1.797-3.086 5.078-5 8.672-5 3.555 0 6.836 1.914 8.633 5 1.797 3.125 1.797 6.914 0 10a9.93 9.93 0 0 1-8.633 5m-.937-6.562v-2.5h-2.5c-.547 0-.938-.391-.938-.938 0-.508.39-.937.938-.937h2.5v-2.5c0-.508.39-.938.937-.938a.95.95 0 0 1 .938.938v2.5h2.5a.95.95 0 0 1 .937.937c0 .547-.43.938-.937.938h-2.5v2.5c0 .546-.43.937-.938.937-.547 0-.937-.39-.937-.937" })
), f = a(x), et = ({
  addNotification: m,
  markAllAsRead: o,
  removeAllNotifications: r,
  shouldAddBanner: i
}) => /* @__PURE__ */ t.jsxs(
  "div",
  {
    style: {
      margin: i ? "100px 20% 100px 80px" : "32px",
      width: "80%"
    },
    children: [
      /* @__PURE__ */ t.jsxs(c, { children: [
        "Testing button",
        i ? "" : "s",
        ":"
      ] }),
      /* @__PURE__ */ t.jsxs(
        e,
        {
          onClick: () => {
            m({
              icon: /* @__PURE__ */ t.jsx(
                f,
                {
                  fill: "rgba(0, 130, 42, 1)",
                  iconSize: "sm"
                }
              ),
              message: "This is a newly added notification with a custom icon.",
              redirectTo: "https://www.example.com",
              timestamp: i ? void 0 : (/* @__PURE__ */ new Date()).getTime(),
              title: "New notification",
              type: s.SUCCESS
            });
          },
          children: [
            "Add new",
            " ",
            i ? "banner in the end" : "notification at the top"
          ]
        }
      ),
      !i && /* @__PURE__ */ t.jsxs(t.Fragment, { children: [
        /* @__PURE__ */ t.jsx(e, { onClick: o, children: "Mark all as read" }),
        /* @__PURE__ */ t.jsx(e, { onClick: r, children: "Remove all notifications" })
      ] })
    ]
  }
);
export {
  et as NotificationTestButtons
};
