// @deriv-com/quill-ui/dist/components/Input/search-field/index.js
import { j as c } from "../../../jsx-runtime-DmTjM30f.js";
import * as o from "react";
import { forwardRef as n, useState as f, useEffect as p } from "react";
import d from "../base/index.js";
import { s as m } from "../../../index-CQbz1p6_.js";
const u = ({ iconSize: a = "md", title: t, titleId: e, ...r }, l) => o.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 32 32", ...m[a], role: "img", ref: l, "aria-labelledby": e, ...r },
  t ? o.createElement("title", { id: e }, t) : null,
  o.createElement("path", { d: "M16 26.5c-3.594 0-6.875-1.875-8.672-5-1.797-3.086-1.797-6.875 0-10 1.797-3.086 5.078-5 8.672-5 3.555 0 6.836 1.914 8.633 5 1.797 3.125 1.797 6.914 0 10a9.93 9.93 0 0 1-8.633 5m-3.164-13.164c-.39.39-.39.976 0 1.328l1.836 1.836-1.836 1.836c-.39.39-.39.977 0 1.328.351.39.937.39 1.289 0l1.836-1.836 1.836 1.836c.39.39.976.39 1.328 0 .39-.352.39-.937 0-1.328L17.289 16.5l1.836-1.836c.39-.351.39-.937 0-1.328-.352-.352-.937-.352-1.328 0l-1.836 1.836-1.836-1.836a.92.92 0 0 0-1.29 0" })
), w = n(u), x = ({ iconSize: a = "md", title: t, titleId: e, ...r }, l) => o.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 32 32", ...m[a], role: "img", ref: l, "aria-labelledby": e, ...r },
  t ? o.createElement("title", { id: e }, t) : null,
  o.createElement("path", { d: "M21 14.625c0-2.422-1.328-4.687-3.437-5.937-2.149-1.211-4.766-1.211-6.875 0a6.87 6.87 0 0 0-3.438 5.937c0 2.46 1.29 4.727 3.438 5.977 2.109 1.21 4.726 1.21 6.874 0 2.11-1.25 3.438-3.516 3.438-5.977m-1.602 6.172c-1.406 1.25-3.28 1.953-5.273 1.953A8.12 8.12 0 0 1 6 14.625C6 10.172 9.633 6.5 14.125 6.5c4.453 0 8.125 3.672 8.125 8.125 0 2.031-.742 3.867-1.953 5.313l5.508 5.507a.66.66 0 0 1 0 .899.66.66 0 0 1-.899 0z" })
), g = n(x), I = n(
  (a, t) => {
    const { value: e, onChange: r } = a, [l, s] = f(!e), i = /* @__PURE__ */ c.jsx(
      "button",
      {
        className: "icon_wrapper",
        onClick: () => {
          s(!0), r == null || r({
            target: { value: "" }
          });
        },
        children: /* @__PURE__ */ c.jsx(
          w,
          {
            fill: "var(--component-textIcon-normal-subtle)",
            iconSize: "sm"
          }
        )
      }
    );
    return p(() => {
      s(!e);
    }, [e]), /* @__PURE__ */ c.jsx(
      d,
      {
        leftIcon: /* @__PURE__ */ c.jsx(
          g,
          {
            fill: "var(--component-textIcon-normal-default)",
            iconSize: "sm"
          }
        ),
        triggerActionIcon: l ? "" : i,
        placeholder: "Search",
        ref: t,
        ...a
      }
    );
  }
);
export {
  I as SearchField,
  I as default
};
