// @deriv-com/quill-ui/dist/components/Chip/dropdown-chip-single-select/index.js
import { j as e } from "../../../jsx-runtime-DmTjM30f.js";
import { forwardRef as w, useState as h, Fragment as b } from "react";
import { Chip as D } from "../base/index.js";
import { c as F } from "../../../clsx-OuTLNxxd.js";
import { I as l, q as I } from "../../../dropdown-chip-single-select-CbCwkSrZ.js";
import { D as S } from "../../../index-C1BOoRZp.js";
import '../../../assets/wheel-picker-base.css';import '../../../assets/password-validation.css';import '../../../assets/date-picker.css';import '../../../assets/divider.css';/* empty css                        */
import "../../Atom/dropdown/title/index.js";
import "../../Atom/dropdown/item-container/index.js";
/* empty css                            */
import "../../Typography/base.js";
import "../../Typography/heading/index.js";
import "../../../dayjs.min-DrYbymKu.js";
/* empty css                                    */
/* empty css                                  */
import "../../../lodash-rsqVyylH.js";
const d = ({ item: r, size: o }) => {
  const p = {
    sm: "sm",
    md: "sm",
    lg: "md"
  };
  return /* @__PURE__ */ e.jsx(l.Option, { value: r, as: b, disabled: r.disabled, children: ({ disabled: a, selected: s }) => /* @__PURE__ */ e.jsx(
    S,
    {
      label: r.label,
      disabled: a,
      selected: s,
      size: p[o]
    }
  ) });
}, N = w(
  ({
    defaultOption: r,
    size: o = "md",
    icon: p,
    labelTag: a,
    disabled: s,
    options: c,
    onSelectionChange: i,
    className: x,
    ...j
  }, u) => {
    const [t, v] = h(r), f = (m) => {
      v(m), i == null || i(m);
    };
    return /* @__PURE__ */ e.jsx(
      l,
      {
        value: t,
        onChange: f,
        disabled: s,
        children: ({ open: m }) => /* @__PURE__ */ e.jsxs(e.Fragment, { children: [
          /* @__PURE__ */ e.jsx(l.Button, { as: "div", children: /* @__PURE__ */ e.jsx(
            D,
            {
              ...j,
              icon: p,
              size: o,
              labelTag: a,
              ref: u,
              dropdown: !0,
              selected: t.value !== r.value,
              className: "dropdown-button",
              isDropdownOpen: m,
              disabled: s,
              label: t == null ? void 0 : t.label
            }
          ) }),
          /* @__PURE__ */ e.jsx(
            I,
            {
              enter: "enter",
              enterFrom: "enter-from",
              enterTo: "enter-to",
              leave: "leave",
              leaveFrom: "enter-from",
              leaveTo: "leave-to",
              children: /* @__PURE__ */ e.jsxs(
                l.Options,
                {
                  className: F(
                    "dropdown",
                    `dropdown__size--${o}`,
                    x
                  ),
                  children: [
                    /* @__PURE__ */ e.jsx(d, { item: r, size: o }),
                    c.map((n) => /* @__PURE__ */ e.jsx(
                      d,
                      {
                        item: n,
                        size: o
                      },
                      `chip-selectable-item-${n.value}`
                    ))
                  ]
                }
              )
            }
          )
        ] })
      }
    );
  }
);
N.displayName = "DropdownChipSingleSelect";
export {
  N as DropdownChipSingleSelect,
  N as default
};
