// @deriv-com/quill-ui/dist/dropdown-chip-single-select-CbCwkSrZ.js
import * as le from "react";
import P, { useEffect as N, useLayoutEffect as it, useRef as y, useState as V, forwardRef as lt, Fragment as Z, isValidElement as at, cloneElement as ut, createElement as st, createContext as ae, useContext as te, useReducer as ct, createRef as dt, useCallback as ie, useMemo as M } from "react";
import './assets/dropdown-chip-single-select.css';var ft = Object.defineProperty, pt = (e, t, n) => t in e ? ft(e, t, { enumerable: !0, configurable: !0, writable: !0, value: n }) : e[t] = n, Le = (e, t, n) => (pt(e, typeof t != "symbol" ? t + "" : t, n), n);
let vt = class {
  constructor() {
    Le(this, "current", this.detect()), Le(this, "handoffState", "pending"), Le(this, "currentId", 0);
  }
  set(t) {
    this.current !== t && (this.handoffState = "pending", this.currentId = 0, this.current = t);
  }
  reset() {
    this.set(this.detect());
  }
  nextId() {
    return ++this.currentId;
  }
  get isServer() {
    return this.current === "server";
  }
  get isClient() {
    return this.current === "client";
  }
  detect() {
    return typeof window > "u" || typeof document > "u" ? "server" : "client";
  }
  handoff() {
    this.handoffState === "pending" && (this.handoffState = "complete");
  }
  get isHandoffComplete() {
    return this.handoffState === "complete";
  }
}, W = new vt(), D = (e, t) => {
  W.isServer ? N(e, t) : it(e, t);
};
function j(e) {
  let t = y(e);
  return D(() => {
    t.current = e;
  }, [e]), t;
}
function ze(e, t) {
  let [n, r] = V(e), o = j(e);
  return D(() => r(o.current), [o, r, ...t]), n;
}
let h = function(e) {
  let t = j(e);
  return P.useCallback((...n) => t.current(...n), [t]);
};
function mt(e, t, n) {
  let [r, o] = V(n), i = e !== void 0, l = y(i), a = y(!1), u = y(!1);
  return i && !l.current && !a.current ? (a.current = !0, l.current = i, console.error("A component is changing from uncontrolled to controlled. This may be caused by the value changing from undefined to a defined value, which should not happen.")) : !i && l.current && !u.current && (u.current = !0, l.current = i, console.error("A component is changing from controlled to uncontrolled. This may be caused by the value changing from a defined value to undefined, which should not happen.")), [i ? e : r, h((s) => (i || o(s), t == null ? void 0 : t(s)))];
}
function bt(e) {
  typeof queueMicrotask == "function" ? queueMicrotask(e) : Promise.resolve().then(e).catch((t) => setTimeout(() => {
    throw t;
  }));
}
function Q() {
  let e = [], t = { addEventListener(n, r, o, i) {
    return n.addEventListener(r, o, i), t.add(() => n.removeEventListener(r, o, i));
  }, requestAnimationFrame(...n) {
    let r = requestAnimationFrame(...n);
    return t.add(() => cancelAnimationFrame(r));
  }, nextFrame(...n) {
    return t.requestAnimationFrame(() => t.requestAnimationFrame(...n));
  }, setTimeout(...n) {
    let r = setTimeout(...n);
    return t.add(() => clearTimeout(r));
  }, microTask(...n) {
    let r = { current: !0 };
    return bt(() => {
      r.current && n[0]();
    }), t.add(() => {
      r.current = !1;
    });
  }, style(n, r, o) {
    let i = n.style.getPropertyValue(r);
    return Object.assign(n.style, { [r]: o }), this.add(() => {
      Object.assign(n.style, { [r]: i });
    });
  }, group(n) {
    let r = Q();
    return n(r), this.add(() => r.dispose());
  }, add(n) {
    return e.push(n), () => {
      let r = e.indexOf(n);
      if (r >= 0) for (let o of e.splice(r, 1)) o();
    };
  }, dispose() {
    for (let n of e.splice(0)) n();
  } };
  return t;
}
function ee() {
  let [e] = V(Q);
  return N(() => () => e.dispose(), [e]), e;
}
function ht() {
  let e = typeof document > "u";
  return "useSyncExternalStore" in le ? ((t) => t.useSyncExternalStore)(le)(() => () => {
  }, () => !1, () => !e) : !1;
}
function Ne() {
  let e = ht(), [t, n] = le.useState(W.isHandoffComplete);
  return t && W.isHandoffComplete === !1 && n(!1), le.useEffect(() => {
    t !== !0 && n(!0);
  }, [t]), le.useEffect(() => W.handoff(), []), e ? !1 : t;
}
var Ue;
let be = (Ue = P.useId) != null ? Ue : function() {
  let e = Ne(), [t, n] = P.useState(e ? () => W.nextId() : null);
  return D(() => {
    t === null && n(W.nextId());
  }, [t]), t != null ? "" + t : void 0;
};
function R(e, t, ...n) {
  if (e in t) {
    let o = t[e];
    return typeof o == "function" ? o(...n) : o;
  }
  let r = new Error(`Tried to handle "${e}" but there is no handler defined. Only defined handlers are: ${Object.keys(t).map((o) => `"${o}"`).join(", ")}.`);
  throw Error.captureStackTrace && Error.captureStackTrace(r, R), r;
}
function Ge(e) {
  return W.isServer ? null : e instanceof Node ? e.ownerDocument : e != null && e.hasOwnProperty("current") && e.current instanceof Node ? e.current.ownerDocument : document;
}
let Be = ["[contentEditable=true]", "[tabindex]", "a[href]", "area[href]", "button:not([disabled])", "iframe", "input:not([disabled])", "select:not([disabled])", "textarea:not([disabled])"].map((e) => `${e}:not([tabindex='-1'])`).join(",");
var gt = ((e) => (e[e.First = 1] = "First", e[e.Previous = 2] = "Previous", e[e.Next = 4] = "Next", e[e.Last = 8] = "Last", e[e.WrapAround = 16] = "WrapAround", e[e.NoScroll = 32] = "NoScroll", e))(gt || {}), xt = ((e) => (e[e.Error = 0] = "Error", e[e.Overflow = 1] = "Overflow", e[e.Success = 2] = "Success", e[e.Underflow = 3] = "Underflow", e))(xt || {}), yt = ((e) => (e[e.Previous = -1] = "Previous", e[e.Next = 1] = "Next", e))(yt || {}), De = ((e) => (e[e.Strict = 0] = "Strict", e[e.Loose = 1] = "Loose", e))(De || {});
function _e(e, t = 0) {
  var n;
  return e === ((n = Ge(e)) == null ? void 0 : n.body) ? !1 : R(t, { 0() {
    return e.matches(Be);
  }, 1() {
    let r = e;
    for (; r !== null; ) {
      if (r.matches(Be)) return !0;
      r = r.parentElement;
    }
    return !1;
  } });
}
var Ot = ((e) => (e[e.Keyboard = 0] = "Keyboard", e[e.Mouse = 1] = "Mouse", e))(Ot || {});
typeof window < "u" && typeof document < "u" && (document.addEventListener("keydown", (e) => {
  e.metaKey || e.altKey || e.ctrlKey || (document.documentElement.dataset.headlessuiFocusVisible = "");
}, !0), document.addEventListener("click", (e) => {
  e.detail === 1 ? delete document.documentElement.dataset.headlessuiFocusVisible : e.detail === 0 && (document.documentElement.dataset.headlessuiFocusVisible = "");
}, !0));
function wt(e, t = (n) => n) {
  return e.slice().sort((n, r) => {
    let o = t(n), i = t(r);
    if (o === null || i === null) return 0;
    let l = o.compareDocumentPosition(i);
    return l & Node.DOCUMENT_POSITION_FOLLOWING ? -1 : l & Node.DOCUMENT_POSITION_PRECEDING ? 1 : 0;
  });
}
function St() {
  return /iPhone/gi.test(window.navigator.platform) || /Mac/gi.test(window.navigator.platform) && window.navigator.maxTouchPoints > 0;
}
function Et() {
  return /Android/gi.test(window.navigator.userAgent);
}
function Lt() {
  return St() || Et();
}
function fe(e, t, n) {
  let r = j(t);
  N(() => {
    function o(i) {
      r.current(i);
    }
    return document.addEventListener(e, o, n), () => document.removeEventListener(e, o, n);
  }, [e, n]);
}
function Tt(e, t, n) {
  let r = j(t);
  N(() => {
    function o(i) {
      r.current(i);
    }
    return window.addEventListener(e, o, n), () => window.removeEventListener(e, o, n);
  }, [e, n]);
}
function Pt(e, t, n = !0) {
  let r = y(!1);
  N(() => {
    requestAnimationFrame(() => {
      r.current = n;
    });
  }, [n]);
  function o(l, a) {
    if (!r.current || l.defaultPrevented) return;
    let u = a(l);
    if (u === null || !u.getRootNode().contains(u) || !u.isConnected) return;
    let s = function c(v) {
      return typeof v == "function" ? c(v()) : Array.isArray(v) || v instanceof Set ? v : [v];
    }(e);
    for (let c of s) {
      if (c === null) continue;
      let v = c instanceof HTMLElement ? c : c.current;
      if (v != null && v.contains(u) || l.composed && l.composedPath().includes(v)) return;
    }
    return !_e(u, De.Loose) && u.tabIndex !== -1 && l.preventDefault(), t(l, u);
  }
  let i = y(null);
  fe("pointerdown", (l) => {
    var a, u;
    r.current && (i.current = ((u = (a = l.composedPath) == null ? void 0 : a.call(l)) == null ? void 0 : u[0]) || l.target);
  }, !0), fe("mousedown", (l) => {
    var a, u;
    r.current && (i.current = ((u = (a = l.composedPath) == null ? void 0 : a.call(l)) == null ? void 0 : u[0]) || l.target);
  }, !0), fe("click", (l) => {
    Lt() || i.current && (o(l, () => i.current), i.current = null);
  }, !0), fe("touchend", (l) => o(l, () => l.target instanceof HTMLElement ? l.target : null), !0), Tt("blur", (l) => o(l, () => window.document.activeElement instanceof HTMLIFrameElement ? window.document.activeElement : null), !0);
}
function qe(e) {
  var t;
  if (e.type) return e.type;
  let n = (t = e.as) != null ? t : "button";
  if (typeof n == "string" && n.toLowerCase() === "button") return "button";
}
function Rt(e, t) {
  let [n, r] = V(() => qe(e));
  return D(() => {
    r(qe(e));
  }, [e.type, e.as]), D(() => {
    n || t.current && t.current instanceof HTMLButtonElement && !t.current.hasAttribute("type") && r("button");
  }, [n, t]), n;
}
let $t = Symbol();
function z(...e) {
  let t = y(e);
  N(() => {
    t.current = e;
  }, [e]);
  let n = h((r) => {
    for (let o of t.current) o != null && (typeof o == "function" ? o(r) : o.current = r);
  });
  return e.every((r) => r == null || (r == null ? void 0 : r[$t])) ? void 0 : n;
}
function Ve(e) {
  return [e.screenX, e.screenY];
}
function Ft() {
  let e = y([-1, -1]);
  return { wasMoved(t) {
    let n = Ve(t);
    return e.current[0] === n[0] && e.current[1] === n[1] ? !1 : (e.current = n, !0);
  }, update(t) {
    e.current = Ve(t);
  } };
}
function ve(...e) {
  return Array.from(new Set(e.flatMap((t) => typeof t == "string" ? t.split(" ") : []))).filter(Boolean).join(" ");
}
var me = ((e) => (e[e.None = 0] = "None", e[e.RenderStrategy = 1] = "RenderStrategy", e[e.Static = 2] = "Static", e))(me || {}), q = ((e) => (e[e.Unmount = 0] = "Unmount", e[e.Hidden = 1] = "Hidden", e))(q || {});
function K({ ourProps: e, theirProps: t, slot: n, defaultTag: r, features: o, visible: i = !0, name: l, mergeRefs: a }) {
  a = a ?? It;
  let u = Ye(t, e);
  if (i) return pe(u, n, r, l, a);
  let s = o ?? 0;
  if (s & 2) {
    let { static: c = !1, ...v } = u;
    if (c) return pe(v, n, r, l, a);
  }
  if (s & 1) {
    let { unmount: c = !0, ...v } = u;
    return R(c ? 0 : 1, { 0() {
      return null;
    }, 1() {
      return pe({ ...v, hidden: !0, style: { display: "none" } }, n, r, l, a);
    } });
  }
  return pe(u, n, r, l, a);
}
function pe(e, t = {}, n, r, o) {
  let { as: i = n, children: l, refName: a = "ref", ...u } = Te(e, ["unmount", "static"]), s = e.ref !== void 0 ? { [a]: e.ref } : {}, c = typeof l == "function" ? l(t) : l;
  "className" in u && u.className && typeof u.className == "function" && (u.className = u.className(t));
  let v = {};
  if (t) {
    let x = !1, g = [];
    for (let [p, m] of Object.entries(t)) typeof m == "boolean" && (x = !0), m === !0 && g.push(p);
    x && (v["data-headlessui-state"] = g.join(" "));
  }
  if (i === Z && Object.keys(Fe(u)).length > 0) {
    if (!at(c) || Array.isArray(c) && c.length > 1) throw new Error(['Passing props on "Fragment"!', "", `The current component <${r} /> is rendering a "Fragment".`, "However we need to passthrough the following props:", Object.keys(u).map((m) => `  - ${m}`).join(`
`), "", "You can apply a few solutions:", ['Add an `as="..."` prop, to ensure that we render an actual element instead of a "Fragment".', "Render a single element as the child so that we can forward the props onto that element."].map((m) => `  - ${m}`).join(`
`)].join(`
`));
    let x = c.props, g = typeof (x == null ? void 0 : x.className) == "function" ? (...m) => ve(x == null ? void 0 : x.className(...m), u.className) : ve(x == null ? void 0 : x.className, u.className), p = g ? { className: g } : {};
    return ut(c, Object.assign({}, Ye(c.props, Fe(Te(u, ["ref"]))), v, s, { ref: o(c.ref, s.ref) }, p));
  }
  return st(i, Object.assign({}, Te(u, ["ref"]), i !== Z && s, i !== Z && v), c);
}
function It(...e) {
  return e.every((t) => t == null) ? void 0 : (t) => {
    for (let n of e) n != null && (typeof n == "function" ? n(t) : n.current = t);
  };
}
function Ye(...e) {
  if (e.length === 0) return {};
  if (e.length === 1) return e[0];
  let t = {}, n = {};
  for (let r of e) for (let o in r) o.startsWith("on") && typeof r[o] == "function" ? (n[o] != null || (n[o] = []), n[o].push(r[o])) : t[o] = r[o];
  if (t.disabled || t["aria-disabled"]) return Object.assign(t, Object.fromEntries(Object.keys(n).map((r) => [r, void 0])));
  for (let r in n) Object.assign(t, { [r](o, ...i) {
    let l = n[r];
    for (let a of l) {
      if ((o instanceof Event || (o == null ? void 0 : o.nativeEvent) instanceof Event) && o.defaultPrevented) return;
      a(o, ...i);
    }
  } });
  return t;
}
function U(e) {
  var t;
  return Object.assign(lt(e), { displayName: (t = e.displayName) != null ? t : e.name });
}
function Fe(e) {
  let t = Object.assign({}, e);
  for (let n in t) t[n] === void 0 && delete t[n];
  return t;
}
function Te(e, t = []) {
  let n = Object.assign({}, e);
  for (let r of t) r in n && delete n[r];
  return n;
}
let Nt = "div";
var Xe = ((e) => (e[e.None = 1] = "None", e[e.Focusable = 2] = "Focusable", e[e.Hidden = 4] = "Hidden", e))(Xe || {});
function Dt(e, t) {
  var n;
  let { features: r = 1, ...o } = e, i = { ref: t, "aria-hidden": (r & 2) === 2 ? !0 : (n = o["aria-hidden"]) != null ? n : void 0, style: { position: "fixed", top: 1, left: 1, width: 1, height: 0, padding: 0, margin: -1, overflow: "hidden", clip: "rect(0, 0, 0, 0)", whiteSpace: "nowrap", borderWidth: "0", ...(r & 4) === 4 && (r & 2) !== 2 && { display: "none" } } };
  return K({ ourProps: i, theirProps: o, slot: {}, defaultTag: Nt, name: "Hidden" });
}
let At = U(Dt), Ae = ae(null);
Ae.displayName = "OpenClosedContext";
var I = ((e) => (e[e.Open = 1] = "Open", e[e.Closed = 2] = "Closed", e[e.Closing = 4] = "Closing", e[e.Opening = 8] = "Opening", e))(I || {});
function Ce() {
  return te(Ae);
}
function Je({ value: e, children: t }) {
  return P.createElement(Ae.Provider, { value: e }, t);
}
function Ct(e) {
  let t = e.parentElement, n = null;
  for (; t && !(t instanceof HTMLFieldSetElement); ) t instanceof HTMLLegendElement && (n = t), t = t.parentElement;
  let r = (t == null ? void 0 : t.getAttribute("disabled")) === "";
  return r && kt(n) ? !1 : r;
}
function kt(e) {
  if (!e) return !1;
  let t = e.previousElementSibling;
  for (; t !== null; ) {
    if (t instanceof HTMLLegendElement) return !1;
    t = t.previousElementSibling;
  }
  return !0;
}
function jt(e) {
  throw new Error("Unexpected object: " + e);
}
var F = ((e) => (e[e.First = 0] = "First", e[e.Previous = 1] = "Previous", e[e.Next = 2] = "Next", e[e.Last = 3] = "Last", e[e.Specific = 4] = "Specific", e[e.Nothing = 5] = "Nothing", e))(F || {});
function Ht(e, t) {
  let n = t.resolveItems();
  if (n.length <= 0) return null;
  let r = t.resolveActiveIndex(), o = r ?? -1;
  switch (e.focus) {
    case 0: {
      for (let i = 0; i < n.length; ++i) if (!t.resolveDisabled(n[i], i, n)) return i;
      return r;
    }
    case 1: {
      for (let i = o - 1; i >= 0; --i) if (!t.resolveDisabled(n[i], i, n)) return i;
      return r;
    }
    case 2: {
      for (let i = o + 1; i < n.length; ++i) if (!t.resolveDisabled(n[i], i, n)) return i;
      return r;
    }
    case 3: {
      for (let i = n.length - 1; i >= 0; --i) if (!t.resolveDisabled(n[i], i, n)) return i;
      return r;
    }
    case 4: {
      for (let i = 0; i < n.length; ++i) if (t.resolveId(n[i], i, n) === e.id) return i;
      return r;
    }
    case 5:
      return null;
    default:
      jt(e);
  }
}
function Ze(e = {}, t = null, n = []) {
  for (let [r, o] of Object.entries(e)) tt(n, et(t, r), o);
  return n;
}
function et(e, t) {
  return e ? e + "[" + t + "]" : t;
}
function tt(e, t, n) {
  if (Array.isArray(n)) for (let [r, o] of n.entries()) tt(e, et(t, r.toString()), o);
  else n instanceof Date ? e.push([t, n.toISOString()]) : typeof n == "boolean" ? e.push([t, n ? "1" : "0"]) : typeof n == "string" ? e.push([t, n]) : typeof n == "number" ? e.push([t, `${n}`]) : n == null ? e.push([t, ""]) : Ze(n, t, e);
}
var T = ((e) => (e.Space = " ", e.Enter = "Enter", e.Escape = "Escape", e.Backspace = "Backspace", e.Delete = "Delete", e.ArrowLeft = "ArrowLeft", e.ArrowUp = "ArrowUp", e.ArrowRight = "ArrowRight", e.ArrowDown = "ArrowDown", e.Home = "Home", e.End = "End", e.PageUp = "PageUp", e.PageDown = "PageDown", e.Tab = "Tab", e))(T || {});
function ke() {
  let e = y(!1);
  return D(() => (e.current = !0, () => {
    e.current = !1;
  }), []), e;
}
let Qe = /([\u2700-\u27BF]|[\uE000-\uF8FF]|\uD83C[\uDC00-\uDFFF]|\uD83D[\uDC00-\uDFFF]|[\u2011-\u26FF]|\uD83E[\uDD10-\uDDFF])/g;
function Ke(e) {
  var t, n;
  let r = (t = e.innerText) != null ? t : "", o = e.cloneNode(!0);
  if (!(o instanceof HTMLElement)) return r;
  let i = !1;
  for (let a of o.querySelectorAll('[hidden],[aria-hidden],[role="img"]')) a.remove(), i = !0;
  let l = i ? (n = o.innerText) != null ? n : "" : r;
  return Qe.test(l) && (l = l.replace(Qe, "")), l;
}
function Mt(e) {
  let t = e.getAttribute("aria-label");
  if (typeof t == "string") return t.trim();
  let n = e.getAttribute("aria-labelledby");
  if (n) {
    let r = n.split(" ").map((o) => {
      let i = document.getElementById(o);
      if (i) {
        let l = i.getAttribute("aria-label");
        return typeof l == "string" ? l.trim() : Ke(i).trim();
      }
      return null;
    }).filter(Boolean);
    if (r.length > 0) return r.join(", ");
  }
  return Ke(e).trim();
}
function Ut(e) {
  let t = y(""), n = y("");
  return h(() => {
    let r = e.current;
    if (!r) return "";
    let o = r.innerText;
    if (t.current === o) return n.current;
    let i = Mt(r).trim().toLowerCase();
    return t.current = o, n.current = i, i;
  });
}
var Bt = ((e) => (e[e.Open = 0] = "Open", e[e.Closed = 1] = "Closed", e))(Bt || {}), qt = ((e) => (e[e.Single = 0] = "Single", e[e.Multi = 1] = "Multi", e))(qt || {}), Vt = ((e) => (e[e.Pointer = 0] = "Pointer", e[e.Other = 1] = "Other", e))(Vt || {}), Qt = ((e) => (e[e.OpenListbox = 0] = "OpenListbox", e[e.CloseListbox = 1] = "CloseListbox", e[e.GoToOption = 2] = "GoToOption", e[e.Search = 3] = "Search", e[e.ClearSearch = 4] = "ClearSearch", e[e.RegisterOption = 5] = "RegisterOption", e[e.UnregisterOption = 6] = "UnregisterOption", e[e.RegisterLabel = 7] = "RegisterLabel", e))(Qt || {});
function Pe(e, t = (n) => n) {
  let n = e.activeOptionIndex !== null ? e.options[e.activeOptionIndex] : null, r = wt(t(e.options.slice()), (i) => i.dataRef.current.domRef.current), o = n ? r.indexOf(n) : null;
  return o === -1 && (o = null), { options: r, activeOptionIndex: o };
}
let Kt = { 1(e) {
  return e.dataRef.current.disabled || e.listboxState === 1 ? e : { ...e, activeOptionIndex: null, listboxState: 1 };
}, 0(e) {
  if (e.dataRef.current.disabled || e.listboxState === 0) return e;
  let t = e.activeOptionIndex, { isSelected: n } = e.dataRef.current, r = e.options.findIndex((o) => n(o.dataRef.current.value));
  return r !== -1 && (t = r), { ...e, listboxState: 0, activeOptionIndex: t };
}, 2(e, t) {
  var n;
  if (e.dataRef.current.disabled || e.listboxState === 1) return e;
  let r = Pe(e), o = Ht(t, { resolveItems: () => r.options, resolveActiveIndex: () => r.activeOptionIndex, resolveId: (i) => i.id, resolveDisabled: (i) => i.dataRef.current.disabled });
  return { ...e, ...r, searchQuery: "", activeOptionIndex: o, activationTrigger: (n = t.trigger) != null ? n : 1 };
}, 3: (e, t) => {
  if (e.dataRef.current.disabled || e.listboxState === 1) return e;
  let n = e.searchQuery !== "" ? 0 : 1, r = e.searchQuery + t.value.toLowerCase(), o = (e.activeOptionIndex !== null ? e.options.slice(e.activeOptionIndex + n).concat(e.options.slice(0, e.activeOptionIndex + n)) : e.options).find((l) => {
    var a;
    return !l.dataRef.current.disabled && ((a = l.dataRef.current.textValue) == null ? void 0 : a.startsWith(r));
  }), i = o ? e.options.indexOf(o) : -1;
  return i === -1 || i === e.activeOptionIndex ? { ...e, searchQuery: r } : { ...e, searchQuery: r, activeOptionIndex: i, activationTrigger: 1 };
}, 4(e) {
  return e.dataRef.current.disabled || e.listboxState === 1 || e.searchQuery === "" ? e : { ...e, searchQuery: "" };
}, 5: (e, t) => {
  let n = { id: t.id, dataRef: t.dataRef }, r = Pe(e, (o) => [...o, n]);
  return e.activeOptionIndex === null && e.dataRef.current.isSelected(t.dataRef.current.value) && (r.activeOptionIndex = r.options.indexOf(n)), { ...e, ...r };
}, 6: (e, t) => {
  let n = Pe(e, (r) => {
    let o = r.findIndex((i) => i.id === t.id);
    return o !== -1 && r.splice(o, 1), r;
  });
  return { ...e, ...n, activationTrigger: 1 };
}, 7: (e, t) => ({ ...e, labelId: t.id }) }, je = ae(null);
je.displayName = "ListboxActionsContext";
function ue(e) {
  let t = te(je);
  if (t === null) {
    let n = new Error(`<${e} /> is missing a parent <Listbox /> component.`);
    throw Error.captureStackTrace && Error.captureStackTrace(n, ue), n;
  }
  return t;
}
let He = ae(null);
He.displayName = "ListboxDataContext";
function se(e) {
  let t = te(He);
  if (t === null) {
    let n = new Error(`<${e} /> is missing a parent <Listbox /> component.`);
    throw Error.captureStackTrace && Error.captureStackTrace(n, se), n;
  }
  return t;
}
function Wt(e, t) {
  return R(t.type, Kt, e, t);
}
let zt = Z;
function Gt(e, t) {
  let { value: n, defaultValue: r, form: o, name: i, onChange: l, by: a = (b, w) => b === w, disabled: u = !1, horizontal: s = !1, multiple: c = !1, ...v } = e;
  const x = s ? "horizontal" : "vertical";
  let g = z(t), [p = c ? [] : void 0, m] = mt(n, l, r), [f, d] = ct(Wt, { dataRef: dt(), listboxState: 1, options: [], searchQuery: "", labelId: null, activeOptionIndex: null, activationTrigger: 1 }), O = y({ static: !1, hold: !1 }), L = y(null), A = y(null), G = y(null), S = h(typeof a == "string" ? (b, w) => {
    let $ = a;
    return (b == null ? void 0 : b[$]) === (w == null ? void 0 : w[$]);
  } : a), C = ie((b) => R(E.mode, { 1: () => p.some((w) => S(w, b)), 0: () => S(p, b) }), [p]), E = M(() => ({ ...f, value: p, disabled: u, mode: c ? 1 : 0, orientation: x, compare: S, isSelected: C, optionsPropsRef: O, labelRef: L, buttonRef: A, optionsRef: G }), [p, u, c, f]);
  D(() => {
    f.dataRef.current = E;
  }, [E]), Pt([E.buttonRef, E.optionsRef], (b, w) => {
    var $;
    d({ type: 1 }), _e(w, De.Loose) || (b.preventDefault(), ($ = E.buttonRef.current) == null || $.focus());
  }, E.listboxState === 0);
  let _ = M(() => ({ open: E.listboxState === 0, disabled: u, value: p }), [E, u, p]), Y = h((b) => {
    let w = E.options.find(($) => $.id === b);
    w && H(w.dataRef.current.value);
  }), ne = h(() => {
    if (E.activeOptionIndex !== null) {
      let { dataRef: b, id: w } = E.options[E.activeOptionIndex];
      H(b.current.value), d({ type: 2, focus: F.Specific, id: w });
    }
  }), X = h(() => d({ type: 0 })), re = h(() => d({ type: 1 })), ye = h((b, w, $) => b === F.Specific ? d({ type: 2, focus: F.Specific, id: w, trigger: $ }) : d({ type: 2, focus: b, trigger: $ })), ce = h((b, w) => (d({ type: 5, id: b, dataRef: w }), () => d({ type: 6, id: b }))), Oe = h((b) => (d({ type: 7, id: b }), () => d({ type: 7, id: null }))), H = h((b) => R(E.mode, { 0() {
    return m == null ? void 0 : m(b);
  }, 1() {
    let w = E.value.slice(), $ = w.findIndex((de) => S(de, b));
    return $ === -1 ? w.push(b) : w.splice($, 1), m == null ? void 0 : m(w);
  } })), we = h((b) => d({ type: 3, value: b })), Se = h(() => d({ type: 4 })), J = M(() => ({ onChange: H, registerOption: ce, registerLabel: Oe, goToOption: ye, closeListbox: re, openListbox: X, selectActiveOption: ne, selectOption: Y, search: we, clearSearch: Se }), []), oe = { ref: g }, k = y(null), Ee = ee();
  return N(() => {
    k.current && r !== void 0 && Ee.addEventListener(k.current, "reset", () => {
      m == null || m(r);
    });
  }, [k, m]), P.createElement(je.Provider, { value: J }, P.createElement(He.Provider, { value: E }, P.createElement(Je, { value: R(E.listboxState, { 0: I.Open, 1: I.Closed }) }, i != null && p != null && Ze({ [i]: p }).map(([b, w], $) => P.createElement(At, { features: Xe.Hidden, ref: $ === 0 ? (de) => {
    var Me;
    k.current = (Me = de == null ? void 0 : de.closest("form")) != null ? Me : null;
  } : void 0, ...Fe({ key: b, as: "input", type: "hidden", hidden: !0, readOnly: !0, form: o, name: b, value: w }) })), K({ ourProps: oe, theirProps: v, slot: _, defaultTag: zt, name: "Listbox" }))));
}
let _t = "button";
function Yt(e, t) {
  var n;
  let r = be(), { id: o = `headlessui-listbox-button-${r}`, ...i } = e, l = se("Listbox.Button"), a = ue("Listbox.Button"), u = z(l.buttonRef, t), s = ee(), c = h((f) => {
    switch (f.key) {
      case T.Space:
      case T.Enter:
      case T.ArrowDown:
        f.preventDefault(), a.openListbox(), s.nextFrame(() => {
          l.value || a.goToOption(F.First);
        });
        break;
      case T.ArrowUp:
        f.preventDefault(), a.openListbox(), s.nextFrame(() => {
          l.value || a.goToOption(F.Last);
        });
        break;
    }
  }), v = h((f) => {
    switch (f.key) {
      case T.Space:
        f.preventDefault();
        break;
    }
  }), x = h((f) => {
    if (Ct(f.currentTarget)) return f.preventDefault();
    l.listboxState === 0 ? (a.closeListbox(), s.nextFrame(() => {
      var d;
      return (d = l.buttonRef.current) == null ? void 0 : d.focus({ preventScroll: !0 });
    })) : (f.preventDefault(), a.openListbox());
  }), g = ze(() => {
    if (l.labelId) return [l.labelId, o].join(" ");
  }, [l.labelId, o]), p = M(() => ({ open: l.listboxState === 0, disabled: l.disabled, value: l.value }), [l]), m = { ref: u, id: o, type: Rt(e, l.buttonRef), "aria-haspopup": "listbox", "aria-controls": (n = l.optionsRef.current) == null ? void 0 : n.id, "aria-expanded": l.listboxState === 0, "aria-labelledby": g, disabled: l.disabled, onKeyDown: c, onKeyUp: v, onClick: x };
  return K({ ourProps: m, theirProps: i, slot: p, defaultTag: _t, name: "Listbox.Button" });
}
let Xt = "label";
function Jt(e, t) {
  let n = be(), { id: r = `headlessui-listbox-label-${n}`, ...o } = e, i = se("Listbox.Label"), l = ue("Listbox.Label"), a = z(i.labelRef, t);
  D(() => l.registerLabel(r), [r]);
  let u = h(() => {
    var c;
    return (c = i.buttonRef.current) == null ? void 0 : c.focus({ preventScroll: !0 });
  }), s = M(() => ({ open: i.listboxState === 0, disabled: i.disabled }), [i]);
  return K({ ourProps: { ref: a, id: r, onClick: u }, theirProps: o, slot: s, defaultTag: Xt, name: "Listbox.Label" });
}
let Zt = "ul", en = me.RenderStrategy | me.Static;
function tn(e, t) {
  var n;
  let r = be(), { id: o = `headlessui-listbox-options-${r}`, ...i } = e, l = se("Listbox.Options"), a = ue("Listbox.Options"), u = z(l.optionsRef, t), s = ee(), c = ee(), v = Ce(), x = v !== null ? (v & I.Open) === I.Open : l.listboxState === 0;
  N(() => {
    var d;
    let O = l.optionsRef.current;
    O && l.listboxState === 0 && O !== ((d = Ge(O)) == null ? void 0 : d.activeElement) && O.focus({ preventScroll: !0 });
  }, [l.listboxState, l.optionsRef]);
  let g = h((d) => {
    switch (c.dispose(), d.key) {
      case T.Space:
        if (l.searchQuery !== "") return d.preventDefault(), d.stopPropagation(), a.search(d.key);
      case T.Enter:
        if (d.preventDefault(), d.stopPropagation(), l.activeOptionIndex !== null) {
          let { dataRef: O } = l.options[l.activeOptionIndex];
          a.onChange(O.current.value);
        }
        l.mode === 0 && (a.closeListbox(), Q().nextFrame(() => {
          var O;
          return (O = l.buttonRef.current) == null ? void 0 : O.focus({ preventScroll: !0 });
        }));
        break;
      case R(l.orientation, { vertical: T.ArrowDown, horizontal: T.ArrowRight }):
        return d.preventDefault(), d.stopPropagation(), a.goToOption(F.Next);
      case R(l.orientation, { vertical: T.ArrowUp, horizontal: T.ArrowLeft }):
        return d.preventDefault(), d.stopPropagation(), a.goToOption(F.Previous);
      case T.Home:
      case T.PageUp:
        return d.preventDefault(), d.stopPropagation(), a.goToOption(F.First);
      case T.End:
      case T.PageDown:
        return d.preventDefault(), d.stopPropagation(), a.goToOption(F.Last);
      case T.Escape:
        return d.preventDefault(), d.stopPropagation(), a.closeListbox(), s.nextFrame(() => {
          var O;
          return (O = l.buttonRef.current) == null ? void 0 : O.focus({ preventScroll: !0 });
        });
      case T.Tab:
        d.preventDefault(), d.stopPropagation();
        break;
      default:
        d.key.length === 1 && (a.search(d.key), c.setTimeout(() => a.clearSearch(), 350));
        break;
    }
  }), p = ze(() => {
    var d;
    return (d = l.buttonRef.current) == null ? void 0 : d.id;
  }, [l.buttonRef.current]), m = M(() => ({ open: l.listboxState === 0 }), [l]), f = { "aria-activedescendant": l.activeOptionIndex === null || (n = l.options[l.activeOptionIndex]) == null ? void 0 : n.id, "aria-multiselectable": l.mode === 1 ? !0 : void 0, "aria-labelledby": p, "aria-orientation": l.orientation, id: o, onKeyDown: g, role: "listbox", tabIndex: 0, ref: u };
  return K({ ourProps: f, theirProps: i, slot: m, defaultTag: Zt, features: en, visible: x, name: "Listbox.Options" });
}
let nn = "li";
function rn(e, t) {
  let n = be(), { id: r = `headlessui-listbox-option-${n}`, disabled: o = !1, value: i, ...l } = e, a = se("Listbox.Option"), u = ue("Listbox.Option"), s = a.activeOptionIndex !== null ? a.options[a.activeOptionIndex].id === r : !1, c = a.isSelected(i), v = y(null), x = Ut(v), g = j({ disabled: o, value: i, domRef: v, get textValue() {
    return x();
  } }), p = z(t, v);
  D(() => {
    if (a.listboxState !== 0 || !s || a.activationTrigger === 0) return;
    let S = Q();
    return S.requestAnimationFrame(() => {
      var C, E;
      (E = (C = v.current) == null ? void 0 : C.scrollIntoView) == null || E.call(C, { block: "nearest" });
    }), S.dispose;
  }, [v, s, a.listboxState, a.activationTrigger, a.activeOptionIndex]), D(() => u.registerOption(r, g), [g, r]);
  let m = h((S) => {
    if (o) return S.preventDefault();
    u.onChange(i), a.mode === 0 && (u.closeListbox(), Q().nextFrame(() => {
      var C;
      return (C = a.buttonRef.current) == null ? void 0 : C.focus({ preventScroll: !0 });
    }));
  }), f = h(() => {
    if (o) return u.goToOption(F.Nothing);
    u.goToOption(F.Specific, r);
  }), d = Ft(), O = h((S) => d.update(S)), L = h((S) => {
    d.wasMoved(S) && (o || s || u.goToOption(F.Specific, r, 0));
  }), A = h((S) => {
    d.wasMoved(S) && (o || s && u.goToOption(F.Nothing));
  }), G = M(() => ({ active: s, selected: c, disabled: o }), [s, c, o]);
  return K({ ourProps: { id: r, ref: p, role: "option", tabIndex: o === !0 ? void 0 : -1, "aria-disabled": o === !0 ? !0 : void 0, "aria-selected": c, disabled: void 0, onClick: m, onFocus: f, onPointerEnter: O, onMouseEnter: O, onPointerMove: L, onMouseMove: L, onPointerLeave: A, onMouseLeave: A }, theirProps: l, slot: G, defaultTag: nn, name: "Listbox.Option" });
}
let on = U(Gt), ln = U(Yt), an = U(Jt), un = U(tn), sn = U(rn), Rn = Object.assign(on, { Button: ln, Label: an, Options: un, Option: sn });
function cn(e = 0) {
  let [t, n] = V(e), r = ke(), o = ie((u) => {
    r.current && n((s) => s | u);
  }, [t, r]), i = ie((u) => !!(t & u), [t]), l = ie((u) => {
    r.current && n((s) => s & ~u);
  }, [n, r]), a = ie((u) => {
    r.current && n((s) => s ^ u);
  }, [n]);
  return { flags: t, addFlag: o, hasFlag: i, removeFlag: l, toggleFlag: a };
}
function dn(e) {
  let t = { called: !1 };
  return (...n) => {
    if (!t.called) return t.called = !0, e(...n);
  };
}
function Re(e, ...t) {
  e && t.length > 0 && e.classList.add(...t);
}
function $e(e, ...t) {
  e && t.length > 0 && e.classList.remove(...t);
}
function fn(e, t) {
  let n = Q();
  if (!e) return n.dispose;
  let { transitionDuration: r, transitionDelay: o } = getComputedStyle(e), [i, l] = [r, o].map((u) => {
    let [s = 0] = u.split(",").filter(Boolean).map((c) => c.includes("ms") ? parseFloat(c) : parseFloat(c) * 1e3).sort((c, v) => v - c);
    return s;
  }), a = i + l;
  if (a !== 0) {
    n.group((s) => {
      s.setTimeout(() => {
        t(), s.dispose();
      }, a), s.addEventListener(e, "transitionrun", (c) => {
        c.target === c.currentTarget && s.dispose();
      });
    });
    let u = n.addEventListener(e, "transitionend", (s) => {
      s.target === s.currentTarget && (t(), u());
    });
  } else t();
  return n.add(() => t()), n.dispose;
}
function pn(e, t, n, r) {
  let o = n ? "enter" : "leave", i = Q(), l = r !== void 0 ? dn(r) : () => {
  };
  o === "enter" && (e.removeAttribute("hidden"), e.style.display = "");
  let a = R(o, { enter: () => t.enter, leave: () => t.leave }), u = R(o, { enter: () => t.enterTo, leave: () => t.leaveTo }), s = R(o, { enter: () => t.enterFrom, leave: () => t.leaveFrom });
  return $e(e, ...t.base, ...t.enter, ...t.enterTo, ...t.enterFrom, ...t.leave, ...t.leaveFrom, ...t.leaveTo, ...t.entered), Re(e, ...t.base, ...a, ...s), i.nextFrame(() => {
    $e(e, ...t.base, ...a, ...s), Re(e, ...t.base, ...a, ...u), fn(e, () => ($e(e, ...t.base, ...a), Re(e, ...t.base, ...t.entered), l()));
  }), i.dispose;
}
function vn({ immediate: e, container: t, direction: n, classes: r, onStart: o, onStop: i }) {
  let l = ke(), a = ee(), u = j(n);
  D(() => {
    e && (u.current = "enter");
  }, [e]), D(() => {
    let s = Q();
    a.add(s.dispose);
    let c = t.current;
    if (c && u.current !== "idle" && l.current) return s.dispose(), o.current(u.current), s.add(pn(c, r.current, u.current === "enter", () => {
      s.dispose(), i.current(u.current);
    })), s.dispose;
  }, [n]);
}
function B(e = "") {
  return e.split(/\s+/).filter((t) => t.length > 1);
}
let he = ae(null);
he.displayName = "TransitionContext";
var mn = ((e) => (e.Visible = "visible", e.Hidden = "hidden", e))(mn || {});
function bn() {
  let e = te(he);
  if (e === null) throw new Error("A <Transition.Child /> is used but it is missing a parent <Transition /> or <Transition.Root />.");
  return e;
}
function hn() {
  let e = te(ge);
  if (e === null) throw new Error("A <Transition.Child /> is used but it is missing a parent <Transition /> or <Transition.Root />.");
  return e;
}
let ge = ae(null);
ge.displayName = "NestingContext";
function xe(e) {
  return "children" in e ? xe(e.children) : e.current.filter(({ el: t }) => t.current !== null).filter(({ state: t }) => t === "visible").length > 0;
}
function nt(e, t) {
  let n = j(e), r = y([]), o = ke(), i = ee(), l = h((g, p = q.Hidden) => {
    let m = r.current.findIndex(({ el: f }) => f === g);
    m !== -1 && (R(p, { [q.Unmount]() {
      r.current.splice(m, 1);
    }, [q.Hidden]() {
      r.current[m].state = "hidden";
    } }), i.microTask(() => {
      var f;
      !xe(r) && o.current && ((f = n.current) == null || f.call(n));
    }));
  }), a = h((g) => {
    let p = r.current.find(({ el: m }) => m === g);
    return p ? p.state !== "visible" && (p.state = "visible") : r.current.push({ el: g, state: "visible" }), () => l(g, q.Unmount);
  }), u = y([]), s = y(Promise.resolve()), c = y({ enter: [], leave: [], idle: [] }), v = h((g, p, m) => {
    u.current.splice(0), t && (t.chains.current[p] = t.chains.current[p].filter(([f]) => f !== g)), t == null || t.chains.current[p].push([g, new Promise((f) => {
      u.current.push(f);
    })]), t == null || t.chains.current[p].push([g, new Promise((f) => {
      Promise.all(c.current[p].map(([d, O]) => O)).then(() => f());
    })]), p === "enter" ? s.current = s.current.then(() => t == null ? void 0 : t.wait.current).then(() => m(p)) : m(p);
  }), x = h((g, p, m) => {
    Promise.all(c.current[p].splice(0).map(([f, d]) => d)).then(() => {
      var f;
      (f = u.current.shift()) == null || f();
    }).then(() => m(p));
  });
  return M(() => ({ children: r, register: a, unregister: l, onStart: v, onStop: x, wait: s, chains: c }), [a, l, r, v, x, c, s]);
}
function gn() {
}
let xn = ["beforeEnter", "afterEnter", "beforeLeave", "afterLeave"];
function We(e) {
  var t;
  let n = {};
  for (let r of xn) n[r] = (t = e[r]) != null ? t : gn;
  return n;
}
function yn(e) {
  let t = y(We(e));
  return N(() => {
    t.current = We(e);
  }, [e]), t;
}
let On = "div", rt = me.RenderStrategy;
function wn(e, t) {
  var n, r;
  let { beforeEnter: o, afterEnter: i, beforeLeave: l, afterLeave: a, enter: u, enterFrom: s, enterTo: c, entered: v, leave: x, leaveFrom: g, leaveTo: p, ...m } = e, f = y(null), d = z(f, t), O = (n = m.unmount) == null || n ? q.Unmount : q.Hidden, { show: L, appear: A, initial: G } = bn(), [S, C] = V(L ? "visible" : "hidden"), E = hn(), { register: _, unregister: Y } = E;
  N(() => _(f), [_, f]), N(() => {
    if (O === q.Hidden && f.current) {
      if (L && S !== "visible") {
        C("visible");
        return;
      }
      return R(S, { hidden: () => Y(f), visible: () => _(f) });
    }
  }, [S, f, _, Y, L, O]);
  let ne = j({ base: B(m.className), enter: B(u), enterFrom: B(s), enterTo: B(c), entered: B(v), leave: B(x), leaveFrom: B(g), leaveTo: B(p) }), X = yn({ beforeEnter: o, afterEnter: i, beforeLeave: l, afterLeave: a }), re = Ne();
  N(() => {
    if (re && S === "visible" && f.current === null) throw new Error("Did you forget to passthrough the `ref` to the actual DOM node?");
  }, [f, S, re]);
  let ye = G && !A, ce = A && L && G, Oe = !re || ye ? "idle" : L ? "enter" : "leave", H = cn(0), we = h((b) => R(b, { enter: () => {
    H.addFlag(I.Opening), X.current.beforeEnter();
  }, leave: () => {
    H.addFlag(I.Closing), X.current.beforeLeave();
  }, idle: () => {
  } })), Se = h((b) => R(b, { enter: () => {
    H.removeFlag(I.Opening), X.current.afterEnter();
  }, leave: () => {
    H.removeFlag(I.Closing), X.current.afterLeave();
  }, idle: () => {
  } })), J = nt(() => {
    C("hidden"), Y(f);
  }, E), oe = y(!1);
  vn({ immediate: ce, container: f, classes: ne, direction: Oe, onStart: j((b) => {
    oe.current = !0, J.onStart(f, b, we);
  }), onStop: j((b) => {
    oe.current = !1, J.onStop(f, b, Se), b === "leave" && !xe(J) && (C("hidden"), Y(f));
  }) });
  let k = m, Ee = { ref: d };
  return ce ? k = { ...k, className: ve(m.className, ...ne.current.enter, ...ne.current.enterFrom) } : oe.current && (k.className = ve(m.className, (r = f.current) == null ? void 0 : r.className), k.className === "" && delete k.className), P.createElement(ge.Provider, { value: J }, P.createElement(Je, { value: R(S, { visible: I.Open, hidden: I.Closed }) | H.flags }, K({ ourProps: Ee, theirProps: k, defaultTag: On, features: rt, visible: S === "visible", name: "Transition.Child" })));
}
function Sn(e, t) {
  let { show: n, appear: r = !1, unmount: o = !0, ...i } = e, l = y(null), a = z(l, t);
  Ne();
  let u = Ce();
  if (n === void 0 && u !== null && (n = (u & I.Open) === I.Open), ![!0, !1].includes(n)) throw new Error("A <Transition /> is used but it is missing a `show={true | false}` prop.");
  let [s, c] = V(n ? "visible" : "hidden"), v = nt(() => {
    c("hidden");
  }), [x, g] = V(!0), p = y([n]);
  D(() => {
    x !== !1 && p.current[p.current.length - 1] !== n && (p.current.push(n), g(!1));
  }, [p, n]);
  let m = M(() => ({ show: n, appear: r, initial: x }), [n, r, x]);
  N(() => {
    if (n) c("visible");
    else if (!xe(v)) c("hidden");
    else {
      let L = l.current;
      if (!L) return;
      let A = L.getBoundingClientRect();
      A.x === 0 && A.y === 0 && A.width === 0 && A.height === 0 && c("hidden");
    }
  }, [n, v]);
  let f = { unmount: o }, d = h(() => {
    var L;
    x && g(!1), (L = e.beforeEnter) == null || L.call(e);
  }), O = h(() => {
    var L;
    x && g(!1), (L = e.beforeLeave) == null || L.call(e);
  });
  return P.createElement(ge.Provider, { value: v }, P.createElement(he.Provider, { value: m }, K({ ourProps: { ...f, as: Z, children: P.createElement(ot, { ref: a, ...f, ...i, beforeEnter: d, beforeLeave: O }) }, theirProps: {}, defaultTag: Z, features: rt, visible: s === "visible", name: "Transition" })));
}
function En(e, t) {
  let n = te(he) !== null, r = Ce() !== null;
  return P.createElement(P.Fragment, null, !n && r ? P.createElement(Ie, { ref: t, ...e }) : P.createElement(ot, { ref: t, ...e }));
}
let Ie = U(Sn), ot = U(wn), Ln = U(En), $n = Object.assign(Ie, { Child: Ln, Root: Ie });
export {
  Rn as I,
  $n as q
};
