// @deriv-com/quill-ui/dist/index-C1BOoRZp.js
import { j as t } from "./jsx-runtime-DmTjM30f.js";
import * as a from "react";
import d, { forwardRef as w } from "react";
import { c as p } from "./clsx-OuTLNxxd.js";
import "./components/Typography/base.js";
import "./components/Typography/heading/index.js";
import { Text as q } from "./components/Typography/text/index.js";
import { s as _ } from "./index-CQbz1p6_.js";
import './assets/index16.css';const E = ({ iconSize: i = "md", title: e, titleId: l, ...r }, o) => a.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 32 32", ..._[i], role: "img", ref: o, "aria-labelledby": l, ...r },
  e ? a.createElement("title", { id: l }, e) : null,
  a.createElement("path", { d: "M22.25 9H9.75c-.703 0-1.25.586-1.25 1.25v12.5c0 .703.547 1.25 1.25 1.25h12.5c.664 0 1.25-.547 1.25-1.25v-12.5c0-.664-.586-1.25-1.25-1.25M9.75 7.75h12.5c1.367 0 2.5 1.133 2.5 2.5v12.5c0 1.406-1.133 2.5-2.5 2.5H9.75a2.47 2.47 0 0 1-2.5-2.5v-12.5c0-1.367 1.094-2.5 2.5-2.5" })
), R = w(E), j = ({ iconSize: i = "md", title: e, titleId: l, ...r }, o) => a.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 32 32", ..._[i], role: "img", ref: o, "aria-labelledby": l, ...r },
  e ? a.createElement("title", { id: l }, e) : null,
  a.createElement("path", { d: "M9.75 7.75h12.5c1.367 0 2.5 1.133 2.5 2.5v12.5c0 1.406-1.133 2.5-2.5 2.5H9.75a2.47 2.47 0 0 1-2.5-2.5v-12.5c0-1.367 1.094-2.5 2.5-2.5m10.664 6.914h-.039c.39-.351.39-.937 0-1.328a.92.92 0 0 0-1.29 0L14.75 17.71l-1.836-1.836c-.39-.39-.976-.39-1.328 0a.856.856 0 0 0 0 1.29l2.5 2.5c.351.39.937.39 1.328 0z" })
), S = w(j), M = d.forwardRef(
  ({
    as: i = "li",
    leftIcon: e,
    rightIcon: l,
    size: r = "md",
    centered: o,
    textAlignment: f = "left",
    label: u,
    disabled: m = !1,
    selected: s = !1,
    className: v,
    checkbox: c = !1,
    labelColor: x,
    ...g
  }, h) => {
    const n = r === "lg" ? "md" : r;
    return d.createElement(
      i,
      {
        ref: h,
        className: p(
          "quill-dropdown-item",
          `quill-dropdown-item__size--${n}`,
          `quill-dropdown-item__align-${f}`,
          c ? `quill-dropdown-item-checkbox__selected--${s}__disabled--${m}` : `quill-dropdown-item__selected--${s}__disabled--${m}`,
          v
        ),
        ...g
      },
      (c || o || e) && /* @__PURE__ */ t.jsx("div", { className: "quill-dropdown-item__icon", children: c ? s ? /* @__PURE__ */ t.jsx(S, { iconSize: "sm" }) : /* @__PURE__ */ t.jsx(R, { iconSize: "sm" }) : e }),
      /* @__PURE__ */ t.jsx(
        q,
        {
          size: n,
          className: p(
            "quill-dropdown-item-label",
            m && "quill-typography__color--disabled"
          ),
          color: x,
          centered: o,
          children: u
        }
      ),
      (o || l) && /* @__PURE__ */ t.jsx("div", { className: "quill-dropdown-item__icon", children: l })
    );
  }
);
export {
  M as D,
  S as F
};
