// @deriv-com/quill-ui/dist/index-C7RXo2Oj.js
import { j as l } from "./jsx-runtime-DmTjM30f.js";
import * as t from "react";
import { forwardRef as f, useState as y, useRef as C, useEffect as F } from "react";
import { c as s } from "./clsx-OuTLNxxd.js";
import "./components/Typography/base.js";
import "./components/Typography/heading/index.js";
import { Text as $ } from "./components/Typography/text/index.js";
import { KEY as E } from "./utils/common-utils.js";
import { F as M } from "./LabelPairedCircleInfoSmRegularIcon-CnUvtxa5.js";
import { s as g } from "./index-CQbz1p6_.js";
import './assets/index20.css';const I = ({ title: n, titleId: e, ...o }, r) => t.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", width: 16, height: 24, viewBox: "0 0 16 24", role: "img", ref: r, "aria-labelledby": e, ...o },
  n ? t.createElement("title", { id: e }, n) : null,
  t.createElement(
    "g",
    null,
    t.createElement("path", { d: "M8 5a6.98 6.98 0 0 0-6.062 3.5c-1.282 2.188-1.282 4.844 0 7C3.188 17.688 5.467 19 8 19c2.5 0 4.781-1.312 6.031-3.5 1.281-2.156 1.281-4.812 0-7C12.781 6.344 10.5 5 8 5m0 15c-2.875 0-5.5-1.5-6.937-4-1.438-2.469-1.438-5.5 0-8A8.02 8.02 0 0 1 8 4c2.844 0 5.469 1.531 6.906 4 1.438 2.5 1.438 5.531 0 8A7.94 7.94 0 0 1 8 20m-1.5-5h1v-3h-.75a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5H8c.25 0 .5.25.5.5V15h1c.25 0 .5.25.5.5 0 .281-.25.5-.5.5h-3a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5M8 9.75A.72.72 0 0 1 7.25 9 .74.74 0 0 1 8 8.25a.76.76 0 0 1 .75.75.74.74 0 0 1-.75.75" })
  ),
  t.createElement(
    "defs",
    null,
    t.createElement(
      "clipPath",
      { id: "37987f4debe5d1de11235bff8c6db891__a" },
      t.createElement("path", { d: "M0 0h16v24H0z" })
    )
  )
), N = f(I), S = ({ iconSize: n = "md", title: e, titleId: o, ...r }, c) => t.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 32 32", ...g[n], role: "img", ref: c, "aria-labelledby": o, ...r },
  e ? t.createElement("title", { id: o }, e) : null,
  t.createElement("path", { d: "M16 26.5c-3.594 0-6.875-1.875-8.672-5-1.797-3.086-1.797-6.875 0-10 1.797-3.086 5.078-5 8.672-5 3.555 0 6.836 1.914 8.633 5 1.797 3.125 1.797 6.914 0 10a9.93 9.93 0 0 1-8.633 5m0-13.75c-1.367 0-2.578.742-3.281 1.875-.664 1.172-.664 2.617 0 3.75.703 1.172 1.914 1.875 3.281 1.875 1.328 0 2.54-.703 3.242-1.875.664-1.133.664-2.578 0-3.75-.703-1.133-1.914-1.875-3.242-1.875" })
), A = f(S), B = ({ iconSize: n = "md", title: e, titleId: o, ...r }, c) => t.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 32 32", ...g[n], role: "img", ref: c, "aria-labelledby": o, ...r },
  e ? t.createElement("title", { id: o }, e) : null,
  t.createElement("path", { d: "M16 7.75a8.73 8.73 0 0 0-7.578 4.375c-1.602 2.734-1.602 6.055 0 8.75C9.984 23.609 12.836 25.25 16 25.25a8.63 8.63 0 0 0 7.54-4.375c1.6-2.695 1.6-6.016 0-8.75C21.976 9.43 19.124 7.75 16 7.75m0 18.75c-3.594 0-6.875-1.875-8.672-5-1.797-3.086-1.797-6.875 0-10 1.797-3.086 5.078-5 8.672-5 3.555 0 6.836 1.914 8.633 5 1.797 3.125 1.797 6.914 0 10a9.93 9.93 0 0 1-8.633 5" })
), D = f(B), J = ({
  children: n,
  className: e,
  classNameInfo: o,
  classNameLabel: r,
  defaultChecked: c,
  disabled: a = !1,
  hasInfo: v,
  id: p,
  onChange: d,
  size: m = "md",
  value: u,
  radioButtonPosition: R = "left",
  ...q
}) => {
  const [_, j] = y(c), h = C(null);
  F(() => {
    j(c);
  }, [c]);
  const w = () => {
    a || b();
  }, x = (i) => {
    !a && (i.key === E.ENTER || i.key === E.SPACE) && b();
  }, b = () => {
    const i = h.current;
    i && (i.checked = !i.checked, d == null || d({
      target: i
    }));
  }, k = () => _ ? /* @__PURE__ */ l.jsx(
    A,
    {
      "data-testid": `dt_checked_icon_${u}_${a}`,
      fill: a ? "var(--component-textIcon-normal-disabled)" : "var(--component-textIcon-normal-prominent)",
      iconSize: m
    }
  ) : /* @__PURE__ */ l.jsx(
    D,
    {
      "data-testid": `dt_unchecked_icon_${u}_${a}`,
      fill: a ? "var(--component-textIcon-normal-disabled)" : "var(--component-textIcon-normal-subtle)",
      iconSize: m,
      tabIndex: 0
    }
  );
  return /* @__PURE__ */ l.jsxs(
    "label",
    {
      htmlFor: p,
      className: s("quill-radio-button", e, {
        "quill-radio-group__item--active": _
      }),
      "data-testid": "dt_quill_radio_button",
      children: [
        /* @__PURE__ */ l.jsx(
          "input",
          {
            className: "quill-radio-button__input",
            type: "radio",
            defaultChecked: _,
            disabled: a,
            ref: h,
            value: u,
            ...q
          }
        ),
        /* @__PURE__ */ l.jsx(
          "span",
          {
            className: s(
              "quill-radio-button__icon",
              {
                "quill-radio-button__icon--disabled": a
              },
              `quill-radio-button__icon--${R}`
            ),
            onClick: w,
            onKeyDown: x,
            id: p,
            children: k()
          }
        ),
        /* @__PURE__ */ l.jsx(
          $,
          {
            size: m,
            as: "span",
            className: s(
              "quill-radio-button__label",
              {
                "quill-radio-button__label--disabled": a
              },
              r
            ),
            onClick: w,
            onKeyDown: x,
            children: n
          }
        ),
        v && (m === "sm" ? /* @__PURE__ */ l.jsx(
          M,
          {
            className: s(
              "quill-radio-button__info",
              o
            ),
            "data-testid": "dt_quill_radio_button_info"
          }
        ) : /* @__PURE__ */ l.jsx(
          N,
          {
            className: s(
              "quill-radio-button__info",
              o
            ),
            "data-testid": "dt_quill_radio_button_info"
          }
        ))
      ]
    }
  );
};
export {
  N as F,
  J as R
};
