// @deriv-com/quill-ui/dist/main.js
import { A as e, e as t, C as p, a as i, D as a, G as m, I as n, c as x, b as f, M as s, T as c, d, f as l } from "./snackbar-controller-ChcV5SXl.js";
import { D as u } from "./index-C1BOoRZp.js";
import { Divider as h } from "./components/Atom/dropdown/divider/index.js";
import { DropdownTitle as P } from "./components/Atom/dropdown/title/index.js";
import { ItemContainer as g } from "./components/Atom/dropdown/item-container/index.js";
import { DatePicker as D } from "./components/Atom/Calendar/index.js";
import { PasswordStrengthValidation as b } from "./components/Atom/PasswordValidation/index.js";
import { WheelPicker as N } from "./components/Atom/WheelPicker/base/index.js";
import { WheelPickerContainer as W } from "./components/Atom/WheelPicker/wheel-picker/index.js";
import { TimeWheelPickerContainer as G } from "./components/Atom/WheelPicker/time-wheel-picker/index.js";
import { Badge as A } from "./components/Badge/index.js";
import { Breadcrumbs as M } from "./components/Breadcrumbs/index.js";
import { Button as L, ButtonSize as U, loaderIcons as V } from "./components/Button/base/index.js";
import { SocialButton as j, SocialButtonSize as q, socialButtonIconSize as y } from "./components/Button/social/index.js";
import { IconButton as J } from "./components/Button/icon-button/index.js";
import { Chip as Q } from "./components/Chip/index.js";
import { PasswordField as Y } from "./components/Input/password-field/index.js";
import { SearchField as _ } from "./components/Input/search-field/index.js";
import { TextField as oo } from "./components/Input/text-field/index.js";
import { TextFieldAddon as eo } from "./components/Input/text-field-addon/index.js";
import { TextArea as po } from "./components/Input/textarea/index.js";
import { Link as ao } from "./components/Link/index.js";
import { Skeleton as no } from "./components/Loader/skeleton/index.js";
import { Spinner as fo } from "./components/Loader/spinner/index.js";
import { Navigation as co } from "./components/Navigation/index.js";
import { NotificationBanner as So } from "./components/Notification/banner/index.js";
import { NotificationBanners as To } from "./components/Notification/banners/index.js";
import { NotificationItem as ko } from "./components/Notification/item/index.js";
import { NotificationItemsList as Bo } from "./components/Notification/items-list/index.js";
import { Pagination as Co } from "./components/Pagination/index.js";
import { SectionMessage as Io } from "./components/SectionMessage/index.js";
import { SegmentedControlSingleChoice as wo } from "./components/SegmentedControl/single-choice/index.js";
import { CheckboxGroup as vo } from "./components/SelectionControl/checkbox/checkbox-group/index.js";
import { R as Fo } from "./index-C7RXo2Oj.js";
import { RadioGroup as zo } from "./components/SelectionControl/radio/radio-group/index.js";
import { ToggleSwitch as Ro } from "./components/SelectionControl/toggle-switch/index.js";
import { HorizontalStepper as Ho } from "./components/Stepper/horizontal-stepper/index.js";
import { VerticalStepper as Uo } from "./components/Stepper/vertical-stepper/index.js";
import { Snackbar as Oo } from "./components/Snackbar/snackbar.js";
import { SnackbarController as qo } from "./components/Snackbar/snackbar-controller.js";
import { Tab as Eo } from "./components/Tab/index.js";
import { Tag as Ko } from "./components/Tag/index.js";
import { CaptionText as Xo } from "./components/Typography/caption/index.js";
import { CodeText as Zo } from "./components/Typography/code/index.js";
import { Heading as $o } from "./components/Typography/heading/index.js";
import { Text as rr } from "./components/Typography/text/index.js";
import { useNotifications as tr } from "./hooks/useNotifications/index.js";
import { UseTheme as ir } from "./hooks/useTheme/index.js";
import { DOTS as mr, UsePaginationRange as nr } from "./hooks/usePaginationRange.js";
import { useSnackbar as fr } from "./hooks/useSnackbar.js";
import { useDropdown as cr } from "./hooks/useDropdown/index.js";
import { ThemeProvider as lr } from "./providers/theme/themeProvider.js";
import { SnackbarProvider as ur } from "./providers/snackbar/snackbarProvider.js";
import { NotificationsProvider as hr } from "./providers/notification/notificationsProvider.js";
import { BreakpointProvider as Pr } from "./providers/breakpoint/breakpointProvider.js";
export {
  e as ActionSheet,
  A as Badge,
  M as Breadcrumbs,
  Pr as BreakpointProvider,
  L as Button,
  U as ButtonSize,
  Xo as CaptionText,
  t as Checkbox,
  vo as CheckboxGroup,
  Q as Chip,
  Zo as CodeText,
  p as CustomDropdown,
  mr as DOTS,
  D as DatePicker,
  i as DatePickerDropdown,
  h as Divider,
  a as DropdownButton,
  u as DropdownItem,
  P as DropdownTitle,
  m as GenericWheelPicker,
  $o as Heading,
  Ho as HorizontalStepper,
  J as IconButton,
  n as InputDropdown,
  x as InputGroupButton,
  f as InputPhoneNumber,
  g as ItemContainer,
  ao as Link,
  s as Modal,
  co as Navigation,
  So as NotificationBanner,
  To as NotificationBanners,
  ko as NotificationItem,
  Bo as NotificationItemsList,
  hr as NotificationsProvider,
  Co as Pagination,
  Y as PasswordField,
  b as PasswordStrengthValidation,
  Fo as RadioButton,
  zo as RadioGroup,
  _ as SearchField,
  Io as SectionMessage,
  wo as SegmentedControlSingleChoice,
  no as Skeleton,
  Oo as Snackbar,
  qo as SnackbarController,
  ur as SnackbarProvider,
  j as SocialButton,
  q as SocialButtonSize,
  fo as Spinner,
  Eo as Tab,
  Ko as Tag,
  rr as Text,
  po as TextArea,
  oo as TextField,
  eo as TextFieldAddon,
  c as TextFieldWithSteppers,
  lr as ThemeProvider,
  d as TimeWheelPicker,
  G as TimeWheelPickerContainer,
  Ro as ToggleSwitch,
  l as Tooltip,
  nr as UsePaginationRange,
  ir as UseTheme,
  Uo as VerticalStepper,
  N as WheelPicker,
  W as WheelPickerContainer,
  V as loaderIcons,
  y as socialButtonIconSize,
  cr as useDropdown,
  tr as useNotifications,
  fr as useSnackbar
};
