// @deriv-com/quill-ui/dist/snackbar-controller-ChcV5SXl.js
import { j as o } from "./jsx-runtime-DmTjM30f.js";
import * as E from "react";
import we, { forwardRef as Y, useContext as Re, useRef as re, useEffect as X, useState as J } from "react";
import { c as L } from "./clsx-OuTLNxxd.js";
import "./components/Typography/base.js";
import { Heading as ut } from "./components/Typography/heading/index.js";
import { Text as We } from "./components/Typography/text/index.js";
import { reactNodeToString as ye, getFormatValue as Ye, KEY as ke } from "./utils/common-utils.js";
import { c as le, g as pt } from "./_commonjsHelpers-BkfeUUK-.js";
import rt from "react-dom";
import { CaptionText as Oe } from "./components/Typography/caption/index.js";
import { Link as mt } from "./components/Link/index.js";
import { Button as ue } from "./components/Button/base/index.js";
import xe from "./components/Input/base/index.js";
import { useDropdown as pe } from "./hooks/useDropdown/index.js";
import { DropdownProvider as Pe } from "./providers/dropdown/dropdownProvider.js";
import { useBreakpoints as Ee } from "./hooks/useBreakpoints/index.js";
import ft, { ActionSheetContext as qe } from "./components/ActionSheet/root/index.js";
import "./components/Button/social/index.js";
import { IconButton as Ce } from "./components/Button/icon-button/index.js";
import ht from "./components/ActionSheet/content/index.js";
import vt from "./components/ActionSheet/handle-bar/index.js";
import gt from "./components/ActionSheet/portal/index.js";
import { DropdownContext as wt } from "./providers/dropdown/dropdownContext.js";
import { D as nt, F as xt } from "./index-C1BOoRZp.js";
import './assets/stepper.css';import './assets/toggle-switch.css';import './assets/segmented-control.css';import './assets/section-message.css';import './assets/pagination-base.css';import './assets/spinner.css';import './assets/breadcrumbs.css';import './assets/date-picker.css';import './assets/wheel-picker-base.css';import './assets/password-validation.css';import './assets/divider.css';import './assets/snackbar-controller.css';/* empty css                 */
import "./components/Atom/dropdown/title/index.js";
import { ItemContainer as bt } from "./components/Atom/dropdown/item-container/index.js";
import { DatePicker as Ge } from "./components/Atom/Calendar/index.js";
/* empty css                             */
/* empty css                           */
import { d as ge } from "./dayjs.min-DrYbymKu.js";
import "./lodash-rsqVyylH.js";
import { s as Me } from "./index-CQbz1p6_.js";
/* empty css                     */
import "./components/Badge/index.js";
/* empty css                     */
import "./providers/breakpoint/breakpointContext.js";
import "./components/Chip/index.js";
import "./components/Input/password-field/index.js";
import { SearchField as Ke } from "./components/Input/search-field/index.js";
import "./components/Input/text-field/index.js";
import "./components/Input/text-field-addon/index.js";
import "./components/Input/textarea/index.js";
import { WheelPickerContainer as it } from "./components/Atom/WheelPicker/wheel-picker/index.js";
import { TimeWheelPickerContainer as _t } from "./components/Atom/WheelPicker/time-wheel-picker/index.js";
import "./components/Loader/skeleton/index.js";
/* empty css                 */
import { ModalHeader as jt } from "./components/Modal/base/modal-header.js";
import { ModalBody as yt } from "./components/Modal/base/modal-body.js";
import "./components/Navigation/index.js";
import "./components/Notification/notification-icon.js";
/* empty css                         */
/* empty css                         */
import "./components/SegmentedControl/segment.js";
/* empty css                           */
import { F as Rt } from "./LabelPairedCircleInfoSmRegularIcon-CnUvtxa5.js";
import { F as Ct } from "./index-C7RXo2Oj.js";
import "./components/SelectionControl/radio/radio-group/index.js";
/* empty css                       */
/* empty css                 */
import "./providers/snackbar/snackbarContext.js";
import "./components/Tab/index.js";
import "./components/Tag/index.js";
import { FlagsMap as ze } from "./components/Input/input-phone-number/country-flags.js";
import { a as Pt } from "./index-EY1gwl5O.js";
import Et from "./components/Button/dropdown/dropdown-content.js";
import Mt from "./components/Button/dropdown/dropdown-head.js";
const Nt = ({ title: e, titleId: t, ...n }, i) => E.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", width: 11, height: 22, viewBox: "0 0 11 22", role: "img", ref: i, "aria-labelledby": t, ...n },
  e ? E.createElement("title", { id: t }, e) : null,
  E.createElement(
    "g",
    null,
    E.createElement("path", { d: "m9.684 7.496-3.282 3.281 3.254 3.254a.6.6 0 0 1 0 .903.6.6 0 0 1-.902 0L5.473 11.68l-3.254 3.254a.6.6 0 0 1-.903 0c-.273-.246-.273-.657 0-.93L4.57 10.75 1.316 7.496c-.273-.246-.273-.656 0-.93.246-.246.657-.246.93 0L5.5 9.848l3.254-3.254c.246-.274.656-.274.93 0a.644.644 0 0 1 0 .902" })
  ),
  E.createElement(
    "defs",
    null,
    E.createElement(
      "clipPath",
      { id: "7621e938296d9c3b45efe83c29f8948a__a" },
      E.createElement("path", { d: "M0 0h11v22H0z" })
    )
  )
), kt = Y(Nt), Ot = ({ title: e, titleId: t, ...n }, i) => E.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", width: 13, height: 22, viewBox: "0 0 13 22", role: "img", ref: i, "aria-labelledby": t, ...n },
  e ? E.createElement("title", { id: t }, e) : null,
  E.createElement(
    "g",
    null,
    E.createElement("path", { d: "M12.188 10.75a.45.45 0 0 1-.438.438H1.25a.43.43 0 0 1-.437-.438.45.45 0 0 1 .437-.437h10.5a.47.47 0 0 1 .438.437" })
  ),
  E.createElement(
    "defs",
    null,
    E.createElement(
      "clipPath",
      { id: "36b698bcc19c519281084bd39aa59f53__a" },
      E.createElement("path", { d: "M0 0h13v22H0z" })
    )
  )
), qt = Y(Ot), $t = ({ title: e, titleId: t, ...n }, i) => E.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", width: 13, height: 22, viewBox: "0 0 13 22", role: "img", ref: i, "aria-labelledby": t, ...n },
  e ? E.createElement("title", { id: t }, e) : null,
  E.createElement(
    "g",
    null,
    E.createElement("path", { d: "M6.938 5.5v4.813h4.812a.47.47 0 0 1 .438.437.45.45 0 0 1-.438.438H6.938V16a.45.45 0 0 1-.438.438.43.43 0 0 1-.437-.438v-4.812H1.25a.43.43 0 0 1-.437-.438.45.45 0 0 1 .437-.437h4.813V5.5a.45.45 0 0 1 .437-.437.47.47 0 0 1 .438.437" })
  ),
  E.createElement(
    "defs",
    null,
    E.createElement(
      "clipPath",
      { id: "5c0d2f6793ade08ad86ad5278d778038__a" },
      E.createElement("path", { d: "M0 0h13v22H0z" })
    )
  )
), St = Y($t), Lt = ({ title: e, titleId: t, ...n }, i) => E.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", width: 12, height: 24, viewBox: "0 0 12 24", role: "img", ref: i, "aria-labelledby": t, ...n },
  e ? E.createElement("title", { id: t }, e) : null,
  E.createElement(
    "g",
    null,
    E.createElement("path", { d: "m10.781 8.281-3.75 3.75 3.719 3.719a.684.684 0 0 1 0 1.031.684.684 0 0 1-1.031 0l-3.75-3.718L2.25 16.78a.684.684 0 0 1-1.031 0c-.313-.281-.313-.75 0-1.062L4.937 12 1.22 8.281c-.313-.281-.313-.75 0-1.062.281-.282.75-.282 1.062 0L6 10.969 9.719 7.25c.281-.312.75-.312 1.062 0a.736.736 0 0 1 0 1.031" })
  ),
  E.createElement(
    "defs",
    null,
    E.createElement(
      "clipPath",
      { id: "3e01371d11741e6b95171840b44da950__a" },
      E.createElement("path", { d: "M0 0h12v24H0z" })
    )
  )
), Tt = Y(Lt), It = ({ iconSize: e = "md", title: t, titleId: n, ...i }, l) => E.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 32 32", ...Me[e], role: "img", ref: l, "aria-labelledby": n, ...i },
  t ? E.createElement("title", { id: n }, t) : null,
  E.createElement("path", { d: "M11.625 6.5c.313 0 .625.313.625.625V9h7.5V7.125c0-.312.273-.625.625-.625.313 0 .625.313.625.625V9h1.25c1.367 0 2.5 1.133 2.5 2.5V24c0 1.406-1.133 2.5-2.5 2.5H9.75a2.47 2.47 0 0 1-2.5-2.5V11.5c0-1.367 1.094-2.5 2.5-2.5H11V7.125c0-.312.273-.625.625-.625M23.5 14h-15v10c0 .703.547 1.25 1.25 1.25h12.5c.664 0 1.25-.547 1.25-1.25zm-1.25-3.75H9.75c-.703 0-1.25.586-1.25 1.25v1.25h15V11.5c0-.664-.586-1.25-1.25-1.25" })
), Ft = Y(It), Dt = ({ iconSize: e = "md", title: t, titleId: n, ...i }, l) => E.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 32 32", ...Me[e], role: "img", ref: l, "aria-labelledby": n, ...i },
  t ? E.createElement("title", { id: n }, t) : null,
  E.createElement("path", { d: "M22.25 9.625H9.75a.64.64 0 0 0-.625.625v12.5c0 .352.273.625.625.625h12.5a.64.64 0 0 0 .625-.625v-12.5c0-.312-.312-.625-.625-.625M9.75 7.75h12.5c1.367 0 2.5 1.133 2.5 2.5v12.5c0 1.406-1.133 2.5-2.5 2.5H9.75a2.47 2.47 0 0 1-2.5-2.5v-12.5c0-1.367 1.094-2.5 2.5-2.5" })
), Ht = Y(Dt), Yt = ({ iconSize: e = "md", title: t, titleId: n, ...i }, l) => E.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 32 32", ...Me[e], role: "img", ref: l, "aria-labelledby": n, ...i },
  t ? E.createElement("title", { id: n }, t) : null,
  E.createElement("path", { d: "M9.75 7.75h12.5c1.367 0 2.5 1.133 2.5 2.5v12.5c0 1.406-1.133 2.5-2.5 2.5H9.75a2.47 2.47 0 0 1-2.5-2.5v-12.5c0-1.367 1.094-2.5 2.5-2.5m3.438 7.813a.925.925 0 0 0-.938.937c0 .547.39.938.938.938h5.624c.508 0 .938-.391.938-.938a.95.95 0 0 0-.937-.937z" })
), zt = Y(Yt), Bt = ({ iconSize: e = "md", title: t, titleId: n, ...i }, l) => E.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 32 32", ...Me[e], role: "img", ref: l, "aria-labelledby": n, ...i },
  t ? E.createElement("title", { id: n }, t) : null,
  E.createElement("path", { d: "m15.336 22.164-7.5-7.5c-.39-.351-.39-.937 0-1.328.351-.352.937-.352 1.328 0L16 20.21l6.836-6.836c.352-.39.937-.39 1.328 0a.92.92 0 0 1 0 1.29l-7.539 7.5a.856.856 0 0 1-1.29 0" })
), Vt = Y(Bt), Wt = ({ iconSize: e = "md", title: t, titleId: n, ...i }, l) => E.createElement(
  "svg",
  { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 32 32", ...Me[e], role: "img", ref: l, "aria-labelledby": n, ...i },
  t ? E.createElement("title", { id: n }, t) : null,
  E.createElement("path", { d: "M18.5 7.75v.625c0 .703-.586 1.25-1.25 1.25h-2.5c-.703 0-1.25-.547-1.25-1.25V7.75h-1.25C11.547 7.75 11 8.336 11 9v15c0 .703.547 1.25 1.25 1.25h7.5c.664 0 1.25-.547 1.25-1.25V9c0-.664-.586-1.25-1.25-1.25zm-1.25 0h-2.5v.625h2.5zM9.75 9c0-1.367 1.094-2.5 2.5-2.5h7.5c1.367 0 2.5 1.133 2.5 2.5v15c0 1.406-1.133 2.5-2.5 2.5h-7.5a2.47 2.47 0 0 1-2.5-2.5z" })
), At = Y(Wt), Xe = Y(
  ({
    children: e,
    className: t,
    isAutocomplete: n = !1,
    value: i,
    onClickDropdown: l,
    fullHeightOnOpen: f = !0,
    onChange: h,
    containerClassName: b,
    contentClassName: a,
    actionSheetFooter: r,
    label: s,
    headComponent: m,
    noActionSheet: d = !1,
    contentAlign: w = "left",
    noAutoClose: x = !1,
    disabled: g,
    actionSheetDropdown: c = !1,
    ...p
  }, u) => {
    if (!Re(wt))
      throw new Error(
        "Custom dropdown must be used within a DropdownProvider"
      );
    const _ = re(null), j = re(null), M = re(null), R = re(null), { isOpen: y, open: $, close: P, selectedValue: O, setSelectedValue: N } = pe(
      [j, M, R],
      x
    ), { isMobile: T } = Ee();
    X(() => {
      i && N(i);
    }, [i]);
    const C = (q) => {
      if (!g) {
        const F = _.current;
        if (F && (F == null || F.focus()), n && y) return;
        l == null || l(q), y ? P() : $();
      }
    }, k = (q) => {
      const { value: F } = q.target;
      N(!h && !F ? "" : F), h == null || h(q);
    };
    return /* @__PURE__ */ o.jsxs(
      "div",
      {
        ref: u,
        className: L(
          "quill-custom-dropdown__container",
          b
        ),
        children: [
          /* @__PURE__ */ o.jsx(
            "div",
            {
              ref: j,
              onClick: C,
              className: "quill-custom-dropdown__head",
              children: m || /* @__PURE__ */ o.jsx(
                xe,
                {
                  dropdown: !0,
                  ref: _,
                  isDropdownOpen: y,
                  readOnly: !n,
                  value: O,
                  className: L(
                    "quill-custom-dropdown__input",
                    `quill-custom-dropdown__input--hasValue--${!!O}`,
                    t
                  ),
                  onChange: k,
                  type: "select",
                  label: s,
                  ...p
                }
              )
            }
          ),
          !c && (!T || d) ? y && /* @__PURE__ */ o.jsx(
            "div",
            {
              className: L(
                "quill-custom-dropdown__content",
                `quill-custom-dropdown__content-align--${w}`,
                a
              ),
              ref: R,
              children: e
            }
          ) : /* @__PURE__ */ o.jsx(D.Root, { isOpen: y, onClose: () => P(), children: /* @__PURE__ */ o.jsxs(
            D.Portal,
            {
              shouldCloseOnDrag: !0,
              fullHeightOnOpen: f,
              ref: M,
              children: [
                s && /* @__PURE__ */ o.jsx(D.Header, { title: s }),
                /* @__PURE__ */ o.jsx(D.Content, { children: e }),
                /* @__PURE__ */ o.jsx(D.Footer, { ...r })
              ]
            }
          ) })
        ]
      }
    );
  }
), Ae = Y(
  ({ children: e, withProvider: t = !0, onOpen: n, onClose: i, ...l }, f) => t ? /* @__PURE__ */ o.jsx(Pe, { onOpen: n, onClose: i, children: /* @__PURE__ */ o.jsx(Xe, { ref: f, ...l, children: e }) }) : /* @__PURE__ */ o.jsx(Xe, { ref: f, ...l, children: e })
);
var at = { exports: {} };
(function(e, t) {
  (function(n, i) {
    e.exports = i();
  })(le, function() {
    var n = { LTS: "h:mm:ss A", LT: "h:mm A", L: "MM/DD/YYYY", LL: "MMMM D, YYYY", LLL: "MMMM D, YYYY h:mm A", LLLL: "dddd, MMMM D, YYYY h:mm A" }, i = /(\[[^[]*\])|([-_:/.,()\s]+)|(A|a|Q|YYYY|YY?|ww?|MM?M?M?|Do|DD?|hh?|HH?|mm?|ss?|S{1,3}|z|ZZ?)/g, l = /\d/, f = /\d\d/, h = /\d\d?/, b = /\d*[^-_:/,()\s\d]+/, a = {}, r = function(c) {
      return (c = +c) + (c > 68 ? 1900 : 2e3);
    }, s = function(c) {
      return function(p) {
        this[c] = +p;
      };
    }, m = [/[+-]\d\d:?(\d\d)?|Z/, function(c) {
      (this.zone || (this.zone = {})).offset = function(p) {
        if (!p || p === "Z") return 0;
        var u = p.match(/([+-]|\d\d)/g), v = 60 * u[1] + (+u[2] || 0);
        return v === 0 ? 0 : u[0] === "+" ? -v : v;
      }(c);
    }], d = function(c) {
      var p = a[c];
      return p && (p.indexOf ? p : p.s.concat(p.f));
    }, w = function(c, p) {
      var u, v = a.meridiem;
      if (v) {
        for (var _ = 1; _ <= 24; _ += 1) if (c.indexOf(v(_, 0, p)) > -1) {
          u = _ > 12;
          break;
        }
      } else u = c === (p ? "pm" : "PM");
      return u;
    }, x = { A: [b, function(c) {
      this.afternoon = w(c, !1);
    }], a: [b, function(c) {
      this.afternoon = w(c, !0);
    }], Q: [l, function(c) {
      this.month = 3 * (c - 1) + 1;
    }], S: [l, function(c) {
      this.milliseconds = 100 * +c;
    }], SS: [f, function(c) {
      this.milliseconds = 10 * +c;
    }], SSS: [/\d{3}/, function(c) {
      this.milliseconds = +c;
    }], s: [h, s("seconds")], ss: [h, s("seconds")], m: [h, s("minutes")], mm: [h, s("minutes")], H: [h, s("hours")], h: [h, s("hours")], HH: [h, s("hours")], hh: [h, s("hours")], D: [h, s("day")], DD: [f, s("day")], Do: [b, function(c) {
      var p = a.ordinal, u = c.match(/\d+/);
      if (this.day = u[0], p) for (var v = 1; v <= 31; v += 1) p(v).replace(/\[|\]/g, "") === c && (this.day = v);
    }], w: [h, s("week")], ww: [f, s("week")], M: [h, s("month")], MM: [f, s("month")], MMM: [b, function(c) {
      var p = d("months"), u = (d("monthsShort") || p.map(function(v) {
        return v.slice(0, 3);
      })).indexOf(c) + 1;
      if (u < 1) throw new Error();
      this.month = u % 12 || u;
    }], MMMM: [b, function(c) {
      var p = d("months").indexOf(c) + 1;
      if (p < 1) throw new Error();
      this.month = p % 12 || p;
    }], Y: [/[+-]?\d+/, s("year")], YY: [f, function(c) {
      this.year = r(c);
    }], YYYY: [/\d{4}/, s("year")], Z: m, ZZ: m };
    function g(c) {
      var p, u;
      p = c, u = a && a.formats;
      for (var v = (c = p.replace(/(\[[^\]]+])|(LTS?|l{1,4}|L{1,4})/g, function(P, O, N) {
        var T = N && N.toUpperCase();
        return O || u[N] || n[N] || u[T].replace(/(\[[^\]]+])|(MMMM|MM|DD|dddd)/g, function(C, k, q) {
          return k || q.slice(1);
        });
      })).match(i), _ = v.length, j = 0; j < _; j += 1) {
        var M = v[j], R = x[M], y = R && R[0], $ = R && R[1];
        v[j] = $ ? { regex: y, parser: $ } : M.replace(/^\[|\]$/g, "");
      }
      return function(P) {
        for (var O = {}, N = 0, T = 0; N < _; N += 1) {
          var C = v[N];
          if (typeof C == "string") T += C.length;
          else {
            var k = C.regex, q = C.parser, F = P.slice(T), G = k.exec(F)[0];
            q.call(O, G), P = P.replace(G, "");
          }
        }
        return function(K) {
          var ee = K.afternoon;
          if (ee !== void 0) {
            var z = K.hours;
            ee ? z < 12 && (K.hours += 12) : z === 12 && (K.hours = 0), delete K.afternoon;
          }
        }(O), O;
      };
    }
    return function(c, p, u) {
      u.p.customParseFormat = !0, c && c.parseTwoDigitYear && (r = c.parseTwoDigitYear);
      var v = p.prototype, _ = v.parse;
      v.parse = function(j) {
        var M = j.date, R = j.utc, y = j.args;
        this.$u = R;
        var $ = y[1];
        if (typeof $ == "string") {
          var P = y[2] === !0, O = y[3] === !0, N = P || O, T = y[2];
          O && (T = y[2]), a = this.$locale(), !P && T && (a = u.Ls[T]), this.$d = function(F, G, K, ee) {
            try {
              if (["x", "X"].indexOf(G) > -1) return new Date((G === "X" ? 1e3 : 1) * F);
              var z = g(G)(F), Z = z.year, I = z.month, te = z.day, W = z.hours, H = z.minutes, oe = z.seconds, me = z.milliseconds, A = z.zone, fe = z.week, he = /* @__PURE__ */ new Date(), ae = te || (Z || I ? 1 : he.getDate()), ie = Z || he.getFullYear(), U = 0;
              Z && !I || (U = I > 0 ? I - 1 : he.getMonth());
              var Q, V = W || 0, B = H || 0, ne = oe || 0, ce = me || 0;
              return A ? new Date(Date.UTC(ie, U, ae, V, B, ne, ce + 60 * A.offset * 1e3)) : K ? new Date(Date.UTC(ie, U, ae, V, B, ne, ce)) : (Q = new Date(ie, U, ae, V, B, ne, ce), fe && (Q = ee(Q).week(fe).toDate()), Q);
            } catch {
              return /* @__PURE__ */ new Date("");
            }
          }(M, $, R, u), this.init(), T && T !== !0 && (this.$L = this.locale(T).$L), N && M != this.format($) && (this.$d = /* @__PURE__ */ new Date("")), a = {};
        } else if ($ instanceof Array) for (var C = $.length, k = 1; k <= C; k += 1) {
          y[1] = $[k - 1];
          var q = u.apply(this, y);
          if (q.isValid()) {
            this.$d = q.$d, this.$L = q.$L, this.init();
            break;
          }
          k === C && (this.$d = /* @__PURE__ */ new Date(""));
        }
        else _.call(this, j);
      };
    };
  });
})(at);
var Ut = at.exports;
const Gt = /* @__PURE__ */ pt(Ut);
ge.extend(Gt);
const be = "DD/MM/YYYY", Kt = Y(
  ({
    label: e,
    textAlignment: t = "left",
    inputSize: n = "lg",
    status: i = "neutral",
    isAutocomplete: l = !1,
    fullHeightOnOpen: f = !0,
    className: h,
    actionSheetFooter: b,
    datePickerProps: a = { selectRange: !1 },
    onSelectDate: r,
    onSearch: s,
    disabled: m,
    value: d,
    ...w
  }, x) => {
    const [g, c] = J(), p = re(null), u = re(null), v = re(null), { isOpen: _, open: j, close: M } = pe([
      u,
      v
    ]), { isMobile: R } = Ee(), {
      selectRange: y,
      className: $,
      ...P
    } = a;
    X(() => {
      c(d);
    }, [d]);
    const O = (C) => {
      var k;
      l && _ && ((k = p.current) != null && k.contains(C.target)) || (_ ? M() : j());
    }, N = (C) => {
      const k = C.target.value;
      s == null || s(k), (ge(k, be, !0).isValid() || k === "") && c(k);
    }, T = ye(e);
    return X(() => {
      var C;
      _ && ((C = p.current) == null || C.focus()), g && p.current && (ge(p.current.value, be, !0).isValid() || (p.current.value = g));
    }, [_]), /* @__PURE__ */ o.jsxs(
      "div",
      {
        className: L(
          "datepicker__wrapper",
          _ && "datepicker__wrapper--open",
          m && "datepicker__wrapper--disabled"
        ),
        ref: u,
        children: [
          /* @__PURE__ */ o.jsx(
            "div",
            {
              onClick: O,
              ref: x,
              "data-testid": "input-container",
              children: /* @__PURE__ */ o.jsx(
                xe,
                {
                  leftIcon: /* @__PURE__ */ o.jsx(
                    Ft,
                    {
                      iconSize: "sm",
                      fill: "var(--semantic-color-monochrome-textIcon-normal-high)"
                    }
                  ),
                  ref: p,
                  "data-testid": "calendar-input",
                  label: T,
                  dropdown: !0,
                  disabled: m,
                  textAlignment: t,
                  inputSize: n,
                  status: i,
                  isDropdownOpen: _,
                  readOnly: !l,
                  value: g,
                  className: L("datepicker__input", h),
                  onChange: N,
                  formatProps: {
                    format: "##/##/####",
                    mask: "_"
                  },
                  ...w
                }
              )
            }
          ),
          /* @__PURE__ */ o.jsx(
            "div",
            {
              className: L(
                "datepicker-dropdown__menu",
                _ ? "datepicker-dropdown__menu--open" : "datepicker-dropdown__menu--close",
                `datepicker-dropdown__menu--${t}`
              ),
              children: R ? /* @__PURE__ */ o.jsx(
                D.Root,
                {
                  isOpen: _,
                  onClose: () => M(),
                  children: /* @__PURE__ */ o.jsxs(
                    D.Portal,
                    {
                      shouldCloseOnDrag: !0,
                      fullHeightOnOpen: f,
                      ref: v,
                      children: [
                        e && /* @__PURE__ */ o.jsx(D.Header, { title: e }),
                        /* @__PURE__ */ o.jsx(D.Content, { children: /* @__PURE__ */ o.jsx(
                          Ge,
                          {
                            "data-testid": "calendar",
                            value: g && ge(g, be).toDate(),
                            hasFixedWidth: !0,
                            onFormattedDate: (C) => {
                              c(C), r == null || r(
                                ge(
                                  C,
                                  be
                                ).toDate()
                              );
                            },
                            onChange: (C) => {
                              r == null || r(C), M();
                            },
                            className: L(
                              "datepicker__container",
                              $
                            ),
                            selectRange: y,
                            ...P
                          }
                        ) }),
                        /* @__PURE__ */ o.jsx(D.Footer, { ...b })
                      ]
                    }
                  )
                }
              ) : _ && /* @__PURE__ */ o.jsx(
                Ge,
                {
                  "data-testid": "calendar",
                  value: g && ge(g, be).toDate(),
                  hasFixedWidth: !0,
                  onFormattedDate: (C) => {
                    c(C), r == null || r(
                      ge(C, be).toDate()
                    );
                  },
                  onChange: (C) => {
                    r == null || r(C), M();
                  },
                  className: L(
                    "datepicker__container",
                    $
                  ),
                  selectRange: y,
                  ...P
                }
              )
            }
          )
        ]
      }
    );
  }
), qr = Y(({ ...e }, t) => /* @__PURE__ */ o.jsx(Pe, { children: /* @__PURE__ */ o.jsx(Kt, { ref: t, ...e }) })), Xt = ({
  value: e,
  selectedText: t,
  ...n
}) => {
  const { isOpen: i, setSelectedValue: l } = pe(), f = re(null);
  return X(() => {
    e && l(e);
  }, [e]), X(() => {
    var h;
    i && ((h = f.current) == null || h.focus());
  }, [i]), /* @__PURE__ */ o.jsx(
    xe,
    {
      ref: f,
      isDropdownOpen: i,
      value: t,
      ...n
    }
  );
}, Zt = ({
  item: e,
  onSelectOption: t,
  setSelectedText: n,
  closeOnItemClick: i,
  ...l
}) => {
  const { selectedValue: f, setSelectedValue: h, close: b } = pe(), a = () => {
    e.value && (h(e.value), n(ye(e.text)), t == null || t(e.value), i && b());
  };
  return /* @__PURE__ */ o.jsx(
    nt,
    {
      selected: (f == null ? void 0 : f.toString()) === e.value,
      onClick: a,
      ...l
    }
  );
}, $r = Y(
  ({
    disabled: e,
    label: t,
    options: n,
    textAlignment: i = "left",
    inputSize: l = "lg",
    status: f = "neutral",
    name: h,
    wrapperClassName: b,
    listHeight: a,
    onSearch: r,
    onSelectOption: s,
    value: m,
    isAutocomplete: d = !1,
    fullHeightOnOpen: w = !1,
    closeOnItemClick: x = !0,
    containerClassName: g,
    noActionSheet: c,
    actionSheetDropdown: p,
    ...u
  }, v) => {
    const [_, j] = J(n), [M, R] = J(""), { isMobile: y } = Ee();
    X(() => {
      j(n);
    }, [n]), X(() => {
      if (m) {
        const P = _.find((O) => O.value === m);
        P && R(ye(P.text));
      }
    }, [m]);
    const $ = !y && !p ? L(
      "dropdown__container",
      a || "dropdown__container--height",
      `dropdown__container--size-${l}`
    ) : void 0;
    return /* @__PURE__ */ o.jsx(
      Ae,
      {
        className: L("dropdown__wrapper", b),
        fullHeightOnOpen: w,
        ref: v,
        isAutocomplete: d,
        containerClassName: g,
        noActionSheet: c,
        actionSheetDropdown: p,
        headComponent: /* @__PURE__ */ o.jsx(
          Xt,
          {
            "data-testid": "dropdown-input",
            disabled: e,
            label: ye(t),
            name: h,
            dropdown: !0,
            textAlignment: i,
            inputSize: l,
            status: f,
            onChange: (P) => {
              const O = P.target.value, N = n.filter(
                (T) => {
                  var C;
                  return (C = T.value) == null ? void 0 : C.toLowerCase().includes(O.toLowerCase());
                }
              );
              j(
                N.length > 0 ? N : n
              ), r == null || r(O);
            },
            readOnly: !d,
            type: "select",
            value: m,
            selectedText: M,
            ...u
          }
        ),
        children: /* @__PURE__ */ o.jsx("div", { className: $, children: _.map((P) => /* @__PURE__ */ o.jsx(
          Zt,
          {
            item: P,
            label: P.text,
            size: l,
            setSelectedText: R,
            textAlignment: i,
            onSelectOption: s,
            closeOnItemClick: x
          },
          P.value
        )) })
      }
    );
  }
), Qt = ({
  addOnVariant: e = "outline",
  addOnStatus: t = "neutral",
  codeIcon: n = !0,
  size: i = "md",
  codeLabel: l,
  disabled: f,
  fillAddonBorderColor: h
}) => {
  const { isOpen: b, selectedValue: a } = pe(), r = e === "fill" ? { borderColor: h } : {};
  return /* @__PURE__ */ o.jsxs(
    "div",
    {
      className: L(
        "quill-country-code",
        `quill-country-code__variant--${e}`,
        `quill-country-code__size--${i}`,
        b && "quill-country-code__opened",
        f && "quill-country-code__disabled"
      ),
      style: r,
      children: [
        n && /* @__PURE__ */ o.jsx(
          At,
          {
            className: "mobile-icon",
            iconSize: "sm"
          }
        ),
        /* @__PURE__ */ o.jsxs("div", { children: [
          l && i === "lg" && /* @__PURE__ */ o.jsx(
            Oe,
            {
              className: L(
                "quill-country-code__label",
                `quill-country-code__label--${t}`
              ),
              children: l
            }
          ),
          /* @__PURE__ */ o.jsx(
            We,
            {
              className: L(
                "quill-country-code__value",
                b && "focus"
              ),
              size: i === "lg" ? "md" : i,
              children: a
            }
          )
        ] }),
        /* @__PURE__ */ o.jsx(
          Vt,
          {
            iconSize: "sm",
            "data-state": b ? "open" : "close",
            className: L(
              "dropdown__transform",
              b && "dropdown__transform-rotate"
            )
          }
        )
      ]
    }
  );
}, Jt = ({
  options: e,
  code: t,
  showFlags: n,
  containerRef: i,
  headcompRef: l,
  onItemClick: f,
  showSearchBar: h = !1
}) => {
  const [b, a] = J(""), { isMobile: r } = Ee(), s = re(null), { isOpen: m, close: d } = pe();
  !r && m && Pt([l, s], d);
  const w = (u) => ze[u] ? ze[u] : ze.XX, x = b.toLowerCase(), g = e.filter(
    (u) => u.name.toLowerCase().includes(x) || u.short_code.toLowerCase().includes(x) || u.phone_code.includes(b)
  ), c = re(null);
  X(() => {
    m && c.current && setTimeout(() => {
      const u = c.current;
      if (u)
        if (r) {
          const v = document.querySelector(
            ".input-phone-number-search"
          ), _ = (v == null ? void 0 : v.clientHeight) || 0, j = u.closest(
            ".action-sheet--content"
          );
          if (j) {
            const M = j.clientHeight, R = u.offsetTop, y = u.offsetHeight;
            R > M - y ? j.scrollTop = R - M + y + _ : (u.scrollIntoView({
              block: "start",
              behavior: "auto"
            }), j.scrollTop -= _);
          }
        } else {
          const v = document.querySelector(
            ".input-phone-number-search"
          ), _ = (v == null ? void 0 : v.clientHeight) || 0, j = s.current;
          if (j) {
            const M = j.clientHeight, R = u.offsetTop, y = u.offsetHeight, $ = j.scrollHeight - M + y, P = Math.min(R, $);
            j.scrollTop = P - _;
          }
        }
    }, 0);
  }, [m, r]);
  const p = () => /* @__PURE__ */ o.jsx("div", { className: "dropdown-wrapper", children: g.map((u) => {
    const v = u.short_code.toLowerCase() === t.toLowerCase();
    return /* @__PURE__ */ o.jsx(
      nt,
      {
        ref: v ? c : void 0,
        label: `${u.name} (${u.phone_code})`,
        selected: u.short_code.toLowerCase() === t.toLowerCase(),
        onClick: () => {
          f(u), d();
        },
        leftIcon: n && w(u.short_code.toUpperCase())
      },
      u.short_code
    );
  }) });
  return /* @__PURE__ */ o.jsx(o.Fragment, { children: r ? /* @__PURE__ */ o.jsxs(o.Fragment, { children: [
    /* @__PURE__ */ o.jsx("div", { className: "phone-code-search input-phone-number-search", children: /* @__PURE__ */ o.jsx(
      Ke,
      {
        inputSize: "sm",
        variant: "fill",
        autoComplete: "off",
        value: b,
        onChange: (u) => {
          a(u.target.value);
        }
      }
    ) }),
    /* @__PURE__ */ o.jsx(p, {})
  ] }) : /* @__PURE__ */ o.jsxs(
    bt,
    {
      size: "md",
      height: "md",
      portalContainer: i.current,
      className: "quill-custom-dropdown__content",
      ref: s,
      children: [
        h && /* @__PURE__ */ o.jsx("div", { className: "phone-code-search input-phone-number-search", children: /* @__PURE__ */ o.jsx(
          Ke,
          {
            inputSize: "sm",
            variant: "fill",
            autoComplete: "off",
            value: b,
            onChange: (u) => {
              a(u.target.value);
            }
          }
        ) }),
        /* @__PURE__ */ o.jsx(p, {})
      ]
    }
  ) });
}, eo = Y(
  ({
    inputSize: e = "lg",
    variant: t = "outline",
    status: n = "neutral",
    codeIcon: i = !0,
    codeLabel: l,
    fillAddonBorderColor: f,
    countryCodes: h,
    shortCode: b,
    onCodeChange: a,
    onValueChange: r,
    showFlags: s = !0,
    showSearchBar: m = !1,
    value: d,
    disabled: w,
    onChange: x,
    ...g
  }, c) => {
    const { isOpen: p } = pe(), u = () => h.find(
      (N) => N.short_code.toLowerCase() === (b == null ? void 0 : b.toLowerCase())
    ) || h[0], [v, _] = J(u().phone_code), [j, M] = J(d || ""), R = re(null), y = re(null);
    b = u().short_code, X(() => {
      M(d || "");
    }, [d]), X(() => {
      r == null || r(`${v}${j}`);
    }, [v, j]);
    const $ = (N) => {
      _(N.phone_code), a == null || a(N);
    }, P = (N) => {
      N.target.value = N.target.value.replace(/\s+/g, ""), M(N.target.value), x == null || x(N);
    }, O = /* @__PURE__ */ o.jsx("div", { className: "quill-phone-input-addon", ref: y, children: /* @__PURE__ */ o.jsx(
      Ae,
      {
        value: v,
        label: l,
        headComponent: /* @__PURE__ */ o.jsx(
          Qt,
          {
            addOnVariant: t,
            codeIcon: i,
            size: e,
            codeLabel: l,
            addOnStatus: n,
            fillAddonBorderColor: f,
            disabled: w
          }
        ),
        noAutoClose: !0,
        withProvider: !1,
        disabled: w,
        children: /* @__PURE__ */ o.jsx(
          Jt,
          {
            options: h,
            code: b,
            showFlags: s,
            showSearchBar: m,
            containerRef: R,
            headcompRef: y,
            onItemClick: $
          }
        )
      }
    ) });
    return /* @__PURE__ */ o.jsx("div", { ref: R, className: "quill-phone-input__container", children: /* @__PURE__ */ o.jsx(
      xe,
      {
        type: "tel",
        inputMode: "numeric",
        value: j,
        addOn: O,
        inputSize: e,
        status: n,
        variant: t,
        onChange: P,
        isDropdownOpen: p,
        disabled: w,
        ...g,
        ref: c
      }
    ) });
  }
), Sr = Y(({ ...e }, t) => /* @__PURE__ */ o.jsx(Pe, { children: /* @__PURE__ */ o.jsx(eo, { ...e, ref: t }) })), Lr = Y(
  ({
    status: e = "neutral",
    button_position: t = "right",
    inputSize: n = "md",
    value: i,
    buttonColor: l = "black",
    label: f,
    buttonIconPosition: h,
    buttonDisabled: b = !1,
    buttonLabel: a,
    disabled: r,
    buttonCallback: s,
    onChange: m,
    ...d
  }, w) => {
    const [x, g] = J(!0);
    X(() => {
      g(!i);
    }, [i]);
    const c = e === "neutral" || r, p = `quill-input__wrapper-with_button-${t}--${n}${f ? ` quill-input__wrapper-with_button-${t}--${n}--has-label` : ` quill-input__wrapper-with_button-${t}--${n}--no-label`}`, u = /* @__PURE__ */ o.jsx(
      ue,
      {
        disabled: x || r || b,
        color: l,
        fullWidth: t === "bottom",
        variant: "primary",
        size: n,
        iconPosition: h,
        label: a,
        onClick: () => s == null ? void 0 : s()
      }
    );
    return /* @__PURE__ */ o.jsx(
      xe,
      {
        inputSize: n === "md" ? "lg" : n,
        className: p,
        label: f,
        status: e,
        disabled: r,
        button_position: t,
        showInputButton: c,
        inputButton: u,
        value: i,
        ref: w,
        onChange: (v) => {
          g(!v.target.value), m == null || m(v);
        },
        ...d
      }
    );
  }
), Tr = Y((e, t) => {
  const {
    value: n,
    onChange: i,
    disabled: l,
    textAlignment: f,
    placeholder: h = 0,
    unitLeft: b = "",
    unitRight: a = "",
    inputSize: r = "md",
    decimals: s = 2,
    minusDisabled: m = !1,
    plusDisabled: d = !1,
    noStatusIcon: w,
    ...x
  } = e, g = () => /* @__PURE__ */ o.jsx(
    Ce,
    {
      onClick: () => {
        i == null || i({
          target: {
            value: Ye(Number(n) - 1, s)
          }
        });
      },
      disabled: l || m,
      variant: "tertiary",
      icon: /* @__PURE__ */ o.jsx(qt, {}),
      color: "black-white",
      size: r
    }
  ), c = /* @__PURE__ */ o.jsxs(o.Fragment, { children: [
    f !== "center" && /* @__PURE__ */ o.jsx(g, {}),
    /* @__PURE__ */ o.jsx(
      Ce,
      {
        onClick: () => {
          i == null || i({
            target: {
              value: Ye(Number(n) + 1, s)
            }
          });
        },
        disabled: l || d,
        variant: "tertiary",
        icon: /* @__PURE__ */ o.jsx(St, {}),
        color: "black-white",
        size: r
      }
    )
  ] });
  return /* @__PURE__ */ o.jsx(
    xe,
    {
      ...x,
      decimals: s,
      ref: t,
      disabled: l,
      inputSize: r === "md" ? "lg" : r,
      leftIcon: f === "center" && /* @__PURE__ */ o.jsx(g, {}),
      leftPlaceholder: b,
      onChange: i,
      placeholder: h.toLocaleString("en", {
        minimumFractionDigits: s
      }),
      rightPlaceholder: a,
      textAlignment: f,
      triggerActionIcon: c,
      allowDecimals: !0,
      value: n && Ye(n, s),
      noStatusIcon: w
    }
  );
}), st = Y(
  ({
    data: e,
    children: t,
    className: n,
    values: i = [],
    onClickDropdown: l,
    onValueChange: f,
    containerClassName: h,
    actionSheetFooter: b,
    label: a,
    leftIcon: r,
    rightIcon: s,
    wheelType: m = "Generic",
    is12Hour: d = !0,
    startTimeIn24Format: w,
    selectedTime: x = "00:00",
    container: g = it,
    containerHeight: c = "100%",
    hoursInterval: p = 1,
    minutesInterval: u = 1,
    ...v
  }, _) => {
    const j = re(null), M = re(null), [R, y] = J(i), [$, P] = J(!1), [O, N] = J(x);
    let T;
    const C = (I, te) => {
      m === "Generic" && y((W) => {
        const H = [...W];
        return H[I] = te, H;
      });
    }, { isOpen: k, open: q, close: F, selectedValue: G, setSelectedValue: K } = pe([j, M]), { isMobile: ee } = Ee();
    m === "Generic" && (T = i.reduce(
      (I, te, W) => {
        if (!(e != null && e[W])) return I;
        const H = e == null ? void 0 : e[W].find(
          (oe) => oe.value === te
        );
        return W === 0 ? ye(H == null ? void 0 : H.label) || (H == null ? void 0 : H.value) : `${I ?? ""} ${ye(H == null ? void 0 : H.label) || (H == null ? void 0 : H.value)}`;
      },
      ""
    )), m === "Time" && (T = (/* @__PURE__ */ new Date(
      `1/1/1 ${O}`
    )).toLocaleTimeString(v.locale || navigator.language, {
      hour: "2-digit",
      minute: "2-digit",
      hour12: d
    }));
    const z = (I) => {
      l == null || l(I), k ? F() : q();
    }, Z = (I) => {
      I.key === ke.ARROW_DOWN && q();
    };
    return X(() => {
      m === "Generic" && (f == null || f(R)), m === "Time" && (f == null || f([G || ""]));
    }, [G]), /* @__PURE__ */ o.jsx(o.Fragment, { children: /* @__PURE__ */ o.jsxs(
      "div",
      {
        ref: j,
        className: L(
          "quill-generic-picker__container",
          h
        ),
        children: [
          /* @__PURE__ */ o.jsx(
            "div",
            {
              ref: _,
              onClick: z,
              onKeyDown: Z,
              onMouseDown: () => P(!0),
              onMouseUp: () => P(!1),
              onMouseLeave: () => P(!1),
              children: /* @__PURE__ */ o.jsx(
                xe,
                {
                  dropdown: !0,
                  isDropdownOpen: k,
                  value: G ?? T,
                  autoFocus: !1,
                  autoComplete: "off",
                  readOnly: !0,
                  className: L(
                    "quill-generic-picker__input",
                    `quill-generic-picker__input--is-open-${k && !$}--${v.variant}--${v.status}`,
                    n
                  ),
                  type: "text",
                  label: a,
                  leftIcon: r,
                  rightIcon: s,
                  ...v
                }
              )
            }
          ),
          /* @__PURE__ */ o.jsx(
            "div",
            {
              className: L(
                "quill-generic-picker__content--container",
                k && "quill-generic-picker__content--container__is-open"
              ),
              children: ee ? /* @__PURE__ */ o.jsx(D.Root, { isOpen: k, onClose: F, children: /* @__PURE__ */ o.jsxs(
                D.Portal,
                {
                  shouldCloseOnDrag: !0,
                  fullHeightOnOpen: !0,
                  ref: M,
                  children: [
                    a && /* @__PURE__ */ o.jsx(D.Header, { title: a }),
                    /* @__PURE__ */ o.jsx(D.Content, { className: "quill-generic-picker__content--hide-scrollbar", children: /* @__PURE__ */ o.jsx(
                      g,
                      {
                        data: e,
                        inputValues: R,
                        close: F,
                        setSelectedValue: K,
                        is12Hour: d,
                        setInputValues: (I, te) => C(
                          I,
                          te
                        ),
                        selectedTime: O,
                        startTimeIn24Format: w,
                        setSelectedTime: N,
                        containerHeight: c,
                        hoursInterval: p,
                        minutesInterval: u,
                        ...v,
                        children: t
                      }
                    ) }),
                    /* @__PURE__ */ o.jsx(
                      D.Footer,
                      {
                        ...b
                      }
                    )
                  ]
                }
              ) }) : /* @__PURE__ */ o.jsx(
                "div",
                {
                  className: L(
                    "quill-generic-picker__content",
                    k && "quill-generic-picker__content__is-open"
                  ),
                  children: /* @__PURE__ */ o.jsx(
                    g,
                    {
                      data: e,
                      inputValues: R,
                      close: F,
                      setSelectedValue: K,
                      is12Hour: d,
                      setInputValues: (I, te) => C(I, te),
                      selectedTime: O,
                      startTimeIn24Format: w,
                      setSelectedTime: N,
                      containerHeight: c,
                      hoursInterval: p,
                      minutesInterval: u,
                      ...v,
                      children: t
                    }
                  )
                }
              )
            }
          )
        ]
      }
    ) });
  }
), Ir = Y(({ children: e, ...t }, n) => /* @__PURE__ */ o.jsx(Pe, { children: /* @__PURE__ */ o.jsx(st, { ref: n, ...t, container: it, children: e }) })), Fr = Y(({ children: e, ...t }, n) => /* @__PURE__ */ o.jsx(Pe, { children: /* @__PURE__ */ o.jsx(
  st,
  {
    ref: n,
    ...t,
    wheelType: "Time",
    container: _t,
    children: e
  }
) })), to = {
  sm: "sm",
  md: "sm",
  lg: "md",
  xl: "md"
}, oo = Y(
  ({
    options: e,
    checkbox: t,
    closeContentOnClick: n,
    contentTitle: i,
    contentHeight: l = "sm",
    contentClass: f,
    actionSheetFooter: h,
    size: b = "md",
    onSelectionChange: a,
    onOpen: r,
    contentAlign: s,
    onClose: m,
    contentCenter: d = !1,
    ...w
  }, x) => {
    const [g, c] = J(e);
    X(() => {
      c(e);
    }, [e]);
    const p = (u) => {
      t && c((v) => {
        const _ = v.map(
          (j) => j.id === u ? { ...j, selected: !j.selected } : j
        );
        return a == null || a(_), _;
      });
    };
    return /* @__PURE__ */ o.jsx(
      Ae,
      {
        headComponent: /* @__PURE__ */ o.jsx(Mt, { size: b, ...w, ref: x }),
        onOpen: r,
        onClose: m,
        contentAlign: s,
        actionSheetFooter: h,
        label: i,
        fullHeightOnOpen: !1,
        contentClassName: "quill__dropdown-button__content",
        children: /* @__PURE__ */ o.jsx(
          Et,
          {
            checkbox: t,
            closeContentOnClick: n,
            label: i,
            height: l,
            size: to[b],
            className: f,
            onItemClick: p,
            options: g,
            centered: d
          }
        )
      }
    );
  }
);
oo.displayName = "DropdownButton";
const ro = ({
  className: e,
  title: t,
  description: n,
  icon: i,
  iconPosition: l = "right",
  closeIcon: f,
  centered: h = !0,
  ...b
}) => {
  const { expandable: a } = Re(qe), { handleClose: r } = Re(qe);
  return /* @__PURE__ */ o.jsxs(
    "div",
    {
      className: L(
        "quill-action-sheet--header",
        `quill-action-sheet--header__expandable--${a}`,
        e
      ),
      "data-testid": "action-sheet-header",
      ...b,
      children: [
        /* @__PURE__ */ o.jsxs("div", { className: "quill-action-sheet--title", children: [
          i && /* @__PURE__ */ o.jsx(
            "div",
            {
              className: L(
                "quill-action-sheet--title--icon",
                `quill-action-sheet--title--icon--${l}`
              ),
              children: i
            }
          ),
          t && /* @__PURE__ */ o.jsx(
            ut.H5,
            {
              centered: h,
              className: "quill-action-sheet--title-text",
              children: t
            }
          ),
          f && /* @__PURE__ */ o.jsx(
            Ce,
            {
              color: "black-white",
              icon: f,
              className: L(
                "quill-action-sheet--title--icon",
                "quill-action-sheet--title--icon--close"
              ),
              onClick: r,
              size: "md",
              variant: "tertiary"
            }
          )
        ] }),
        n && /* @__PURE__ */ o.jsx(
          We,
          {
            centered: h,
            className: "quill-action-sheet--description",
            children: n
          }
        )
      ]
    }
  );
}, no = ({
  primaryAction: e,
  secondaryAction: t,
  alignment: n,
  className: i,
  shouldCloseOnPrimaryButtonClick: l = !0,
  shouldCloseOnSecondaryButtonClick: f = !0,
  isPrimaryButtonDisabled: h,
  isSecondaryButtonDisabled: b,
  primaryButtonColor: a = "black-white",
  ...r
}) => {
  const { handleClose: s } = Re(qe);
  if (!e && !t) return null;
  const m = () => {
    e == null || e.onAction(), l && (s == null || s());
  }, d = () => {
    t == null || t.onAction(), f && (s == null || s());
  };
  return /* @__PURE__ */ o.jsxs(
    "div",
    {
      className: L(
        "quill-action-sheet--footer",
        `quill-action-sheet--footer__variant--${n}`,
        i
      ),
      ...r,
      children: [
        e && /* @__PURE__ */ o.jsx(
          ue,
          {
            onClick: m,
            color: a,
            size: "lg",
            label: e.content,
            fullWidth: !0,
            disabled: h
          }
        ),
        t && /* @__PURE__ */ o.jsx(
          ue,
          {
            onClick: d,
            variant: "secondary",
            color: "black-white",
            size: "lg",
            label: t.content,
            fullWidth: !0,
            disabled: b
          }
        )
      ]
    }
  );
}, lt = ({
  iconComponent: e,
  iconClassName: t,
  ...n
}) => {
  const { handleOpen: i } = Re(qe);
  return e ? /* @__PURE__ */ o.jsx(ue, { onClick: i, ...n, children: /* @__PURE__ */ o.jsx(
    e,
    {
      "data-testid": "dt-actionsheet-icon-button",
      className: t
    }
  ) }) : /* @__PURE__ */ o.jsx(ue, { onClick: i, ...n });
};
lt.displayName = "Trigger";
const D = ({
  children: e
}) => /* @__PURE__ */ o.jsx(o.Fragment, { children: e });
D.Root = ft;
D.Header = ro;
D.Content = ht;
D.Footer = no;
D.HandleBar = vt;
D.Trigger = lt;
D.Portal = gt;
const ct = (e) => {
  const {
    isOpened: t = !1,
    className: n,
    children: i,
    showCrossIcon: l = !0,
    showPrimaryButton: f = !0,
    showSecondaryButton: h = !1,
    shouldCloseOnPrimaryButtonClick: b = !1,
    shouldCloseOnSecondaryButtonClick: a = !1,
    shouldCloseModalOnSwipeDown: r = !1,
    toggleModal: s,
    disableCloseOnOverlay: m = !1,
    hasFooter: d = !0,
    portalId: w,
    primaryButtonLabel: x,
    isPrimaryButtonDisabled: g,
    isSecondaryButtonDisabled: c,
    primaryButtonCallback: p,
    secondaryButtonCallback: u,
    secondaryButtonLabel: v,
    showHandleBar: _,
    isMobile: j,
    isNonExpandable: M = !1,
    buttonColor: R = "black-white",
    handleBarIndex: y,
    ...$
  } = e, [P, O] = J(t), N = w && document.getElementById(w) || document.getElementById("modal-root") || document.body;
  X(() => {
    O(t);
  }, [t]);
  const T = () => {
    p == null || p(), b && k();
  }, C = () => {
    u == null || u(), a && k();
  }, k = () => {
    if (s && P) return s(!1);
    O(!1);
  }, q = () => /* @__PURE__ */ o.jsxs(o.Fragment, { children: [
    /* @__PURE__ */ o.jsx("div", { className: "quill-modal__content-wrapper", children: i }),
    d && /* @__PURE__ */ o.jsxs("div", { className: "quill-modal__button-wrapper", children: [
      f && /* @__PURE__ */ o.jsx(
        ue,
        {
          color: R,
          fullWidth: !0,
          size: "lg",
          label: x,
          disabled: g,
          onClick: T
        }
      ),
      h && /* @__PURE__ */ o.jsx(
        ue,
        {
          color: "black-white",
          fullWidth: !0,
          size: "lg",
          label: v,
          disabled: c,
          variant: "secondary",
          className: "quill-modal__button",
          onClick: C
        }
      )
    ] })
  ] });
  return !j && P ? rt.createPortal(
    /* @__PURE__ */ o.jsx(
      "div",
      {
        ...$,
        className: "quill-modal__background",
        "data-testid": "dt_overlay",
        onClick: () => {
          m || k();
        },
        children: /* @__PURE__ */ o.jsxs(
          "div",
          {
            className: L("quill-modal__container", n),
            onClick: (F) => F.stopPropagation(),
            children: [
              l && /* @__PURE__ */ o.jsx(
                Ce,
                {
                  variant: "tertiary",
                  color: "black-white",
                  size: "lg",
                  className: "quill-modal__close-icon",
                  onClick: k,
                  icon: /* @__PURE__ */ o.jsx(Tt, {})
                }
              ),
              /* @__PURE__ */ o.jsx(q, {})
            ]
          }
        )
      }
    ),
    N
  ) : /* @__PURE__ */ o.jsx(
    D.Root,
    {
      expandable: !M,
      isOpen: t,
      onClose: k,
      children: /* @__PURE__ */ o.jsx(
        D.Portal,
        {
          ...$,
          showHandlebar: _,
          handleBarPosition: "absolute",
          disableCloseOnOverlay: m,
          portalId: w,
          shouldCloseOnDrag: r,
          handleBarIndex: y,
          children: /* @__PURE__ */ o.jsx(q, {})
        }
      )
    }
  );
};
ct.Header = jt;
ct.Body = yt;
var Be = {}, $e = {};
Object.defineProperty($e, "__esModule", { value: !0 });
$e.PopoverPortal = void 0;
var io = we, ao = rt, so = function(e) {
  var t = e.container, n = e.element, i = e.scoutElement, l = e.children;
  return (0, io.useLayoutEffect)(function() {
    return t.appendChild(n), t.appendChild(i), function() {
      t.removeChild(n), t.removeChild(i);
    };
  }, [t, n, i]), (0, ao.createPortal)(l, n);
};
$e.PopoverPortal = so;
var Se = {};
(function(e) {
  var t = le && le.__assign || function() {
    return t = Object.assign || function(a) {
      for (var r, s = 1, m = arguments.length; s < m; s++) {
        r = arguments[s];
        for (var d in r) Object.prototype.hasOwnProperty.call(r, d) && (a[d] = r[d]);
      }
      return a;
    }, t.apply(this, arguments);
  };
  Object.defineProperty(e, "__esModule", { value: !0 }), e.getNudgedPopoverRect = e.getNewPopoverRect = e.popoverRectForPosition = e.createContainer = e.rectsAreEqual = e.createRect = e.EMPTY_RECT = void 0, e.EMPTY_RECT = {
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    width: 0,
    height: 0
  };
  var n = function(a) {
    var r = a.top, s = a.left, m = a.width, d = a.height;
    return {
      top: r,
      left: s,
      width: m,
      height: d,
      right: s + m,
      bottom: r + d
    };
  };
  e.createRect = n;
  var i = function(a, r) {
    return a === r || (a == null ? void 0 : a.bottom) === (r == null ? void 0 : r.bottom) && (a == null ? void 0 : a.height) === (r == null ? void 0 : r.height) && (a == null ? void 0 : a.left) === (r == null ? void 0 : r.left) && (a == null ? void 0 : a.right) === (r == null ? void 0 : r.right) && (a == null ? void 0 : a.top) === (r == null ? void 0 : r.top) && (a == null ? void 0 : a.width) === (r == null ? void 0 : r.width);
  };
  e.rectsAreEqual = i;
  var l = function(a) {
    var r = a.containerStyle, s = a.containerClassName, m = a.id, d = window.document.createElement("div");
    return d.id = m, s && (d.className = s), Object.assign(d.style, r), d;
  };
  e.createContainer = l;
  var f = function(a, r, s, m, d) {
    var w = r.left + r.width / 2, x = r.top + r.height / 2, g = s.width, c = s.height, p, u;
    switch (a) {
      case "left":
        p = x - c / 2, u = r.left - m - g, d === "start" && (p = r.top), d === "end" && (p = r.bottom - c);
        break;
      case "bottom":
        p = r.bottom + m, u = w - g / 2, d === "start" && (u = r.left), d === "end" && (u = r.right - g);
        break;
      case "right":
        p = x - c / 2, u = r.right + m, d === "start" && (p = r.top), d === "end" && (p = r.bottom - c);
        break;
      default:
        p = r.top - c - m, u = w - g / 2, d === "start" && (u = r.left), d === "end" && (u = r.right - g);
        break;
    }
    return (0, e.createRect)({ left: u, top: p, width: g, height: c });
  };
  e.popoverRectForPosition = f;
  var h = function(a, r) {
    var s = a.position, m = a.align, d = a.childRect, w = a.popoverRect, x = a.boundaryRect, g = a.padding, c = a.reposition, p = (0, e.popoverRectForPosition)(s, d, w, g, m), u = c && (s === "top" && p.top < x.top + r || s === "left" && p.left < x.left + r || s === "right" && p.right > x.right - r || s === "bottom" && p.bottom > x.bottom - r);
    return {
      rect: p,
      boundaryViolation: u
    };
  };
  e.getNewPopoverRect = h;
  var b = function(a, r, s) {
    var m = r.top + s, d = r.left + s, w = r.right - s, x = r.bottom - s, g = a.top < m ? m : a.top;
    g = g + a.height > x ? x - a.height : g;
    var c = a.left < d ? d : a.left;
    return c = c + a.width > w ? w - a.width : c, (0, e.createRect)(t(t({}, a), { top: g, left: c }));
  };
  e.getNudgedPopoverRect = b;
})(Se);
var Le = {}, Te = {};
Object.defineProperty(Te, "__esModule", { value: !0 });
Te.useElementRef = void 0;
var Ze = we, lo = Se, co = function(e) {
  var t = e.containerClassName, n = e.containerStyle, i = e.id, l = (0, Ze.useRef)(), f = (0, Ze.useMemo)(function() {
    return (0, lo.createContainer)({ containerStyle: n, containerClassName: t, id: i });
  }, [t, n, i]);
  return l.current = f, l;
};
Te.useElementRef = co;
Object.defineProperty(Le, "__esModule", { value: !0 });
Le.usePopover = void 0;
var uo = we, _e = Se, Qe = Te, po = {
  position: "fixed",
  overflow: "visible",
  top: "0px",
  left: "0px"
}, mo = {
  position: "fixed",
  top: "0px",
  left: "0px",
  width: "0px",
  height: "0px",
  visibility: "hidden"
}, fo = function(e) {
  var t = e.isOpen, n = e.childRef, i = e.positions, l = e.containerClassName, f = e.parentElement, h = e.transform, b = e.transformMode, a = e.align, r = e.padding, s = e.reposition, m = e.boundaryInset, d = e.boundaryElement, w = e.onPositionPopover, x = (0, Qe.useElementRef)({ id: "react-tiny-popover-scout", containerStyle: mo }), g = (0, Qe.useElementRef)({
    id: "react-tiny-popover-container",
    containerClassName: l,
    containerStyle: po
  }), c = (0, uo.useCallback)(function(p) {
    var u, v, _ = p === void 0 ? {} : p, j = _.positionIndex, M = j === void 0 ? 0 : j, R = _.parentRect, y = R === void 0 ? f.getBoundingClientRect() : R, $ = _.childRect, P = $ === void 0 ? (u = n == null ? void 0 : n.current) === null || u === void 0 ? void 0 : u.getBoundingClientRect() : $, O = _.scoutRect, N = O === void 0 ? (v = x == null ? void 0 : x.current) === null || v === void 0 ? void 0 : v.getBoundingClientRect() : O, T = _.popoverRect, C = T === void 0 ? g.current.getBoundingClientRect() : T, k = _.boundaryRect, q = k === void 0 ? d === f ? y : d.getBoundingClientRect() : k;
    if (!(!P || !y || !t)) {
      if (h && b === "absolute") {
        var F = typeof h == "function" ? h({
          childRect: P,
          popoverRect: C,
          parentRect: y,
          boundaryRect: q,
          padding: r,
          align: a,
          nudgedTop: 0,
          nudgedLeft: 0,
          boundaryInset: m,
          violations: _e.EMPTY_RECT,
          hasViolations: !1
        }) : h, G = F.top, K = F.left, ee = Math.round(y.left + K - N.left), z = Math.round(y.top + G - N.top);
        g.current.style.transform = "translate(".concat(ee, "px, ").concat(z, "px)"), w({
          childRect: P,
          popoverRect: (0, _e.createRect)({
            left: ee,
            top: z,
            width: C.width,
            height: C.height
          }),
          parentRect: y,
          boundaryRect: q,
          padding: r,
          align: a,
          transform: { top: G, left: K },
          nudgedTop: 0,
          nudgedLeft: 0,
          boundaryInset: m,
          violations: _e.EMPTY_RECT,
          hasViolations: !1
        });
        return;
      }
      var Z = M === i.length, I = Z ? i[0] : i[M], te = (0, _e.getNewPopoverRect)({
        childRect: P,
        popoverRect: C,
        boundaryRect: q,
        position: I,
        align: a,
        padding: r,
        reposition: s
      }, m), W = te.rect, H = te.boundaryViolation;
      if (H && s && !Z) {
        c({
          positionIndex: M + 1,
          childRect: P,
          popoverRect: C,
          parentRect: y,
          boundaryRect: q
        });
        return;
      }
      var oe = W.top, me = W.left, A = W.width, fe = W.height, he = s && !Z, ae = (0, _e.getNudgedPopoverRect)(W, q, m), ie = ae.left, U = ae.top, Q = oe, V = me;
      he && (Q = U, V = ie), Q = Math.round(Q - N.top), V = Math.round(V - N.left), g.current.style.transform = "translate(".concat(V, "px, ").concat(Q, "px)");
      var B = {
        top: q.top + m - Q,
        left: q.left + m - V,
        right: V + A - q.right + m,
        bottom: Q + fe - q.bottom + m
      }, ne = {
        childRect: P,
        popoverRect: (0, _e.createRect)({ left: V, top: Q, width: A, height: fe }),
        parentRect: y,
        boundaryRect: q,
        position: I,
        align: a,
        padding: r,
        nudgedTop: U - oe,
        nudgedLeft: ie - me,
        boundaryInset: m,
        violations: {
          top: B.top <= 0 ? 0 : B.top,
          left: B.left <= 0 ? 0 : B.left,
          right: B.right <= 0 ? 0 : B.right,
          bottom: B.bottom <= 0 ? 0 : B.bottom
        },
        hasViolations: B.top > 0 || B.left > 0 || B.right > 0 || B.bottom > 0
      };
      if (h) {
        w(ne);
        var ce = typeof h == "function" ? h(ne) : h, se = ce.top, S = ce.left;
        g.current.style.transform = "translate(".concat(Math.round(V + (S ?? 0)), "px, ").concat(Math.round(Q + (se ?? 0)), "px)"), ne.nudgedLeft += S ?? 0, ne.nudgedTop += se ?? 0, ne.transform = { top: se, left: S };
      }
      w(ne);
    }
  }, [
    f,
    n,
    x,
    g,
    d,
    t,
    h,
    b,
    i,
    a,
    r,
    s,
    m,
    w
  ]);
  return { positionPopover: c, popoverRef: g, scoutRef: x };
};
Le.usePopover = fo;
var Ie = {};
Object.defineProperty(Ie, "__esModule", { value: !0 });
Ie.useMemoizedArray = void 0;
var Je = we, ho = function(e) {
  var t = (0, Je.useRef)(e), n = (0, Je.useMemo)(function() {
    if (t.current === e)
      return t.current;
    if (t.current.length !== e.length)
      return t.current = e, e;
    for (var i = 0; i < e.length; i += 1)
      if (e[i] !== t.current[i])
        return t.current = e, e;
    return t.current;
  }, [e]);
  return n;
};
Ie.useMemoizedArray = ho;
var Ne = {}, Ve = le && le.__assign || function() {
  return Ve = Object.assign || function(e) {
    for (var t, n = 1, i = arguments.length; n < i; n++) {
      t = arguments[n];
      for (var l in t) Object.prototype.hasOwnProperty.call(t, l) && (e[l] = t[l]);
    }
    return e;
  }, Ve.apply(this, arguments);
};
Object.defineProperty(Ne, "__esModule", { value: !0 });
Ne.useArrowContainer = void 0;
var et = we, vo = function(e) {
  var t = e.childRect, n = e.popoverRect, i = e.position, l = e.arrowSize, f = e.arrowColor, h = (0, et.useMemo)(function() {
    return {
      padding: l
    };
  }, [l]), b = (0, et.useMemo)(function() {
    return Ve({ position: "absolute" }, function() {
      var a = l * 2, r = t.top - n.top + t.height / 2 - a / 2, s = t.left - n.left + t.width / 2 - a / 2, m = l, d = n.width - l, w = n.height - l;
      switch (s = s < m ? m : s, s = s + a > d ? d - a : s, r = r < m ? m : r, r = r + a > w ? w - a : r, r = Number.isNaN(r) ? 0 : r, s = Number.isNaN(s) ? 0 : s, i) {
        case "right":
          return {
            borderTop: "".concat(l, "px solid transparent"),
            borderBottom: "".concat(l, "px solid transparent"),
            borderRight: "".concat(l, "px solid ").concat(f),
            left: 0,
            top: r
          };
        case "left":
          return {
            borderTop: "".concat(l, "px solid transparent"),
            borderBottom: "".concat(l, "px solid transparent"),
            borderLeft: "".concat(l, "px solid ").concat(f),
            right: 0,
            top: r
          };
        case "bottom":
          return {
            borderLeft: "".concat(l, "px solid transparent"),
            borderRight: "".concat(l, "px solid transparent"),
            borderBottom: "".concat(l, "px solid ").concat(f),
            top: 0,
            left: s
          };
        case "top":
          return {
            borderLeft: "".concat(l, "px solid transparent"),
            borderRight: "".concat(l, "px solid transparent"),
            borderTop: "".concat(l, "px solid ").concat(f),
            bottom: 0,
            left: s
          };
        default:
          return {
            display: "hidden"
          };
      }
    }());
  }, [
    f,
    l,
    t.height,
    t.left,
    t.top,
    t.width,
    n.height,
    n.left,
    n.top,
    n.width,
    i
  ]);
  return {
    arrowContainerStyle: h,
    arrowStyle: b
  };
};
Ne.useArrowContainer = vo;
var Fe = {}, je = le && le.__assign || function() {
  return je = Object.assign || function(e) {
    for (var t, n = 1, i = arguments.length; n < i; n++) {
      t = arguments[n];
      for (var l in t) Object.prototype.hasOwnProperty.call(t, l) && (e[l] = t[l]);
    }
    return e;
  }, je.apply(this, arguments);
};
Object.defineProperty(Fe, "__esModule", { value: !0 });
Fe.ArrowContainer = void 0;
var tt = o, ot = we, go = Ne, wo = function(e) {
  var t = e.childRect, n = e.popoverRect, i = e.position, l = e.arrowColor, f = e.arrowSize, h = e.arrowClassName, b = e.arrowStyle, a = e.className, r = e.children, s = e.style, m = (0, go.useArrowContainer)({
    childRect: t,
    popoverRect: n,
    position: i,
    arrowColor: l,
    arrowSize: f
  }), d = m.arrowContainerStyle, w = m.arrowStyle, x = (0, ot.useMemo)(function() {
    return je(je({}, d), s);
  }, [d, s]), g = (0, ot.useMemo)(function() {
    return je(je({}, w), b);
  }, [w, b]);
  return (0, tt.jsxs)("div", { className: a, style: x, children: [(0, tt.jsx)("div", { style: g, className: h }), r] });
};
Fe.ArrowContainer = wo;
(function(e) {
  var t = le && le.__assign || function() {
    return t = Object.assign || function(d) {
      for (var w, x = 1, g = arguments.length; x < g; x++) {
        w = arguments[x];
        for (var c in w) Object.prototype.hasOwnProperty.call(w, c) && (d[c] = w[c]);
      }
      return d;
    }, t.apply(this, arguments);
  };
  Object.defineProperty(e, "__esModule", { value: !0 }), e.Popover = e.usePopover = e.ArrowContainer = e.useArrowContainer = void 0;
  var n = o, i = we, l = $e, f = Se, h = Le;
  Object.defineProperty(e, "usePopover", { enumerable: !0, get: function() {
    return h.usePopover;
  } });
  var b = Ie, a = Ne;
  Object.defineProperty(e, "useArrowContainer", { enumerable: !0, get: function() {
    return a.useArrowContainer;
  } });
  var r = Fe;
  Object.defineProperty(e, "ArrowContainer", { enumerable: !0, get: function() {
    return r.ArrowContainer;
  } });
  var s = ["top", "left", "right", "bottom"], m = (0, i.forwardRef)(function(d, w) {
    var x = d.isOpen, g = d.children, c = d.content, p = d.positions, u = p === void 0 ? s : p, v = d.align, _ = v === void 0 ? "center" : v, j = d.padding, M = j === void 0 ? 0 : j, R = d.reposition, y = R === void 0 ? !0 : R, $ = d.parentElement, P = $ === void 0 ? window.document.body : $, O = d.boundaryElement, N = O === void 0 ? P : O, T = d.containerClassName, C = d.containerStyle, k = d.transform, q = d.transformMode, F = q === void 0 ? "absolute" : q, G = d.boundaryInset, K = G === void 0 ? 0 : G, ee = d.onClickOutside, z = d.clickOutsideCapture, Z = z === void 0 ? !1 : z, I = (0, b.useMemoizedArray)(Array.isArray(u) ? u : [u]), te = (0, i.useRef)(!1), W = (0, i.useRef)(), H = (0, i.useRef)(y), oe = (0, i.useRef)(), me = (0, i.useState)({
      align: _,
      nudgedLeft: 0,
      nudgedTop: 0,
      position: I[0],
      padding: M,
      childRect: f.EMPTY_RECT,
      popoverRect: f.EMPTY_RECT,
      parentRect: f.EMPTY_RECT,
      boundaryRect: f.EMPTY_RECT,
      boundaryInset: K,
      violations: f.EMPTY_RECT,
      hasViolations: !1
    }), A = me[0], fe = me[1], he = (0, i.useCallback)(function(S) {
      return fe(S);
    }, []), ae = (0, h.usePopover)({
      isOpen: x,
      childRef: oe,
      containerClassName: T,
      parentElement: P,
      boundaryElement: N,
      transform: k,
      transformMode: F,
      positions: I,
      align: _,
      padding: M,
      boundaryInset: K,
      reposition: y,
      onPositionPopover: he
    }), ie = ae.positionPopover, U = ae.popoverRef, Q = ae.scoutRef;
    (0, i.useLayoutEffect)(function() {
      var S = !0, de = function() {
        var ve, De;
        if (x && S) {
          var Ue = (ve = oe == null ? void 0 : oe.current) === null || ve === void 0 ? void 0 : ve.getBoundingClientRect(), He = (De = U == null ? void 0 : U.current) === null || De === void 0 ? void 0 : De.getBoundingClientRect();
          Ue != null && He != null && (!(0, f.rectsAreEqual)(Ue, A.childRect) || He.width !== A.popoverRect.width || He.height !== A.popoverRect.height || A.padding !== M || A.align !== _ || I !== W.current || y !== H.current) && ie(), I !== W.current && (W.current = I), y !== H.current && (H.current = y), S && window.requestAnimationFrame(de);
        }
        te.current = x;
      };
      return window.requestAnimationFrame(de), function() {
        S = !1;
      };
    }, [
      _,
      x,
      M,
      U,
      A.align,
      A.childRect,
      A.padding,
      A.popoverRect.height,
      A.popoverRect.width,
      ie,
      I,
      y
    ]), (0, i.useEffect)(function() {
      var S = U.current;
      return Object.assign(S.style, C), function() {
        Object.keys(C ?? {}).forEach(function(de) {
          return delete S.style[de];
        });
      };
    }, [C, x, U]);
    var V = (0, i.useCallback)(function(S) {
      var de, ve;
      x && !(!((de = U.current) === null || de === void 0) && de.contains(S.target)) && !(!((ve = oe.current) === null || ve === void 0) && ve.contains(S.target)) && (ee == null || ee(S));
    }, [x, ee, U]), B = (0, i.useCallback)(function() {
      oe.current && window.requestAnimationFrame(function() {
        return ie();
      });
    }, [ie]);
    (0, i.useEffect)(function() {
      var S = P.ownerDocument.body;
      return S.addEventListener("click", V, Z), S.addEventListener("contextmenu", V, Z), S.addEventListener("resize", B), function() {
        S.removeEventListener("click", V, Z), S.removeEventListener("contextmenu", V, Z), S.removeEventListener("resize", B);
      };
    }, [Z, V, B, P]);
    var ne = (0, i.useCallback)(function(S) {
      oe.current = S, w != null && (typeof w == "object" ? w.current = S : typeof w == "function" && w(S));
    }, [w]), ce = function() {
      return (0, i.cloneElement)(g, { ref: ne });
    }, se = function() {
      return x ? (0, n.jsx)(l.PopoverPortal, { element: U.current, scoutElement: Q.current, container: P, children: typeof c == "function" ? c(A) : c }) : null;
    };
    return (0, n.jsxs)(n.Fragment, { children: [ce(), se()] });
  });
  e.Popover = (0, i.forwardRef)(function(d, w) {
    return typeof window > "u" ? d.children : (0, n.jsx)(m, t({}, d, { ref: w }));
  });
})(Be);
const dt = ({
  as: e,
  tooltipContent: t,
  shouldCloseToolTipOnMouseLeave: n = !0,
  tooltipPosition: i,
  variant: l = "base",
  popoverAlign: f = "center",
  title: h,
  actionText: b,
  tooltipColor: a = "var(--component-textIcon-normal-prominent)",
  tooltipAction: r = "link",
  tooltipActionProps: s,
  hasArrow: m = !0,
  children: d,
  ...w
}) => {
  const [x, g] = J(!1), c = () => g(!0), p = () => g(!1), u = e || "div";
  return /* @__PURE__ */ o.jsx(
    Be.Popover,
    {
      isOpen: x,
      positions: i,
      align: f,
      content: ({ position: v, childRect: _, popoverRect: j }) => {
        const M = /* @__PURE__ */ o.jsx(
          "div",
          {
            style: { backgroundColor: a },
            className: L(
              "tooltip-content",
              `tooltip-content__variant-${l}`
            ),
            children: l === "rich" ? /* @__PURE__ */ o.jsxs("div", { className: "tooltip-content__variant-rich-content", children: [
              /* @__PURE__ */ o.jsxs("div", { className: "tooltip-content__variant-rich-text", children: [
                /* @__PURE__ */ o.jsxs("div", { className: "tooltip-content__variant-rich-header", children: [
                  /* @__PURE__ */ o.jsx(
                    Oe,
                    {
                      color: "var(--component-textIcon-inverse-prominent)",
                      bold: !0,
                      className: "tooltip-content__variant-rich-title",
                      children: h
                    }
                  ),
                  /* @__PURE__ */ o.jsx(
                    Ce,
                    {
                      icon: /* @__PURE__ */ o.jsx(kt, { fill: "var(--component-textIcon-inverse-prominent)" }),
                      onClick: () => g(!1),
                      className: "tooltip-content__variant-rich-icon",
                      size: "sm",
                      color: "white-black",
                      variant: "tertiary"
                    }
                  )
                ] }),
                /* @__PURE__ */ o.jsx(Oe, { color: "var(--component-textIcon-inverse-default)", children: t })
              ] }),
              r === "link" ? /* @__PURE__ */ o.jsx(
                mt,
                {
                  color: "white",
                  hasChevron: !0,
                  children: b,
                  ...s
                }
              ) : /* @__PURE__ */ o.jsx(
                ue,
                {
                  color: "white-black",
                  variant: "secondary",
                  ...s,
                  label: b
                }
              )
            ] }) : /* @__PURE__ */ o.jsx(Oe, { color: "var(--component-textIcon-inverse-default)", children: t })
          }
        );
        return m ? /* @__PURE__ */ o.jsx(
          Be.ArrowContainer,
          {
            position: v,
            childRect: _,
            popoverRect: j,
            arrowColor: a,
            arrowSize: 4,
            children: M
          }
        ) : M;
      },
      children: /* @__PURE__ */ o.jsx(
        u,
        {
          onMouseEnter: c,
          onMouseLeave: n && p,
          ...w,
          children: d
        }
      )
    }
  );
};
dt.displayName = "Tooltip";
const Dr = Y(
  ({
    checked: e = !1,
    disabled: t = !1,
    indeterminate: n = !1,
    infoIconMessage: i,
    infoIconClassName: l,
    size: f = "sm",
    label: h,
    labelClassName: b,
    name: a,
    onChange: r,
    className: s,
    checkboxPosition: m = "left",
    ...d
  }, w) => {
    const [x, g] = J(e), [c, p] = J(n);
    X(() => {
      g(e);
    }, [e]), X(() => {
      p(n);
    }, [n]);
    const u = (R) => {
      c ? (p(!c), g(!1)) : g(!x), r == null || r(R);
    }, v = (R) => {
      R.stopPropagation(), u(R);
    }, _ = (R) => {
      R.stopPropagation(), R.key === ke.SPACE && R.preventDefault(), (R.key === ke.ENTER || R.key === ke.SPACE) && u(R);
    }, j = x ? /* @__PURE__ */ o.jsx(
      xt,
      {
        iconSize: "sm",
        className: "quill-checkbox__box-icon"
      }
    ) : /* @__PURE__ */ o.jsx(
      Ht,
      {
        iconSize: "sm",
        className: L(
          "quill-checkbox__box-icon",
          "quill-checkbox__box-icon--pale"
        )
      }
    ), M = f === "sm" ? Rt : Ct;
    return /* @__PURE__ */ o.jsxs(
      "div",
      {
        className: L(
          "quill-checkbox",
          {
            "quill-checkbox--disabled": t
          },
          s
        ),
        children: [
          /* @__PURE__ */ o.jsxs(
            "div",
            {
              className: L(
                "quill-checkbox__wrapper",
                `quill-checkbox__wrapper--${m}`
              ),
              children: [
                /* @__PURE__ */ o.jsx(
                  "input",
                  {
                    ...d,
                    id: d.id ?? a,
                    className: "quill-checkbox__box",
                    type: "checkbox",
                    checked: x,
                    disabled: t,
                    ref: w,
                    name: a,
                    onChange: v,
                    onKeyDown: _
                  }
                ),
                c ? /* @__PURE__ */ o.jsx(
                  zt,
                  {
                    iconSize: "sm",
                    className: "quill-checkbox__box-icon"
                  }
                ) : j
              ]
            }
          ),
          /* @__PURE__ */ o.jsx("label", { htmlFor: d.id ?? a, children: /* @__PURE__ */ o.jsx(
            We,
            {
              size: f,
              as: "span",
              className: L(
                "quill-checkbox__label",
                b
              ),
              children: h
            }
          ) }),
          i && /* @__PURE__ */ o.jsx(
            dt,
            {
              shouldCloseToolTipOnMouseLeave: !0,
              tooltipContent: i,
              children: /* @__PURE__ */ o.jsx(
                M,
                {
                  className: L(
                    "quill-checkbox__info-icon",
                    l
                  )
                }
              )
            }
          )
        ]
      }
    );
  }
);
export {
  D as A,
  Ae as C,
  oo as D,
  kt as F,
  Ir as G,
  ro as H,
  $r as I,
  ct as M,
  Tr as T,
  st as W,
  qr as a,
  Sr as b,
  Lr as c,
  Fr as d,
  Dr as e,
  dt as f,
  Qt as g,
  Jt as h,
  no as i,
  lt as j
};
