// @deriv-com/quill-ui/dist/components/Modal/base/modal-header.js
import { j as o } from "../../../jsx-runtime-DmTjM30f.js";
import { c as e } from "../../../clsx-OuTLNxxd.js";
import "react";
import "../../Typography/base.js";
import { Heading as m } from "../../Typography/heading/index.js";
const p = ({
  className: t,
  height: i,
  image: l,
  src: a,
  style: r,
  title: n
}) => /* @__PURE__ */ o.jsxs(o.Fragment, { children: [
  a && /* @__PURE__ */ o.jsx(
    "div",
    {
      className: e(
        "quill-modal__content-image",
        i || "quill-modal__content-image-size",
        t
      ),
      style: {
        background: `url(${a}) 50% / cover no-repeat`,
        backgroundColor: "var(--colour-green-opacity-100)",
        ...r
      },
      "data-testid": "dt_modal_image"
    }
  ),
  l && /* @__PURE__ */ o.jsx(
    "div",
    {
      style: r,
      className: e("quill-modal__content-image", t),
      children: l
    }
  ),
  n && /* @__PURE__ */ o.jsx(
    m.H4,
    {
      className: e(
        l || a ? "quill-modal__content-title-has-image" : "quill-modal__content-title",
        t
      ),
      children: n
    }
  )
] });
export {
  p as ModalHeader
};
